---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# SNP Annotation

## Libraries
```{r}
library(reactome.db)
library(tidyverse)
library(data.table)
library(WebGestaltR)
library(kableExtra)
library(biomaRt)
library(ReactomePA)
library(gridExtra)
```

## Data
Read in: SNP gene/exon/cds information
```{r}
gwas_snp <- 
  read.csv("Output Files/allsamples_gwas.csv")

fst_sd_genes <- 
  fread("Input Files/allsamples_fst_sd_outlier_genes_snps.txt")

fst_sd_exons <- 
  fread("Input Files/allsamples_fst_sd_outlier_exons_snps.txt")

fst_sd_cds <- 
  fread("Input Files/allsamples_fst_sd_outlier_cds_snps.txt")

fst_sd_fst <- 
  read.csv("Output Files/allsamples_fst_sd_outlierposition.csv")

pv_genes <- 
  fread("Input Files/GSC_HSeal_1.0_HiC_genes.gff3")
```


## Tidy Harbor seal genes
```{r}
pv_genes_tidy <- 
  pv_genes %>% 
  mutate(gene = V9,
         gene = str_split_fixed(gene, ";", 5),
         gene = str_extract(gene[,4], "(?<=to ).*(?=:)")) %>% 
  filter(!is.na(gene))
```


## FST 5 SD

### ID genes
```{r}
fst_sd_genes_tidy <- 
  fst_sd_genes %>% 
  mutate(gene = V9,
         gene = str_split_fixed(gene, ";", 5),
         gene = str_extract(gene[,4], "(?<=to ).*(?=:)")) %>% 
  filter(!is.na(gene)) %>% 
  mutate(classification = 
           ifelse(V4 %in% fst_sd_cds$V4, "cds", 
                  ifelse(V4 %in% fst_sd_exons$V4, "exon", "gene")))

write.csv(fst_sd_genes_tidy, "Output Files/allsamples_fst_sd_snp_geneid.csv")
```

### Over-representation
#### GO Terms
```{r}
go_bp_sd <- 
  WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
              enrichDatabase="geneontology_Biological_Process",
              interestGene = fst_sd_genes_tidy$gene, interestGeneType = "genesymbol", 
              referenceGene = pv_genes_tidy$gene, referenceGeneType = "genesymbol",
              sigMethod = "fdr", fdrThr = 0.05, 
              isOutput = FALSE)

go_cc_sd <- 
  WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
              enrichDatabase="geneontology_Cellular_Component",
              interestGene = fst_sd_genes_tidy$gene, interestGeneType = "genesymbol", 
              referenceGene = pv_genes_tidy$gene, referenceGeneType = "genesymbol",
              sigMethod = "fdr", fdrThr = 0.05, 
              isOutput = FALSE)
go_mf_sd <- 
  WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
              enrichDatabase="geneontology_Molecular_Function",
              interestGene = fst_sd_genes_tidy$gene, interestGeneType = "genesymbol", 
              referenceGene = pv_genes_tidy$gene, referenceGeneType = "genesymbol",
              sigMethod = "fdr", fdrThr = 0.05, 
              isOutput = FALSE)
```

#### KEGG Pathway
```{r}
kegg_sd <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", 
              enrichDatabase = "pathway_KEGG",
              interestGene = fst_sd_genes_tidy$gene, interestGeneType = "genesymbol", 
              referenceGene = pv_genes_tidy$gene, referenceGeneType = "genesymbol",
              minNum = 5, sigMethod = "fdr", isOutput = FALSE) 
kegg_sd %>%
  kable() %>% 
  kable_styling("basic")
```


### Biomart annotation
Each gene was manually annotated with Ensembl identifier to query Ensembl's Biomart
```{r}
snp_ensembl <-
  read.csv("Input Files/allsamples_fst_sd_snp_ensemblid.csv") 

ensembl <- 
  useEnsembl(biomart = "ensembl", 
             dataset = "hsapiens_gene_ensembl")

filters <- 
  listFilters(ensembl)

attributes <- 
  listAttributes(ensembl)
```


#### Entrez description 
```{r}
snp_entrez_description <- 
  getBM(attributes = c("ensembl_gene_id",
                       "entrezgene_description"),
        filter = "ensembl_gene_id",
        values = snp_ensembl$Ensembl.ID,
        mart = ensembl,
        uniqueRows = TRUE)

snp_entrez_description %>% 
  kable() %>% 
  kable_styling("basic")
```

#### GO terms
```{r}
snp_go <- 
  getBM(attributes = c("ensembl_gene_id",
                       "go_id", "name_1006", "definition_1006", "namespace_1003"),
        filter = "ensembl_gene_id",
        values = snp_ensembl$Ensembl.ID,
        mart = ensembl,
        uniqueRows = TRUE)

snp_go_tidy <-
  snp_go %>% 
  group_by(name_1006) %>% 
  summarize(freq = n())

snp_go %>% 
  filter(namespace_1003 == "biological_process") %>% 
  group_by(name_1006) %>% 
  summarize(freq = n()) %>% 
  filter(freq > 3) %>% 
  ggplot(., aes(x = fct_reorder(name_1006, freq, .desc=TRUE), 
                y = freq)) +
  geom_col() +
  theme_bw() +
  labs(x = "Biological Process", y = "Frequency") +
  theme(panel.grid = element_blank()) +
  coord_flip() 

snp_go %>% 
  filter(namespace_1003 == "molecular_function") %>% 
  group_by(name_1006) %>% 
  summarize(freq = n()) %>% 
  filter(freq > 3) %>% 
  ggplot(., aes(x = fct_reorder(name_1006, freq, .desc=TRUE), 
                y = freq)) +
  geom_col() +
  theme_bw() +
  labs(x = "Molecular Function", y = "Frequency") +
  theme(panel.grid = element_blank()) +
  coord_flip() 

snp_go %>% 
  filter(namespace_1003 == "cellular_component") %>% 
  group_by(name_1006) %>% 
  summarize(freq = n()) %>% 
  filter(freq > 3) %>% 
  ggplot(., aes(x = fct_reorder(name_1006, freq, .desc=TRUE), 
                y = freq)) +
  geom_col() +
  theme_bw() +
  labs(x = "Cellular Component", y = "Frequency") +
  theme(panel.grid = element_blank()) +
  coord_flip() 

snp_go_output <- 
  snp_go %>% 
  group_by(name_1006) %>% 
  summarize(freq = n()) %>% 
  filter(freq > 3) %>% 
  filter(!name_1006 == "") %>% 
  merge(., snp_go, by = "name_1006", all.x = TRUE, all.y = FALSE) %>% 
  filter(!duplicated(go_id))

write.csv(snp_go_output, 
          "Output Files/allsamples_fst_sd_snp_goterm.csv", 
          row.names = FALSE)
```

#### Reactome pathways
```{r}
all_reactome <- 
  as.data.frame(reactomePATHID2NAME)

snp_reactome <- 
  getBM(attributes = c("ensembl_gene_id",
                       "reactome"),
        filter = "ensembl_gene_id",
        values = snp_ensembl$Ensembl.ID,
        mart = ensembl,
        uniqueRows = TRUE) %>% 
  filter(!reactome == "") %>% 
  merge(., all_reactome, by.x = "reactome", by.y = "DB_ID") %>% 
  mutate(path_name = gsub("Homo sapiens: ", "", path_name))

snp_reactome %>% 
  group_by(path_name) %>% 
  summarize(freq = n()) %>% 
  filter(freq > 3) %>% 
  ggplot(., aes(x = fct_reorder(path_name, freq, .desc=TRUE), 
                y = freq)) +
  geom_col() +
  theme_bw() +
  labs(x = "Reactome Pathway", y = "Frequeny") +
  theme(panel.grid = element_blank()) +
  coord_flip()
```

#### Reactome Pathway Over-representation
```{r}
# Convert to entrez gene ID
snp_entrez_id <- 
  getBM(attributes = c("ensembl_gene_id", "entrezgene_id"),
        filter = "ensembl_gene_id",
        values = snp_ensembl$Ensembl.ID,
        mart = ensembl,
        uniqueRows = TRUE)

snp_overrep_reactome <- 
  enrichPathway(gene = snp_entrez_id$entrezgene_id,
                organism = "human",
                pvalueCutoff = 0.05,
                pAdjustMethod = "fdr",
                readable = TRUE,
                universe = pv_genes_tidy$gene,
                minGSSize = 2)$result
```

### Exonic, CDS Genes

#### Identify genes
```{r}
fst_sd_cds_tidy <- 
  fst_sd_cds %>%
  merge(., fst_sd_genes_tidy[,c(12:14)], by = "V12") %>% 
  merge(., snp_ensembl, by.x = "gene", by.y = "Gene")

fst_sd_exons_tidy <- 
  fst_sd_exons %>%
  merge(., fst_sd_genes_tidy[,c(12:14)], by = "V12") %>% 
  merge(., snp_ensembl, by.x = "gene", by.y = "Gene")
```

#### GO terms
```{r}
fst_sd_cds_tidy %>% 
  merge(., snp_go, by.x = "Ensembl.ID", by.y = "ensembl_gene_id") %>% 
  dplyr::select(name_1006, namespace_1003) %>% 
  arrange(namespace_1003) %>% 
  kable() %>% 
  kable_styling("basic")

fst_sd_exons_tidy %>% 
  filter(!classification == "cds") %>% 
  merge(., snp_go, by.x = "Ensembl.ID", by.y = "ensembl_gene_id") %>% 
  dplyr::select(name_1006, namespace_1003) %>% 
  filter(!name_1006 == "") %>% 
  group_by(name_1006, namespace_1003) %>% 
  summarize(freq = n()) %>% 
  arrange(namespace_1003) %>% 
  kable() %>% 
  kable_styling("basic")
```

#### Reactome Pathways
```{r}
fst_sd_cds_tidy %>% 
  merge(., snp_reactome, by.x = "Ensembl.ID", by.y = "ensembl_gene_id") %>% 
  dplyr::select(reactome, path_name) %>% 
  arrange(path_name) %>% 
  kable() %>% 
  kable_styling("basic")

fst_sd_exons_tidy %>% 
  filter(!classification == "cds") %>% 
  merge(., snp_reactome, by.x = "Ensembl.ID", by.y = "ensembl_gene_id") %>%
  arrange(path_name) %>% 
  dplyr::select(reactome, path_name) %>% 
  kable() %>% 
  kable_styling("basic")
```


## GWAS
```{r}
gwas_snp <- 
  gwas_snp %>% 
  filter(fdr < 0.05) %>% 
  merge(., fst_sd_genes_tidy, by.x = "BP", by.y = "V4") 

snp_ensembl %>% 
  filter(Gene == "HAPLN1") %>% 
  merge(., snp_go, by.x = "Ensembl.ID", by.y = "ensembl_gene_id") %>% 
  filter(!name_1006 == "") %>% 
  arrange(namespace_1003) %>% 
  dplyr::select(name_1006, namespace_1003) %>% 
  kable() %>% 
  kable_styling("basic")

snp_ensembl %>% 
  filter(Gene == "HAPLN1") %>% 
  merge(., snp_reactome, by.x = "Ensembl.ID", by.y = "ensembl_gene_id") %>% 
  kable() %>% 
  kable_styling("basic")
```

HAPLN1:  
Predicted to enable hyaluronic acid binding activity. Predicted to be an extracellular matrix structural constituent conferring compression resistance. Predicted to be involved in nervous system development; positive regulation of neuroblast proliferation; and skeletal system development.

## Follow up for Discussion

### Innate Immune System
```{r}
innate_go <- 
  snp_go %>% 
  filter(grepl("innate immune|cytokine|pattern recognition receptor", name_1006, 
               ignore.case = TRUE)) %>% 
  merge(., snp_ensembl, by.x = "ensembl_gene_id", by.y = "Ensembl.ID") %>% 
  merge(., snp_entrez_description, by = "ensembl_gene_id") 

innate_reactome <-
  snp_reactome %>% 
  filter(grepl("innate|cytokine|interleukin", path_name, ignore.case = TRUE)) %>% 
  merge(., snp_ensembl, by.x = "ensembl_gene_id", by.y = "Ensembl.ID")

innate_gene <- 
  c(innate_go$Gene, innate_reactome$Gene) %>% 
  data.frame() %>% 
  distinct()
```

### Signal Transduction
```{r}
signal_go <- 
  snp_go %>% 
  filter(grepl("signal", name_1006, ignore.case = TRUE)) %>% 
  merge(., snp_ensembl, by.x = "ensembl_gene_id", by.y = "Ensembl.ID") %>% 
  merge(., snp_entrez_description, by = "ensembl_gene_id") 

signal_reactome <- 
  snp_reactome %>% 
  filter(grepl("signal", path_name, ignore.case = TRUE)) %>% 
  merge(., snp_ensembl, by.x = "ensembl_gene_id", by.y = "Ensembl.ID")

signal_gene <- 
  c(signal_go$Gene, signal_reactome$Gene) %>% 
  data.frame() %>% 
  distinct()
```

### Cytokines
```{r}
cytokine_go <- 
  snp_go %>% 
  filter(grepl("cytokine ", name_1006, ignore.case = TRUE)) %>% 
  merge(., snp_ensembl, by.x = "ensembl_gene_id", by.y = "Ensembl.ID") %>% 
  merge(., snp_entrez_description, by = "ensembl_gene_id") 

cytokine_reactome <- 
  snp_reactome %>% 
  filter(grepl("cytokine ", path_name, ignore.case = TRUE)) %>% 
  merge(., snp_ensembl, by.x = "ensembl_gene_id", by.y = "Ensembl.ID")

cytokine_gene <- 
  c(cytokine_go$Gene, cytokine_reactome$Gene) %>% 
  data.frame() %>% 
  distinct()
```

### Adaptive Immune System
```{r}
adaptive_go <- 
  snp_go %>% 
  filter(grepl("adaptive|antigen |MHC", name_1006, ignore.case = TRUE)) %>% 
  merge(., snp_ensembl, by.x = "ensembl_gene_id", by.y = "Ensembl.ID") 

adaptive_reactome <- 
  snp_reactome %>% 
  filter(grepl("adaptive|antigen |MHC|complement", path_name, ignore.case = TRUE)) %>% 
  merge(., snp_ensembl, by.x = "ensembl_gene_id", by.y = "Ensembl.ID")

adaptive_gene <-
  c(adaptive_go$Gene, adaptive_reactome$Gene) %>% 
  data.frame() %>% 
  distinct()
```

### Nervous System
```{r}
nervous_go <- 
  snp_go %>% 
  filter(grepl("nervous|neur|axon|synap|dendrit|brain|dopamine|norepinephrine|sensory", 
               name_1006, ignore.case = TRUE)) %>% 
  merge(., snp_ensembl, by.x = "ensembl_gene_id", by.y = "Ensembl.ID")

nervous_reactome <- 
  snp_reactome %>% 
  filter(grepl("nervous|neur|synap|axon", path_name, ignore.case = TRUE)) %>% 
  merge(., snp_ensembl, by.x = "ensembl_gene_id", by.y = "Ensembl.ID")

nervous_gene <- 
  c(nervous_go$Gene, nervous_reactome$Gene) %>% 
  data.frame() %>% 
  distinct()
```

## Manuscript Figure
```{r}
bp_freq <- 
  snp_go %>% 
  filter(namespace_1003 == "biological_process") %>% 
  group_by(name_1006) %>% 
  summarize(freq = n()) %>% 
  filter(freq > 3) %>% 
  mutate(category = "GO Biological Process") %>% 
  setNames(c("title", "freq", "category"))

reactome_freq <- 
  snp_reactome %>% 
  group_by(path_name) %>% 
  summarize(freq = n()) %>% 
  filter(freq > 3) %>% 
  mutate(category = "Reactome Pathway") %>% 
  setNames(c("title", "freq", "category"))

ggplot(bp_freq, aes(x = fct_reorder(title, freq), y = freq)) +
  geom_bar(stat = "identity", position = "dodge") +
  labs(x=NULL, y="Number of Genes", title = "GO Biological Process") +
  scale_x_discrete(labels = function(description) str_wrap(description, width = 60)) +
  coord_flip() +
  theme_bw() +
  theme(panel.grid=element_blank(), text=element_text(size=20))

ggplot(reactome_freq, aes(x = fct_reorder(title, freq), y = freq)) +
  geom_bar(stat = "identity", position = "dodge") +
  labs(x=NULL, y="Number of Genes", title = "Reactome Pathway") +
  scale_x_discrete(labels = function(description) str_wrap(description, width = 35)) +
  coord_flip() +
  theme_bw() +
  theme(panel.grid=element_blank(), text=element_text(size=20))


bp_reactome_freq <- 
  rbind(bp_freq, reactome_freq) %>% 
  ggplot(., aes(x=fct_reorder(title, freq), y = freq)) +
  geom_bar(stat = "identity", position = "dodge") +
  scale_x_discrete(labels = function(title) str_wrap(title, width = 35))+
  labs(x = NULL, y = "Number of Genes") +
  facet_wrap(~category, scale = "free") +
  coord_flip() +
  theme_bw() + 
  theme(panel.grid = element_blank())
bp_reactome_freq

ggsave("Figures/snp_bp_reactome.jpg", bp_reactome_freq,
       width = 9, height = 7, units = "in")
```

## Presentation Figure
```{r}
adaptive_gene <- 
  adaptive_gene %>% 
  mutate(role = "Adaptive Immunity") 

innate_gene <- 
  innate_gene %>% 
  mutate(role = "Innate Immunity") 

nervous_gene <- 
  nervous_gene %>% 
  mutate(role = "Nervous System")

genes_all <-
  rbind(adaptive_gene, innate_gene, nervous_gene) %>% 
  group_by(role) %>% 
  summarize(sum = n()) %>% 
  mutate(role = factor(role, 
                       levels = c("Innate Immunity", 
                                  "Adaptive Immunity", 
                                  "Nervous System")))

snp_genes_plot <- 
  genes_all %>% 
  ggplot(., aes(x = role, y = sum, fill = role)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("Innate Immunity" = "#B5CDD3", 
                               "Adaptive Immunity" = "#AFBABB",
                               "Nervous System" = "#7F7B9A")) +
  labs(x = NULL, y = "Number of Genes") +
  scale_x_discrete(labels = function(role) str_wrap(role, width = 8)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none",
        text = element_text(size = 20))

ggsave("Figures/snp_annotation_num.jpeg", snp_genes_plot,
       width = 5, height = 5, units = "in")
```

## Link genes to SNPs for figure
```{r}
snp_immune_genes <-
  c(adaptive_gene$., innate_gene$.) %>% 
  data.frame() %>% 
  distinct() %>% 
  mutate(role = "Immune System") 

snp_immunenervous_genes <-
  intersect(snp_immune_genes$., nervous_gene$.) %>% 
  data.frame() %>% 
  mutate(role = "Immune & Nervous System")

snp_immune_genes <- 
  snp_immune_genes %>% 
  filter(!. %in% snp_immunenervous_genes$.)

snp_nervous_genes <- 
  nervous_gene %>% 
  filter(!. %in% snp_immunenervous_genes$.)

snp_genes <- 
  rbind(snp_immunenervous_genes, snp_immune_genes, snp_nervous_genes) %>% 
  setNames(c("gene", "role")) %>% 
  merge(., fst_sd_genes_tidy, by = "gene", all.x = TRUE, all.y = FALSE) %>% 
  mutate(snp = paste(V1, "_", V4, sep = "")) %>% 
  dplyr::select(c(gene, role, snp))

write.csv(snp_genes, "Output Files/snp_gene_roles.csv")
```