<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 All Samples | Pv-WGS-Analysis</title>
  <meta name="description" content="McCosker Dissertation Chapter 3 Analyses" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="5 All Samples | Pv-WGS-Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="McCosker Dissertation Chapter 3 Analyses" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 All Samples | Pv-WGS-Analysis" />
  
  <meta name="twitter:description" content="McCosker Dissertation Chapter 3 Analyses" />
  

<meta name="author" content="Christina McCosker" />


<meta name="date" content="2025-03-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="density-plot-filter-metrics.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Pv-WGS-Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a></li>
<li class="chapter" data-level="2" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html"><i class="fa fa-check"></i><b>2</b> Tidy Data &amp; Exploratory</a>
<ul>
<li class="chapter" data-level="2.1" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#libraries"><i class="fa fa-check"></i><b>2.1</b> Libraries</a></li>
<li class="chapter" data-level="2.2" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#data"><i class="fa fa-check"></i><b>2.2</b> Data</a></li>
<li class="chapter" data-level="2.3" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#descriptive-statistics"><i class="fa fa-check"></i><b>2.3</b> Descriptive Statistics</a></li>
<li class="chapter" data-level="2.4" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#exclude-samples"><i class="fa fa-check"></i><b>2.4</b> Exclude samples</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#sample-statistics"><i class="fa fa-check"></i><b>2.4.1</b> Sample statistics</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#map-of-strandings"><i class="fa fa-check"></i><b>2.5</b> Map of Strandings</a></li>
<li class="chapter" data-level="2.6" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#case-v-control-factors"><i class="fa fa-check"></i><b>2.6</b> Case v Control ~ Factors</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#duringafter-outbreak"><i class="fa fa-check"></i><b>2.6.1</b> During/After Outbreak</a></li>
<li class="chapter" data-level="2.6.2" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#sex"><i class="fa fa-check"></i><b>2.6.2</b> Sex</a></li>
<li class="chapter" data-level="2.6.3" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#age"><i class="fa fa-check"></i><b>2.6.3</b> Age</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html"><i class="fa fa-check"></i><b>3</b> PCA - High Coverage</a>
<ul>
<li class="chapter" data-level="3.1" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#libraries-1"><i class="fa fa-check"></i><b>3.1</b> Libraries</a></li>
<li class="chapter" data-level="3.2" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#data-1"><i class="fa fa-check"></i><b>3.2</b> Data</a></li>
<li class="chapter" data-level="3.3" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#vcf-to-gds"><i class="fa fa-check"></i><b>3.3</b> VCF to GDS</a></li>
<li class="chapter" data-level="3.4" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#ld-based-snp-pruning"><i class="fa fa-check"></i><b>3.4</b> LD-based SNP Pruning</a></li>
<li class="chapter" data-level="3.5" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#pca"><i class="fa fa-check"></i><b>3.5</b> PCA</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#pca-original"><i class="fa fa-check"></i><b>3.5.1</b> PCA Original</a></li>
<li class="chapter" data-level="3.5.2" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#pca-wo-outliers"><i class="fa fa-check"></i><b>3.5.2</b> PCA w/o Outliers</a></li>
<li class="chapter" data-level="3.5.3" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#pca-by-chromosome"><i class="fa fa-check"></i><b>3.5.3</b> PCA by Chromosome</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#relatedness"><i class="fa fa-check"></i><b>3.6</b> Relatedness</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html"><i class="fa fa-check"></i><b>4</b> Density Plot Filter Metrics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#libraries-2"><i class="fa fa-check"></i><b>4.1</b> Libraries</a></li>
<li class="chapter" data-level="4.2" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#data-2"><i class="fa fa-check"></i><b>4.2</b> Data</a></li>
<li class="chapter" data-level="4.3" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#density-plots"><i class="fa fa-check"></i><b>4.3</b> Density Plots</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-qd"><i class="fa fa-check"></i><b>4.3.1</b> Metric: QD</a></li>
<li class="chapter" data-level="4.3.2" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-fs"><i class="fa fa-check"></i><b>4.3.2</b> Metric: FS</a></li>
<li class="chapter" data-level="4.3.3" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-sor"><i class="fa fa-check"></i><b>4.3.3</b> Metric: SOR</a></li>
<li class="chapter" data-level="4.3.4" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-mq"><i class="fa fa-check"></i><b>4.3.4</b> Metric: MQ</a></li>
<li class="chapter" data-level="4.3.5" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-mqranksum"><i class="fa fa-check"></i><b>4.3.5</b> Metric: MQRankSum</a></li>
<li class="chapter" data-level="4.3.6" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-readposranksum"><i class="fa fa-check"></i><b>4.3.6</b> Metric: ReadPosRankSum</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="all-samples.html"><a href="all-samples.html"><i class="fa fa-check"></i><b>5</b> All Samples</a>
<ul>
<li class="chapter" data-level="5.1" data-path="all-samples.html"><a href="all-samples.html#gwas---rrblup"><i class="fa fa-check"></i><b>5.1</b> GWAS - rrBLUP</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="all-samples.html"><a href="all-samples.html#libraries-3"><i class="fa fa-check"></i><b>5.1.1</b> Libraries</a></li>
<li class="chapter" data-level="5.1.2" data-path="all-samples.html"><a href="all-samples.html#data-3"><i class="fa fa-check"></i><b>5.1.2</b> Data</a></li>
<li class="chapter" data-level="5.1.3" data-path="all-samples.html"><a href="all-samples.html#calculating-grm"><i class="fa fa-check"></i><b>5.1.3</b> Calculating GRM</a></li>
<li class="chapter" data-level="5.1.4" data-path="all-samples.html"><a href="all-samples.html#population-structure"><i class="fa fa-check"></i><b>5.1.4</b> Population structure</a></li>
<li class="chapter" data-level="5.1.5" data-path="all-samples.html"><a href="all-samples.html#configure-data"><i class="fa fa-check"></i><b>5.1.5</b> Configure data</a></li>
<li class="chapter" data-level="5.1.6" data-path="all-samples.html"><a href="all-samples.html#gwas-model-1-genomic-relationship-matrix"><i class="fa fa-check"></i><b>5.1.6</b> GWAS Model 1: genomic relationship matrix</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="all-samples.html"><a href="all-samples.html#fst-based-approach"><i class="fa fa-check"></i><b>5.2</b> Fst-based Approach</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="all-samples.html"><a href="all-samples.html#libraries-4"><i class="fa fa-check"></i><b>5.2.1</b> Libraries</a></li>
<li class="chapter" data-level="5.2.2" data-path="all-samples.html"><a href="all-samples.html#data-4"><i class="fa fa-check"></i><b>5.2.2</b> Data</a></li>
<li class="chapter" data-level="5.2.3" data-path="all-samples.html"><a href="all-samples.html#visualize-snp-fst-distribution"><i class="fa fa-check"></i><b>5.2.3</b> Visualize SNP Fst distribution</a></li>
<li class="chapter" data-level="5.2.4" data-path="all-samples.html"><a href="all-samples.html#select-outlier-snps"><i class="fa fa-check"></i><b>5.2.4</b> Select outlier SNPs</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="all-samples.html"><a href="all-samples.html#snp-annotation"><i class="fa fa-check"></i><b>5.3</b> SNP Annotation</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="all-samples.html"><a href="all-samples.html#libraries-5"><i class="fa fa-check"></i><b>5.3.1</b> Libraries</a></li>
<li class="chapter" data-level="5.3.2" data-path="all-samples.html"><a href="all-samples.html#data-5"><i class="fa fa-check"></i><b>5.3.2</b> Data</a></li>
<li class="chapter" data-level="5.3.3" data-path="all-samples.html"><a href="all-samples.html#tidy-harbor-seal-genes"><i class="fa fa-check"></i><b>5.3.3</b> Tidy Harbor seal genes</a></li>
<li class="chapter" data-level="5.3.4" data-path="all-samples.html"><a href="all-samples.html#fst-5-sd"><i class="fa fa-check"></i><b>5.3.4</b> FST 5 SD</a></li>
<li class="chapter" data-level="5.3.5" data-path="all-samples.html"><a href="all-samples.html#gwas"><i class="fa fa-check"></i><b>5.3.5</b> GWAS</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/christinamccosker/pv-wgs.git" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pv-WGS-Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="all-samples" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> All Samples<a href="all-samples.html#all-samples" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="gwas---rrblup" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> GWAS - rrBLUP<a href="all-samples.html#gwas---rrblup" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="libraries-3" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Libraries<a href="all-samples.html#libraries-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="all-samples.html#cb160-1" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb160-2"><a href="all-samples.html#cb160-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb160-3"><a href="all-samples.html#cb160-3" tabindex="-1"></a><span class="fu">library</span>(rrBLUP)</span>
<span id="cb160-4"><a href="all-samples.html#cb160-4" tabindex="-1"></a><span class="fu">library</span>(qqman)</span>
<span id="cb160-5"><a href="all-samples.html#cb160-5" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb160-6"><a href="all-samples.html#cb160-6" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span></code></pre></div>
</div>
<div id="data-3" class="section level3 hasAnchor" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> Data<a href="all-samples.html#data-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="all-samples.html#cb161-1" tabindex="-1"></a>snp.matrix <span class="ot">&lt;-</span></span>
<span id="cb161-2"><a href="all-samples.html#cb161-2" tabindex="-1"></a>  <span class="fu">fread</span>(<span class="st">&quot;C:/Users/Christina/OneDrive - University of Maine System/pv-wgs/allsamples_maf_prune.raw&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb161-3"><a href="all-samples.html#cb161-3" tabindex="-1"></a>  <span class="fu">rename_with</span>(gsub, </span>
<span id="cb161-4"><a href="all-samples.html#cb161-4" tabindex="-1"></a>              <span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">_[A-Z]{1}$&quot;</span>, </span>
<span id="cb161-5"><a href="all-samples.html#cb161-5" tabindex="-1"></a>              <span class="at">replacement =</span> <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(.))</span>
<span id="cb161-6"><a href="all-samples.html#cb161-6" tabindex="-1"></a></span>
<span id="cb161-7"><a href="all-samples.html#cb161-7" tabindex="-1"></a>snp.info <span class="ot">&lt;-</span> </span>
<span id="cb161-8"><a href="all-samples.html#cb161-8" tabindex="-1"></a>  <span class="fu">fread</span>(<span class="st">&quot;C:/Users/Christina/OneDrive - University of Maine System/pv-wgs/allsamples_maf_prune.map&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb161-9"><a href="all-samples.html#cb161-9" tabindex="-1"></a>  <span class="fu">setNames</span>(<span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;snp&quot;</span>, <span class="st">&quot;cM&quot;</span>, <span class="st">&quot;pos&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb161-10"><a href="all-samples.html#cb161-10" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>cM) <span class="sc">%&gt;%</span> </span>
<span id="cb161-11"><a href="all-samples.html#cb161-11" tabindex="-1"></a>  <span class="fu">filter</span>(snp <span class="sc">%in%</span> <span class="fu">colnames</span>(snp.matrix))</span>
<span id="cb161-12"><a href="all-samples.html#cb161-12" tabindex="-1"></a>  </span>
<span id="cb161-13"><a href="all-samples.html#cb161-13" tabindex="-1"></a>meta <span class="ot">&lt;-</span> </span>
<span id="cb161-14"><a href="all-samples.html#cb161-14" tabindex="-1"></a>  <span class="fu">fread</span>(<span class="st">&quot;Output Files/metadata_tidy.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb161-15"><a href="all-samples.html#cb161-15" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq.id =</span> <span class="fu">gsub</span>(<span class="st">&quot;B5GAN&quot;</span>, <span class="st">&quot;&quot;</span>, seq.id)) <span class="sc">%&gt;%</span> </span>
<span id="cb161-16"><a href="all-samples.html#cb161-16" tabindex="-1"></a>  <span class="fu">filter</span>(seq.id <span class="sc">%in%</span> snp.matrix<span class="sc">$</span>IID)</span></code></pre></div>
</div>
<div id="calculating-grm" class="section level3 hasAnchor" number="5.1.3">
<h3><span class="header-section-number">5.1.3</span> Calculating GRM<a href="all-samples.html#calculating-grm" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>SNP calls need to be changed from 0, 1, 2 to -1, 0, 1 (for rrBLUP)</p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="all-samples.html#cb162-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> </span>
<span id="cb162-2"><a href="all-samples.html#cb162-2" tabindex="-1"></a>  snp.matrix[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)] <span class="sc">%&gt;%</span> </span>
<span id="cb162-3"><a href="all-samples.html#cb162-3" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb162-4"><a href="all-samples.html#cb162-4" tabindex="-1"></a></span>
<span id="cb162-5"><a href="all-samples.html#cb162-5" tabindex="-1"></a><span class="co"># Change from 0,1,2 to -1,0,1 for rrBLUP</span></span>
<span id="cb162-6"><a href="all-samples.html#cb162-6" tabindex="-1"></a>x <span class="ot">&lt;-</span> x<span class="dv">-1</span></span>
<span id="cb162-7"><a href="all-samples.html#cb162-7" tabindex="-1"></a></span>
<span id="cb162-8"><a href="all-samples.html#cb162-8" tabindex="-1"></a><span class="co"># Row names = sample ID</span></span>
<span id="cb162-9"><a href="all-samples.html#cb162-9" tabindex="-1"></a><span class="fu">rownames</span>(x) <span class="ot">&lt;-</span> snp.matrix<span class="sc">$</span>IID</span>
<span id="cb162-10"><a href="all-samples.html#cb162-10" tabindex="-1"></a></span>
<span id="cb162-11"><a href="all-samples.html#cb162-11" tabindex="-1"></a>grm <span class="ot">&lt;-</span> <span class="fu">A.mat</span>(x)</span></code></pre></div>
</div>
<div id="population-structure" class="section level3 hasAnchor" number="5.1.4">
<h3><span class="header-section-number">5.1.4</span> Population structure<a href="all-samples.html#population-structure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="heatmap" class="section level4 hasAnchor" number="5.1.4.1">
<h4><span class="header-section-number">5.1.4.1</span> Heatmap<a href="all-samples.html#heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>No obvious clustering of samples</p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="all-samples.html#cb163-1" tabindex="-1"></a><span class="fu">heatmap</span>(grm, <span class="at">col=</span><span class="fu">rev</span>(<span class="fu">heat.colors</span>(<span class="dv">75</span>)))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-54-1.png" width="672" /></p>
</div>
<div id="pca-1" class="section level4 hasAnchor" number="5.1.4.2">
<h4><span class="header-section-number">5.1.4.2</span> PCA<a href="all-samples.html#pca-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Again, seemingly no sample clustering</p>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="all-samples.html#cb164-1" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(grm, <span class="at">scale=</span><span class="cn">TRUE</span>, <span class="at">center=</span><span class="cn">TRUE</span>)</span>
<span id="cb164-2"><a href="all-samples.html#cb164-2" tabindex="-1"></a></span>
<span id="cb164-3"><a href="all-samples.html#cb164-3" tabindex="-1"></a><span class="co"># Variance explained by first 8 PCs</span></span>
<span id="cb164-4"><a href="all-samples.html#cb164-4" tabindex="-1"></a>eigenvalues <span class="ot">&lt;-</span> pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb164-5"><a href="all-samples.html#cb164-5" tabindex="-1"></a>prop <span class="ot">&lt;-</span> <span class="fu">round</span>((eigenvalues<span class="sc">/</span><span class="fu">sum</span>(eigenvalues)<span class="sc">*</span><span class="dv">100</span>), <span class="dv">2</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>] </span>
<span id="cb164-6"><a href="all-samples.html#cb164-6" tabindex="-1"></a>prop</span></code></pre></div>
<pre><code>## [1] 2.72 2.10 1.37 1.27 1.24 1.23 1.21 1.20</code></pre>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="all-samples.html#cb166-1" tabindex="-1"></a><span class="fu">fviz_eig</span>(pca, <span class="at">ncp =</span> <span class="dv">8</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-55-1.png" width="672" /></p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="all-samples.html#cb167-1" tabindex="-1"></a>pca.data <span class="ot">&lt;-</span></span>
<span id="cb167-2"><a href="all-samples.html#cb167-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb167-3"><a href="all-samples.html#cb167-3" tabindex="-1"></a>    <span class="at">sample =</span> <span class="fu">rownames</span>(pca<span class="sc">$</span>x),</span>
<span id="cb167-4"><a href="all-samples.html#cb167-4" tabindex="-1"></a>    <span class="at">x =</span> pca<span class="sc">$</span>x[,<span class="dv">1</span>],</span>
<span id="cb167-5"><a href="all-samples.html#cb167-5" tabindex="-1"></a>    <span class="at">y =</span> pca<span class="sc">$</span>x[,<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb167-6"><a href="all-samples.html#cb167-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb167-7"><a href="all-samples.html#cb167-7" tabindex="-1"></a></span>
<span id="cb167-8"><a href="all-samples.html#cb167-8" tabindex="-1"></a>labelx <span class="ot">&lt;-</span> </span>
<span id="cb167-9"><a href="all-samples.html#cb167-9" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, prop[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb167-10"><a href="all-samples.html#cb167-10" tabindex="-1"></a>labely <span class="ot">&lt;-</span> </span>
<span id="cb167-11"><a href="all-samples.html#cb167-11" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, prop[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb167-12"><a href="all-samples.html#cb167-12" tabindex="-1"></a></span>
<span id="cb167-13"><a href="all-samples.html#cb167-13" tabindex="-1"></a><span class="fu">ggplot</span>(pca.data, <span class="fu">aes</span>(<span class="at">x=</span>x, <span class="at">y=</span>y, <span class="at">col=</span>classification)) <span class="sc">+</span></span>
<span id="cb167-14"><a href="all-samples.html#cb167-14" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb167-15"><a href="all-samples.html#cb167-15" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb167-16"><a href="all-samples.html#cb167-16" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>)) <span class="sc">+</span></span>
<span id="cb167-17"><a href="all-samples.html#cb167-17" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb167-18"><a href="all-samples.html#cb167-18" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb167-19"><a href="all-samples.html#cb167-19" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb167-20"><a href="all-samples.html#cb167-20" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-55-2.png" width="672" /></p>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="all-samples.html#cb168-1" tabindex="-1"></a><span class="fu">ggplot</span>(pca.data, <span class="fu">aes</span>(<span class="at">x=</span>x, <span class="at">y=</span>y, <span class="at">col=</span>pdv)) <span class="sc">+</span></span>
<span id="cb168-2"><a href="all-samples.html#cb168-2" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb168-3"><a href="all-samples.html#cb168-3" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb168-4"><a href="all-samples.html#cb168-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>, <span class="st">&quot;coral&quot;</span>)) <span class="sc">+</span></span>
<span id="cb168-5"><a href="all-samples.html#cb168-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb168-6"><a href="all-samples.html#cb168-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb168-7"><a href="all-samples.html#cb168-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb168-8"><a href="all-samples.html#cb168-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-55-3.png" width="672" /></p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="all-samples.html#cb169-1" tabindex="-1"></a><span class="fu">ggplot</span>(pca.data, <span class="fu">aes</span>(<span class="at">x=</span>x, <span class="at">y=</span>y, <span class="at">col=</span>sex)) <span class="sc">+</span></span>
<span id="cb169-2"><a href="all-samples.html#cb169-2" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb169-3"><a href="all-samples.html#cb169-3" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb169-4"><a href="all-samples.html#cb169-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>, <span class="st">&quot;coral&quot;</span>)) <span class="sc">+</span></span>
<span id="cb169-5"><a href="all-samples.html#cb169-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb169-6"><a href="all-samples.html#cb169-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb169-7"><a href="all-samples.html#cb169-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb169-8"><a href="all-samples.html#cb169-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span></code></pre></div>
<pre><code>## Too few points to calculate an ellipse</code></pre>
<pre><code>## Warning: Removed 1 row containing missing values or values outside
## the scale range (`geom_path()`).</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-55-4.png" width="672" /></p>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="all-samples.html#cb172-1" tabindex="-1"></a><span class="fu">ggplot</span>(pca.data, <span class="fu">aes</span>(<span class="at">x=</span>x, <span class="at">y=</span>y, <span class="at">col=</span>age.class)) <span class="sc">+</span></span>
<span id="cb172-2"><a href="all-samples.html#cb172-2" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb172-3"><a href="all-samples.html#cb172-3" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb172-4"><a href="all-samples.html#cb172-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>, <span class="st">&quot;coral&quot;</span>, <span class="st">&quot;magenta&quot;</span>, <span class="st">&quot;black&quot;</span>)) <span class="sc">+</span></span>
<span id="cb172-5"><a href="all-samples.html#cb172-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb172-6"><a href="all-samples.html#cb172-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb172-7"><a href="all-samples.html#cb172-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb172-8"><a href="all-samples.html#cb172-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span></code></pre></div>
<pre><code>## Too few points to calculate an ellipse</code></pre>
<pre><code>## Warning in MASS::cov.trob(data[, vars]): Probable convergence failure
## Warning in MASS::cov.trob(data[, vars]): Removed 1 row containing missing values or values outside
## the scale range (`geom_path()`).</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-55-5.png" width="672" /></p>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="all-samples.html#cb175-1" tabindex="-1"></a><span class="fu">ggplot</span>(pca.data, <span class="fu">aes</span>(<span class="at">x=</span>x, <span class="at">y=</span>y, <span class="at">col=</span>condition)) <span class="sc">+</span></span>
<span id="cb175-2"><a href="all-samples.html#cb175-2" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb175-3"><a href="all-samples.html#cb175-3" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb175-4"><a href="all-samples.html#cb175-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>, <span class="st">&quot;coral&quot;</span>, <span class="st">&quot;magenta&quot;</span>)) <span class="sc">+</span></span>
<span id="cb175-5"><a href="all-samples.html#cb175-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb175-6"><a href="all-samples.html#cb175-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb175-7"><a href="all-samples.html#cb175-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb175-8"><a href="all-samples.html#cb175-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span></code></pre></div>
<pre><code>## Too few points to calculate an ellipse</code></pre>
<pre><code>## Warning: Removed 1 row containing missing values or values outside
## the scale range (`geom_path()`).</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-55-6.png" width="672" /></p>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="all-samples.html#cb178-1" tabindex="-1"></a><span class="fu">ggplot</span>(pca.data, <span class="fu">aes</span>(<span class="at">x=</span>x, <span class="at">y=</span>y, <span class="at">col=</span><span class="fu">factor</span>(seq.run))) <span class="sc">+</span></span>
<span id="cb178-2"><a href="all-samples.html#cb178-2" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb178-3"><a href="all-samples.html#cb178-3" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb178-4"><a href="all-samples.html#cb178-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>)) <span class="sc">+</span></span>
<span id="cb178-5"><a href="all-samples.html#cb178-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb178-6"><a href="all-samples.html#cb178-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb178-7"><a href="all-samples.html#cb178-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb178-8"><a href="all-samples.html#cb178-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-55-7.png" width="672" /></p>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="all-samples.html#cb179-1" tabindex="-1"></a><span class="fu">ggplot</span>(pca.data, <span class="fu">aes</span>(<span class="at">x=</span>x, <span class="at">y=</span>y, <span class="at">col=</span>coverage)) <span class="sc">+</span></span>
<span id="cb179-2"><a href="all-samples.html#cb179-2" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb179-3"><a href="all-samples.html#cb179-3" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb179-4"><a href="all-samples.html#cb179-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>)) <span class="sc">+</span></span>
<span id="cb179-5"><a href="all-samples.html#cb179-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb179-6"><a href="all-samples.html#cb179-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb179-7"><a href="all-samples.html#cb179-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb179-8"><a href="all-samples.html#cb179-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-55-8.png" width="672" /></p>
</div>
</div>
<div id="configure-data" class="section level3 hasAnchor" number="5.1.5">
<h3><span class="header-section-number">5.1.5</span> Configure data<a href="all-samples.html#configure-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="create-matrix" class="section level4 hasAnchor" number="5.1.5.1">
<h4><span class="header-section-number">5.1.5.1</span> Create matrix<a href="all-samples.html#create-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Add marker location and marker dosage</p>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="all-samples.html#cb180-1" tabindex="-1"></a>snp.info <span class="ot">&lt;-</span> </span>
<span id="cb180-2"><a href="all-samples.html#cb180-2" tabindex="-1"></a>  <span class="fu">bind_cols</span>(snp.info, <span class="fu">as.data.frame</span>(<span class="fu">t</span>(x)))</span></code></pre></div>
</div>
<div id="configure-phenotype-data" class="section level4 hasAnchor" number="5.1.5.2">
<h4><span class="header-section-number">5.1.5.2</span> Configure phenotype data<a href="all-samples.html#configure-phenotype-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Non-survivor = 0, Survivor = 1</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="all-samples.html#cb181-1" tabindex="-1"></a>phenotypes <span class="ot">&lt;-</span> </span>
<span id="cb181-2"><a href="all-samples.html#cb181-2" tabindex="-1"></a>  meta <span class="sc">%&gt;%</span> </span>
<span id="cb181-3"><a href="all-samples.html#cb181-3" tabindex="-1"></a>  <span class="fu">select</span>(seq.id, classification) <span class="sc">%&gt;%</span> </span>
<span id="cb181-4"><a href="all-samples.html#cb181-4" tabindex="-1"></a>  <span class="fu">setNames</span>(<span class="fu">c</span>(<span class="st">&quot;id&quot;</span>, <span class="st">&quot;phenotype&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb181-5"><a href="all-samples.html#cb181-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">phenotype =</span> <span class="fu">ifelse</span>(phenotype<span class="sc">==</span><span class="st">&quot;Non-survivor&quot;</span>, <span class="dv">0</span>, <span class="dv">1</span>))</span>
<span id="cb181-6"><a href="all-samples.html#cb181-6" tabindex="-1"></a></span>
<span id="cb181-7"><a href="all-samples.html#cb181-7" tabindex="-1"></a><span class="co"># Make sure sample ID order matches</span></span>
<span id="cb181-8"><a href="all-samples.html#cb181-8" tabindex="-1"></a><span class="fu">unique</span>(phenotypes<span class="sc">$</span>id <span class="sc">==</span> <span class="fu">rownames</span>(x)) <span class="co"># no</span></span></code></pre></div>
<pre><code>## [1]  TRUE FALSE</code></pre>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="all-samples.html#cb183-1" tabindex="-1"></a>phenotypes <span class="ot">&lt;-</span> </span>
<span id="cb183-2"><a href="all-samples.html#cb183-2" tabindex="-1"></a>  phenotypes[<span class="fu">order</span>(<span class="fu">match</span>(phenotypes<span class="sc">$</span>id, <span class="fu">rownames</span>(x))), ]</span>
<span id="cb183-3"><a href="all-samples.html#cb183-3" tabindex="-1"></a></span>
<span id="cb183-4"><a href="all-samples.html#cb183-4" tabindex="-1"></a><span class="fu">unique</span>(phenotypes<span class="sc">$</span>id <span class="sc">==</span> <span class="fu">rownames</span>(grm)) <span class="co"># yes</span></span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="all-samples.html#cb185-1" tabindex="-1"></a><span class="fu">unique</span>(phenotypes<span class="sc">$</span>id <span class="sc">==</span> <span class="fu">rownames</span>(x)) <span class="co"># yes</span></span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
</div>
<div id="configure-grm-snp-info" class="section level4 hasAnchor" number="5.1.5.3">
<h4><span class="header-section-number">5.1.5.3</span> Configure grm &amp; snp info<a href="all-samples.html#configure-grm-snp-info" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="all-samples.html#cb187-1" tabindex="-1"></a>vec <span class="ot">&lt;-</span> <span class="fu">colnames</span>(grm) <span class="sc">%in%</span> phenotypes<span class="sc">$</span>id</span>
<span id="cb187-2"><a href="all-samples.html#cb187-2" tabindex="-1"></a>grm <span class="ot">&lt;-</span> grm[vec,vec]</span>
<span id="cb187-3"><a href="all-samples.html#cb187-3" tabindex="-1"></a></span>
<span id="cb187-4"><a href="all-samples.html#cb187-4" tabindex="-1"></a>snp.info <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(snp.info)</span>
<span id="cb187-5"><a href="all-samples.html#cb187-5" tabindex="-1"></a>snp.info <span class="ot">&lt;-</span> snp.info[,<span class="fu">c</span>(<span class="cn">TRUE</span>,<span class="cn">TRUE</span>,<span class="cn">TRUE</span>,vec)]</span></code></pre></div>
</div>
</div>
<div id="gwas-model-1-genomic-relationship-matrix" class="section level3 hasAnchor" number="5.1.6">
<h3><span class="header-section-number">5.1.6</span> GWAS Model 1: genomic relationship matrix<a href="all-samples.html#gwas-model-1-genomic-relationship-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="all-samples.html#cb188-1" tabindex="-1"></a>model <span class="ot">&lt;-</span> </span>
<span id="cb188-2"><a href="all-samples.html#cb188-2" tabindex="-1"></a>  <span class="fu">GWAS</span>(<span class="at">pheno =</span> phenotypes, </span>
<span id="cb188-3"><a href="all-samples.html#cb188-3" tabindex="-1"></a>       <span class="at">geno =</span> snp.info, </span>
<span id="cb188-4"><a href="all-samples.html#cb188-4" tabindex="-1"></a>       <span class="at">K =</span> grm, </span>
<span id="cb188-5"><a href="all-samples.html#cb188-5" tabindex="-1"></a>       <span class="at">fixed =</span> <span class="cn">NULL</span>, </span>
<span id="cb188-6"><a href="all-samples.html#cb188-6" tabindex="-1"></a>       <span class="at">plot =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;GWAS for trait: phenotype&quot;
## [1] &quot;Variance components estimated. Testing markers.&quot;</code></pre>
<div id="extract-results" class="section level4 hasAnchor" number="5.1.6.1">
<h4><span class="header-section-number">5.1.6.1</span> Extract results<a href="all-samples.html#extract-results" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Set names to defaults for manhattan plot.<br />
GWAS outputs -log10(pvalues), convert to pvalues.<br />
Try FDR and Bonferroni p-adjustment methods.</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="all-samples.html#cb190-1" tabindex="-1"></a>gwasResults <span class="ot">&lt;-</span></span>
<span id="cb190-2"><a href="all-samples.html#cb190-2" tabindex="-1"></a>  model <span class="sc">%&gt;%</span> </span>
<span id="cb190-3"><a href="all-samples.html#cb190-3" tabindex="-1"></a>  <span class="fu">select</span>(snp, chr, pos, phenotype) <span class="sc">%&gt;%</span> </span>
<span id="cb190-4"><a href="all-samples.html#cb190-4" tabindex="-1"></a>  <span class="fu">setNames</span>(<span class="fu">c</span>(<span class="st">&quot;SNP&quot;</span>, <span class="st">&quot;CHR&quot;</span>, <span class="st">&quot;BP&quot;</span>, <span class="st">&quot;logP&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb190-5"><a href="all-samples.html#cb190-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">CHR =</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;HiC_scaffold_&quot;</span>, <span class="st">&quot;&quot;</span>, CHR))) <span class="sc">%&gt;%</span> </span>
<span id="cb190-6"><a href="all-samples.html#cb190-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">P =</span> <span class="dv">10</span><span class="sc">^</span>(<span class="sc">-</span>(logP)),</span>
<span id="cb190-7"><a href="all-samples.html#cb190-7" tabindex="-1"></a>         <span class="at">fdr =</span> <span class="fu">p.adjust</span>(P, <span class="at">method =</span> <span class="st">&quot;fdr&quot;</span>),</span>
<span id="cb190-8"><a href="all-samples.html#cb190-8" tabindex="-1"></a>         <span class="at">bonf =</span> <span class="fu">p.adjust</span>(P, <span class="at">method =</span> <span class="st">&quot;bonferroni&quot;</span>))</span>
<span id="cb190-9"><a href="all-samples.html#cb190-9" tabindex="-1"></a></span>
<span id="cb190-10"><a href="all-samples.html#cb190-10" tabindex="-1"></a><span class="fu">write.csv</span>(gwasResults, <span class="st">&quot;Output Files/allsamples_gwas.csv&quot;</span>,</span>
<span id="cb190-11"><a href="all-samples.html#cb190-11" tabindex="-1"></a>          <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="manhattan-plot" class="section level4 hasAnchor" number="5.1.6.2">
<h4><span class="header-section-number">5.1.6.2</span> Manhattan Plot<a href="all-samples.html#manhattan-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="all-samples.html#cb191-1" tabindex="-1"></a><span class="co"># Threshold for significance</span></span>
<span id="cb191-2"><a href="all-samples.html#cb191-2" tabindex="-1"></a>bf <span class="ot">&lt;-</span> (<span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span> <span class="sc">/</span> <span class="fu">nrow</span>(snp.info)))</span>
<span id="cb191-3"><a href="all-samples.html#cb191-3" tabindex="-1"></a>fdr <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>)</span>
<span id="cb191-4"><a href="all-samples.html#cb191-4" tabindex="-1"></a></span>
<span id="cb191-5"><a href="all-samples.html#cb191-5" tabindex="-1"></a><span class="co"># Bonferroni</span></span>
<span id="cb191-6"><a href="all-samples.html#cb191-6" tabindex="-1"></a><span class="fu">manhattan</span>(gwasResults, </span>
<span id="cb191-7"><a href="all-samples.html#cb191-7" tabindex="-1"></a>          <span class="at">suggestiveline=</span><span class="cn">FALSE</span>, </span>
<span id="cb191-8"><a href="all-samples.html#cb191-8" tabindex="-1"></a>          <span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb191-9"><a href="all-samples.html#cb191-9" tabindex="-1"></a>          <span class="at">genomewideline =</span> bf, </span>
<span id="cb191-10"><a href="all-samples.html#cb191-10" tabindex="-1"></a>          <span class="at">p =</span> <span class="st">&quot;P&quot;</span>, <span class="at">logp =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-61-1.png" width="672" /></p>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="all-samples.html#cb192-1" tabindex="-1"></a><span class="co"># FDR</span></span>
<span id="cb192-2"><a href="all-samples.html#cb192-2" tabindex="-1"></a><span class="fu">manhattan</span>(gwasResults, </span>
<span id="cb192-3"><a href="all-samples.html#cb192-3" tabindex="-1"></a>          <span class="at">suggestiveline=</span><span class="cn">FALSE</span>, </span>
<span id="cb192-4"><a href="all-samples.html#cb192-4" tabindex="-1"></a>          <span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb192-5"><a href="all-samples.html#cb192-5" tabindex="-1"></a>          <span class="at">genomewideline =</span> fdr, </span>
<span id="cb192-6"><a href="all-samples.html#cb192-6" tabindex="-1"></a>          <span class="at">p =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">logp =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-61-2.png" width="672" /></p>

</div>
</div>
</div>
<div id="fst-based-approach" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Fst-based Approach<a href="all-samples.html#fst-based-approach" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To complement GWAS, conducting Fst test between cases and controls, selecting top 10% of SNPs as likely to be related to disease susceptibility.</p>
<div id="libraries-4" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Libraries<a href="all-samples.html#libraries-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="all-samples.html#cb193-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
</div>
<div id="data-4" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Data<a href="all-samples.html#data-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="all-samples.html#cb194-1" tabindex="-1"></a>fst <span class="ot">&lt;-</span> </span>
<span id="cb194-2"><a href="all-samples.html#cb194-2" tabindex="-1"></a>  <span class="fu">read.table</span>(<span class="st">&quot;C:/Users/Christina/OneDrive - University of Maine System/pv-wgs/allsamples.casevcontrol.weir.fst&quot;</span>,</span>
<span id="cb194-3"><a href="all-samples.html#cb194-3" tabindex="-1"></a>             <span class="at">header =</span> <span class="dv">1</span>) </span></code></pre></div>
</div>
<div id="visualize-snp-fst-distribution" class="section level3 hasAnchor" number="5.2.3">
<h3><span class="header-section-number">5.2.3</span> Visualize SNP Fst distribution<a href="all-samples.html#visualize-snp-fst-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="all-samples.html#cb195-1" tabindex="-1"></a><span class="fu">hist</span>(fst<span class="sc">$</span>WEIR_AND_COCKERHAM_FST)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-64-1.png" width="672" /></p>
</div>
<div id="select-outlier-snps" class="section level3 hasAnchor" number="5.2.4">
<h3><span class="header-section-number">5.2.4</span> Select outlier SNPs<a href="all-samples.html#select-outlier-snps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="top-10" class="section level4 hasAnchor" number="5.2.4.1">
<h4><span class="header-section-number">5.2.4.1</span> Top 10%<a href="all-samples.html#top-10" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Outlier = Within top 10% of Fst differentiation</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="all-samples.html#cb196-1" tabindex="-1"></a>ten <span class="ot">&lt;-</span> </span>
<span id="cb196-2"><a href="all-samples.html#cb196-2" tabindex="-1"></a>  <span class="fu">quantile</span>(fst<span class="sc">$</span>WEIR_AND_COCKERHAM_FST, <span class="at">probs =</span> <span class="fu">c</span>(<span class="fl">0.9</span>))</span>
<span id="cb196-3"><a href="all-samples.html#cb196-3" tabindex="-1"></a></span>
<span id="cb196-4"><a href="all-samples.html#cb196-4" tabindex="-1"></a>fst <span class="ot">&lt;-</span> </span>
<span id="cb196-5"><a href="all-samples.html#cb196-5" tabindex="-1"></a>  fst <span class="sc">%&gt;%</span> </span>
<span id="cb196-6"><a href="all-samples.html#cb196-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">outlier.10 =</span> <span class="fu">ifelse</span>(WEIR_AND_COCKERHAM_FST <span class="sc">&gt;</span> ten, <span class="st">&quot;outlier&quot;</span>, <span class="st">&quot;snp&quot;</span>))</span>
<span id="cb196-7"><a href="all-samples.html#cb196-7" tabindex="-1"></a></span>
<span id="cb196-8"><a href="all-samples.html#cb196-8" tabindex="-1"></a><span class="fu">nrow</span>(<span class="fu">filter</span>(fst, outlier<span class="fl">.10</span> <span class="sc">==</span> <span class="st">&quot;outlier&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 14884</code></pre>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="all-samples.html#cb198-1" tabindex="-1"></a><span class="fu">ggplot</span>(fst, <span class="fu">aes</span>(WEIR_AND_COCKERHAM_FST, <span class="at">fill =</span> outlier<span class="fl">.10</span>)) <span class="sc">+</span></span>
<span id="cb198-2"><a href="all-samples.html#cb198-2" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="fl">0.005</span>) <span class="sc">+</span></span>
<span id="cb198-3"><a href="all-samples.html#cb198-3" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;slategray3&quot;</span>, <span class="st">&quot;gray40&quot;</span>)) <span class="sc">+</span></span>
<span id="cb198-4"><a href="all-samples.html#cb198-4" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;FST&quot;</span>) <span class="sc">+</span></span>
<span id="cb198-5"><a href="all-samples.html#cb198-5" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb198-6"><a href="all-samples.html#cb198-6" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb198-7"><a href="all-samples.html#cb198-7" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-65-1.png" width="672" /></p>
</div>
<div id="sd" class="section level4 hasAnchor" number="5.2.4.2">
<h4><span class="header-section-number">5.2.4.2</span> &gt; 5SD<a href="all-samples.html#sd" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="all-samples.html#cb199-1" tabindex="-1"></a>sd <span class="ot">&lt;-</span> </span>
<span id="cb199-2"><a href="all-samples.html#cb199-2" tabindex="-1"></a>  <span class="fu">mean</span>(fst<span class="sc">$</span>WEIR_AND_COCKERHAM_FST) <span class="sc">+</span></span>
<span id="cb199-3"><a href="all-samples.html#cb199-3" tabindex="-1"></a>  <span class="dv">5</span><span class="sc">*</span><span class="fu">sd</span>(fst<span class="sc">$</span>WEIR_AND_COCKERHAM_FST)</span>
<span id="cb199-4"><a href="all-samples.html#cb199-4" tabindex="-1"></a></span>
<span id="cb199-5"><a href="all-samples.html#cb199-5" tabindex="-1"></a>fst <span class="ot">&lt;-</span> </span>
<span id="cb199-6"><a href="all-samples.html#cb199-6" tabindex="-1"></a>  fst <span class="sc">%&gt;%</span> </span>
<span id="cb199-7"><a href="all-samples.html#cb199-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">outlier.sd =</span> <span class="fu">ifelse</span>(WEIR_AND_COCKERHAM_FST <span class="sc">&gt;</span> sd, <span class="st">&quot;outlier&quot;</span>, <span class="st">&quot;snp&quot;</span>))</span>
<span id="cb199-8"><a href="all-samples.html#cb199-8" tabindex="-1"></a></span>
<span id="cb199-9"><a href="all-samples.html#cb199-9" tabindex="-1"></a><span class="fu">nrow</span>(fst <span class="sc">%&gt;%</span> <span class="fu">filter</span>(outlier.sd <span class="sc">==</span> <span class="st">&quot;outlier&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 539</code></pre>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="all-samples.html#cb201-1" tabindex="-1"></a><span class="fu">ggplot</span>(fst, <span class="fu">aes</span>(WEIR_AND_COCKERHAM_FST, <span class="at">fill =</span> outlier.sd)) <span class="sc">+</span></span>
<span id="cb201-2"><a href="all-samples.html#cb201-2" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="fl">0.005</span>) <span class="sc">+</span></span>
<span id="cb201-3"><a href="all-samples.html#cb201-3" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;slategray3&quot;</span>, <span class="st">&quot;gray40&quot;</span>)) <span class="sc">+</span></span>
<span id="cb201-4"><a href="all-samples.html#cb201-4" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;FST&quot;</span>) <span class="sc">+</span></span>
<span id="cb201-5"><a href="all-samples.html#cb201-5" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb201-6"><a href="all-samples.html#cb201-6" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb201-7"><a href="all-samples.html#cb201-7" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-66-1.png" width="672" /></p>
</div>
<div id="extract-snps-600-bp-positions" class="section level4 hasAnchor" number="5.2.4.3">
<h4><span class="header-section-number">5.2.4.3</span> Extract SNPs &amp; 600 bp positions<a href="all-samples.html#extract-snps-600-bp-positions" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="all-samples.html#cb202-1" tabindex="-1"></a>fst<span class="fl">.10</span> <span class="ot">&lt;-</span> </span>
<span id="cb202-2"><a href="all-samples.html#cb202-2" tabindex="-1"></a>  fst <span class="sc">%&gt;%</span> </span>
<span id="cb202-3"><a href="all-samples.html#cb202-3" tabindex="-1"></a>  <span class="fu">filter</span>(outlier<span class="fl">.10</span> <span class="sc">==</span> <span class="st">&quot;outlier&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb202-4"><a href="all-samples.html#cb202-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">minbp =</span> POS <span class="sc">-</span> <span class="dv">600</span>,</span>
<span id="cb202-5"><a href="all-samples.html#cb202-5" tabindex="-1"></a>         <span class="at">maxbp =</span> POS <span class="sc">+</span> <span class="dv">600</span>,</span>
<span id="cb202-6"><a href="all-samples.html#cb202-6" tabindex="-1"></a>         <span class="at">flank =</span> <span class="fu">paste0</span>(CHROM, <span class="st">&quot;:&quot;</span>, minbp, <span class="st">&quot;-&quot;</span>, maxbp, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb202-7"><a href="all-samples.html#cb202-7" tabindex="-1"></a></span>
<span id="cb202-8"><a href="all-samples.html#cb202-8" tabindex="-1"></a><span class="fu">write.csv</span>(fst<span class="fl">.10</span>, <span class="st">&quot;Output Files/allsamples_fst_10_outlierposition.csv&quot;</span>, </span>
<span id="cb202-9"><a href="all-samples.html#cb202-9" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb202-10"><a href="all-samples.html#cb202-10" tabindex="-1"></a></span>
<span id="cb202-11"><a href="all-samples.html#cb202-11" tabindex="-1"></a>fst.sd <span class="ot">&lt;-</span> </span>
<span id="cb202-12"><a href="all-samples.html#cb202-12" tabindex="-1"></a>  fst <span class="sc">%&gt;%</span> </span>
<span id="cb202-13"><a href="all-samples.html#cb202-13" tabindex="-1"></a>  <span class="fu">filter</span>(outlier.sd <span class="sc">==</span> <span class="st">&quot;outlier&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb202-14"><a href="all-samples.html#cb202-14" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">minbp =</span> POS <span class="sc">-</span> <span class="dv">600</span>,</span>
<span id="cb202-15"><a href="all-samples.html#cb202-15" tabindex="-1"></a>         <span class="at">maxbp =</span> POS <span class="sc">+</span> <span class="dv">600</span>,</span>
<span id="cb202-16"><a href="all-samples.html#cb202-16" tabindex="-1"></a>         <span class="at">flank =</span> <span class="fu">paste0</span>(CHROM, <span class="st">&quot;:&quot;</span>, minbp, <span class="st">&quot;-&quot;</span>, maxbp, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) </span>
<span id="cb202-17"><a href="all-samples.html#cb202-17" tabindex="-1"></a></span>
<span id="cb202-18"><a href="all-samples.html#cb202-18" tabindex="-1"></a><span class="fu">write.csv</span>(fst.sd, <span class="st">&quot;Output Files/allsamples_fst_sd_outlierposition.csv&quot;</span>, </span>
<span id="cb202-19"><a href="all-samples.html#cb202-19" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>

</div>
</div>
</div>
<div id="snp-annotation" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> SNP Annotation<a href="all-samples.html#snp-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="libraries-5" class="section level3 hasAnchor" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Libraries<a href="all-samples.html#libraries-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="all-samples.html#cb203-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb203-2"><a href="all-samples.html#cb203-2" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb203-3"><a href="all-samples.html#cb203-3" tabindex="-1"></a><span class="fu">library</span>(WebGestaltR)</span></code></pre></div>
<pre><code>## ******************************************</code></pre>
<pre><code>## *                                        *</code></pre>
<pre><code>## *          Welcome to WebGestaltR !      *</code></pre>
<pre><code>## *                                        *</code></pre>
<pre><code>## ******************************************</code></pre>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="all-samples.html#cb209-1" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb209-2"><a href="all-samples.html#cb209-2" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb209-3"><a href="all-samples.html#cb209-3" tabindex="-1"></a><span class="fu">library</span>(reactome.db)</span></code></pre></div>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:gridExtra&#39;:
## 
##     combine</code></pre>
<pre><code>## The following objects are masked from &#39;package:lubridate&#39;:
## 
##     intersect, setdiff, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
##     Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
##     table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:data.table&#39;:
## 
##     first, second</code></pre>
<pre><code>## The following objects are masked from &#39;package:lubridate&#39;:
## 
##     second, second&lt;-</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     first, rename</code></pre>
<pre><code>## The following object is masked from &#39;package:tidyr&#39;:
## 
##     expand</code></pre>
<pre><code>## The following object is masked from &#39;package:utils&#39;:
## 
##     findMatches</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## 
## Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:data.table&#39;:
## 
##     shift</code></pre>
<pre><code>## The following object is masked from &#39;package:lubridate&#39;:
## 
##     %within%</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     collapse, desc, slice</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     reduce</code></pre>
<pre><code>## The following object is masked from &#39;package:grDevices&#39;:
## 
##     windows</code></pre>
<pre><code>## 
## Attaching package: &#39;AnnotationDbi&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     select</code></pre>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="all-samples.html#cb238-1" tabindex="-1"></a><span class="fu">library</span>(ReactomePA)</span></code></pre></div>
<pre><code>## </code></pre>
<pre><code>## ReactomePA v1.50.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/
## 
## Please cite:
## 
## Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for
## reactome pathway analysis and visualization. Molecular BioSystems.
## 2016, 12(2):477-479</code></pre>
</div>
<div id="data-5" class="section level3 hasAnchor" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> Data<a href="all-samples.html#data-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Read in: SNP gene/exon/cds information</p>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="all-samples.html#cb241-1" tabindex="-1"></a>gwas_snp <span class="ot">&lt;-</span> </span>
<span id="cb241-2"><a href="all-samples.html#cb241-2" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/allsamples_gwas.csv&quot;</span>)</span>
<span id="cb241-3"><a href="all-samples.html#cb241-3" tabindex="-1"></a></span>
<span id="cb241-4"><a href="all-samples.html#cb241-4" tabindex="-1"></a>fst_sd_genes <span class="ot">&lt;-</span> </span>
<span id="cb241-5"><a href="all-samples.html#cb241-5" tabindex="-1"></a>  <span class="fu">fread</span>(<span class="st">&quot;Input Files/allsamples_fst_sd_outlier_genes_snps.txt&quot;</span>)</span>
<span id="cb241-6"><a href="all-samples.html#cb241-6" tabindex="-1"></a></span>
<span id="cb241-7"><a href="all-samples.html#cb241-7" tabindex="-1"></a>fst_sd_exons <span class="ot">&lt;-</span> </span>
<span id="cb241-8"><a href="all-samples.html#cb241-8" tabindex="-1"></a>  <span class="fu">fread</span>(<span class="st">&quot;Input Files/allsamples_fst_sd_outlier_exons_snps.txt&quot;</span>)</span>
<span id="cb241-9"><a href="all-samples.html#cb241-9" tabindex="-1"></a></span>
<span id="cb241-10"><a href="all-samples.html#cb241-10" tabindex="-1"></a>fst_sd_cds <span class="ot">&lt;-</span> </span>
<span id="cb241-11"><a href="all-samples.html#cb241-11" tabindex="-1"></a>  <span class="fu">fread</span>(<span class="st">&quot;Input Files/allsamples_fst_sd_outlier_cds_snps.txt&quot;</span>)</span>
<span id="cb241-12"><a href="all-samples.html#cb241-12" tabindex="-1"></a></span>
<span id="cb241-13"><a href="all-samples.html#cb241-13" tabindex="-1"></a>fst_sd_fst <span class="ot">&lt;-</span> </span>
<span id="cb241-14"><a href="all-samples.html#cb241-14" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/allsamples_fst_sd_outlierposition.csv&quot;</span>)</span>
<span id="cb241-15"><a href="all-samples.html#cb241-15" tabindex="-1"></a></span>
<span id="cb241-16"><a href="all-samples.html#cb241-16" tabindex="-1"></a>pv_genes <span class="ot">&lt;-</span> </span>
<span id="cb241-17"><a href="all-samples.html#cb241-17" tabindex="-1"></a>  <span class="fu">fread</span>(<span class="st">&quot;Input Files/GSC_HSeal_1.0_HiC_genes.gff3&quot;</span>)</span></code></pre></div>
</div>
<div id="tidy-harbor-seal-genes" class="section level3 hasAnchor" number="5.3.3">
<h3><span class="header-section-number">5.3.3</span> Tidy Harbor seal genes<a href="all-samples.html#tidy-harbor-seal-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="all-samples.html#cb242-1" tabindex="-1"></a>pv_genes_tidy <span class="ot">&lt;-</span> </span>
<span id="cb242-2"><a href="all-samples.html#cb242-2" tabindex="-1"></a>  pv_genes <span class="sc">%&gt;%</span> </span>
<span id="cb242-3"><a href="all-samples.html#cb242-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene =</span> V9,</span>
<span id="cb242-4"><a href="all-samples.html#cb242-4" tabindex="-1"></a>         <span class="at">gene =</span> <span class="fu">str_split_fixed</span>(gene, <span class="st">&quot;;&quot;</span>, <span class="dv">5</span>),</span>
<span id="cb242-5"><a href="all-samples.html#cb242-5" tabindex="-1"></a>         <span class="at">gene =</span> <span class="fu">str_extract</span>(gene[,<span class="dv">4</span>], <span class="st">&quot;(?&lt;=to ).*(?=:)&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb242-6"><a href="all-samples.html#cb242-6" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene))</span></code></pre></div>
</div>
<div id="fst-5-sd" class="section level3 hasAnchor" number="5.3.4">
<h3><span class="header-section-number">5.3.4</span> FST 5 SD<a href="all-samples.html#fst-5-sd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="id-genes" class="section level4 hasAnchor" number="5.3.4.1">
<h4><span class="header-section-number">5.3.4.1</span> ID genes<a href="all-samples.html#id-genes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="all-samples.html#cb243-1" tabindex="-1"></a>fst_sd_genes_tidy <span class="ot">&lt;-</span> </span>
<span id="cb243-2"><a href="all-samples.html#cb243-2" tabindex="-1"></a>  fst_sd_genes <span class="sc">%&gt;%</span> </span>
<span id="cb243-3"><a href="all-samples.html#cb243-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene =</span> V9,</span>
<span id="cb243-4"><a href="all-samples.html#cb243-4" tabindex="-1"></a>         <span class="at">gene =</span> <span class="fu">str_split_fixed</span>(gene, <span class="st">&quot;;&quot;</span>, <span class="dv">5</span>),</span>
<span id="cb243-5"><a href="all-samples.html#cb243-5" tabindex="-1"></a>         <span class="at">gene =</span> <span class="fu">str_extract</span>(gene[,<span class="dv">4</span>], <span class="st">&quot;(?&lt;=to ).*(?=:)&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb243-6"><a href="all-samples.html#cb243-6" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb243-7"><a href="all-samples.html#cb243-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> </span>
<span id="cb243-8"><a href="all-samples.html#cb243-8" tabindex="-1"></a>           <span class="fu">ifelse</span>(V4 <span class="sc">%in%</span> fst_sd_cds<span class="sc">$</span>V4, <span class="st">&quot;cds&quot;</span>, </span>
<span id="cb243-9"><a href="all-samples.html#cb243-9" tabindex="-1"></a>                  <span class="fu">ifelse</span>(V4 <span class="sc">%in%</span> fst_sd_exons, <span class="st">&quot;exon&quot;</span>, <span class="st">&quot;gene&quot;</span>)))</span>
<span id="cb243-10"><a href="all-samples.html#cb243-10" tabindex="-1"></a></span>
<span id="cb243-11"><a href="all-samples.html#cb243-11" tabindex="-1"></a><span class="fu">write.csv</span>(fst_sd_genes_tidy, <span class="st">&quot;Output Files/allsamples_fst_sd_snp_geneid.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="go-terms" class="section level4 hasAnchor" number="5.3.4.2">
<h4><span class="header-section-number">5.3.4.2</span> GO Terms<a href="all-samples.html#go-terms" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="all-samples.html#cb244-1" tabindex="-1"></a>go_bp_sd <span class="ot">&lt;-</span> </span>
<span id="cb244-2"><a href="all-samples.html#cb244-2" tabindex="-1"></a>  <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb244-3"><a href="all-samples.html#cb244-3" tabindex="-1"></a>              <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Biological_Process&quot;</span>,</span>
<span id="cb244-4"><a href="all-samples.html#cb244-4" tabindex="-1"></a>              <span class="at">interestGene =</span> fst_sd_genes_tidy<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb244-5"><a href="all-samples.html#cb244-5" tabindex="-1"></a>              <span class="at">referenceGene =</span> pv_genes_tidy<span class="sc">$</span>gene, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>,</span>
<span id="cb244-6"><a href="all-samples.html#cb244-6" tabindex="-1"></a>              <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb244-7"><a href="all-samples.html#cb244-7" tabindex="-1"></a>              <span class="at">isOutput =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
<pre><code>## Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
## minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="all-samples.html#cb247-1" tabindex="-1"></a>go_cc_sd <span class="ot">&lt;-</span> </span>
<span id="cb247-2"><a href="all-samples.html#cb247-2" tabindex="-1"></a>  <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb247-3"><a href="all-samples.html#cb247-3" tabindex="-1"></a>              <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Cellular_Component&quot;</span>,</span>
<span id="cb247-4"><a href="all-samples.html#cb247-4" tabindex="-1"></a>              <span class="at">interestGene =</span> fst_sd_genes_tidy<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb247-5"><a href="all-samples.html#cb247-5" tabindex="-1"></a>              <span class="at">referenceGene =</span> pv_genes_tidy<span class="sc">$</span>gene, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>,</span>
<span id="cb247-6"><a href="all-samples.html#cb247-6" tabindex="-1"></a>              <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb247-7"><a href="all-samples.html#cb247-7" tabindex="-1"></a>              <span class="at">isOutput =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
<pre><code>## Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
## minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="all-samples.html#cb250-1" tabindex="-1"></a>go_mf_sd <span class="ot">&lt;-</span> </span>
<span id="cb250-2"><a href="all-samples.html#cb250-2" tabindex="-1"></a>  <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb250-3"><a href="all-samples.html#cb250-3" tabindex="-1"></a>              <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Molecular_Function&quot;</span>,</span>
<span id="cb250-4"><a href="all-samples.html#cb250-4" tabindex="-1"></a>              <span class="at">interestGene =</span> fst_sd_genes_tidy<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb250-5"><a href="all-samples.html#cb250-5" tabindex="-1"></a>              <span class="at">referenceGene =</span> pv_genes_tidy<span class="sc">$</span>gene, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>,</span>
<span id="cb250-6"><a href="all-samples.html#cb250-6" tabindex="-1"></a>              <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb250-7"><a href="all-samples.html#cb250-7" tabindex="-1"></a>              <span class="at">isOutput =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
<pre><code>## Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
## minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
</div>
<div id="kegg-pathway" class="section level4 hasAnchor" number="5.3.4.3">
<h4><span class="header-section-number">5.3.4.3</span> KEGG Pathway<a href="all-samples.html#kegg-pathway" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="all-samples.html#cb253-1" tabindex="-1"></a>kegg_sd <span class="ot">&lt;-</span></span>
<span id="cb253-2"><a href="all-samples.html#cb253-2" tabindex="-1"></a>  <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod =</span> <span class="st">&quot;ORA&quot;</span>, <span class="at">organism =</span> <span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb253-3"><a href="all-samples.html#cb253-3" tabindex="-1"></a>              <span class="at">enrichDatabase =</span> <span class="st">&quot;pathway_KEGG&quot;</span>,</span>
<span id="cb253-4"><a href="all-samples.html#cb253-4" tabindex="-1"></a>              <span class="at">interestGene =</span> fst_sd_genes_tidy<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb253-5"><a href="all-samples.html#cb253-5" tabindex="-1"></a>              <span class="at">referenceGene =</span> pv_genes_tidy<span class="sc">$</span>gene, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>,</span>
<span id="cb253-6"><a href="all-samples.html#cb253-6" tabindex="-1"></a>              <span class="at">minNum =</span> <span class="dv">5</span>, <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">isOutput =</span> <span class="cn">FALSE</span>) </span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="all-samples.html#cb255-1" tabindex="-1"></a>kegg_sd <span class="sc">%&gt;%</span></span>
<span id="cb255-2"><a href="all-samples.html#cb255-2" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb255-3"><a href="all-samples.html#cb255-3" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">&quot;basic&quot;</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
link
</th>
<th style="text-align:right;">
size
</th>
<th style="text-align:right;">
overlap
</th>
<th style="text-align:right;">
expect
</th>
<th style="text-align:right;">
enrichmentRatio
</th>
<th style="text-align:right;">
pValue
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:left;">
overlapId
</th>
<th style="text-align:left;">
userId
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
hsa04931
</td>
<td style="text-align:left;">
Insulin resistance
</td>
<td style="text-align:left;">
<a href="http://www.kegg.jp/kegg-bin/show_pathway?hsa04931+11001+32+5291+5524+5792+5836" class="uri">http://www.kegg.jp/kegg-bin/show_pathway?hsa04931+11001+32+5291+5524+5792+5836</a>
</td>
<td style="text-align:right;">
101
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.6380715
</td>
<td style="text-align:right;">
9.403335
</td>
<td style="text-align:right;">
3.45e-05
</td>
<td style="text-align:right;">
0.0111415
</td>
<td style="text-align:left;">
11001;32;5291;5524;5792;5836
</td>
<td style="text-align:left;">
PTPA;ACACB;PTPRF;SLC27A2;PYGL;PIK3CB
</td>
</tr>
</tbody>
</table>
</div>
<div id="biomart-annotation" class="section level4 hasAnchor" number="5.3.4.4">
<h4><span class="header-section-number">5.3.4.4</span> Biomart annotation<a href="all-samples.html#biomart-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Each gene was manually annotated with Ensembl identifier to query Ensembls Biomart</p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="all-samples.html#cb256-1" tabindex="-1"></a>snp_ensembl <span class="ot">&lt;-</span></span>
<span id="cb256-2"><a href="all-samples.html#cb256-2" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Input Files/allsamples_fst_sd_snp_ensemblid.csv&quot;</span>) </span>
<span id="cb256-3"><a href="all-samples.html#cb256-3" tabindex="-1"></a></span>
<span id="cb256-4"><a href="all-samples.html#cb256-4" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> </span>
<span id="cb256-5"><a href="all-samples.html#cb256-5" tabindex="-1"></a>  <span class="fu">useEnsembl</span>(<span class="at">biomart =</span> <span class="st">&quot;ensembl&quot;</span>, </span>
<span id="cb256-6"><a href="all-samples.html#cb256-6" tabindex="-1"></a>             <span class="at">dataset =</span> <span class="st">&quot;hsapiens_gene_ensembl&quot;</span>, </span>
<span id="cb256-7"><a href="all-samples.html#cb256-7" tabindex="-1"></a>             <span class="at">mirror =</span> <span class="st">&quot;useast&quot;</span>)</span></code></pre></div>
<pre><code>## Ensembl site unresponsive, trying www mirror</code></pre>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="all-samples.html#cb258-1" tabindex="-1"></a>filters <span class="ot">&lt;-</span> </span>
<span id="cb258-2"><a href="all-samples.html#cb258-2" tabindex="-1"></a>  <span class="fu">listFilters</span>(ensembl)</span>
<span id="cb258-3"><a href="all-samples.html#cb258-3" tabindex="-1"></a></span>
<span id="cb258-4"><a href="all-samples.html#cb258-4" tabindex="-1"></a>attributes <span class="ot">&lt;-</span> </span>
<span id="cb258-5"><a href="all-samples.html#cb258-5" tabindex="-1"></a>  <span class="fu">listAttributes</span>(ensembl)</span></code></pre></div>
<div id="gene-descriptions" class="section level5 hasAnchor" number="5.3.4.4.1">
<h5><span class="header-section-number">5.3.4.4.1</span> Gene Descriptions<a href="all-samples.html#gene-descriptions" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="all-samples.html#cb259-1" tabindex="-1"></a>snp_description <span class="ot">&lt;-</span></span>
<span id="cb259-2"><a href="all-samples.html#cb259-2" tabindex="-1"></a>  <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>, <span class="st">&quot;description&quot;</span>), </span>
<span id="cb259-3"><a href="all-samples.html#cb259-3" tabindex="-1"></a>        <span class="at">filter =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>,</span>
<span id="cb259-4"><a href="all-samples.html#cb259-4" tabindex="-1"></a>        <span class="at">values =</span> snp_ensembl<span class="sc">$</span>Ensembl.ID,</span>
<span id="cb259-5"><a href="all-samples.html#cb259-5" tabindex="-1"></a>        <span class="at">mart =</span> ensembl,</span>
<span id="cb259-6"><a href="all-samples.html#cb259-6" tabindex="-1"></a>        <span class="at">uniqueRows =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="entrez-description" class="section level5 hasAnchor" number="5.3.4.4.2">
<h5><span class="header-section-number">5.3.4.4.2</span> Entrez description<a href="all-samples.html#entrez-description" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="all-samples.html#cb260-1" tabindex="-1"></a>snp_entrez_description <span class="ot">&lt;-</span> </span>
<span id="cb260-2"><a href="all-samples.html#cb260-2" tabindex="-1"></a>  <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,</span>
<span id="cb260-3"><a href="all-samples.html#cb260-3" tabindex="-1"></a>                       <span class="st">&quot;entrezgene_description&quot;</span>),</span>
<span id="cb260-4"><a href="all-samples.html#cb260-4" tabindex="-1"></a>        <span class="at">filter =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>,</span>
<span id="cb260-5"><a href="all-samples.html#cb260-5" tabindex="-1"></a>        <span class="at">values =</span> snp_ensembl<span class="sc">$</span>Ensembl.ID,</span>
<span id="cb260-6"><a href="all-samples.html#cb260-6" tabindex="-1"></a>        <span class="at">mart =</span> ensembl,</span>
<span id="cb260-7"><a href="all-samples.html#cb260-7" tabindex="-1"></a>        <span class="at">uniqueRows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb260-8"><a href="all-samples.html#cb260-8" tabindex="-1"></a></span>
<span id="cb260-9"><a href="all-samples.html#cb260-9" tabindex="-1"></a>snp_entrez_description <span class="sc">%&gt;%</span> </span>
<span id="cb260-10"><a href="all-samples.html#cb260-10" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb260-11"><a href="all-samples.html#cb260-11" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">&quot;basic&quot;</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
ensembl_gene_id
</th>
<th style="text-align:left;">
entrezgene_description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSG00000004948
</td>
<td style="text-align:left;">
calcitonin receptor
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000005206
</td>
<td style="text-align:left;">
signal peptide peptidase like 2B
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000006555
</td>
<td style="text-align:left;">
tetratricopeptide repeat domain 22
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000014123
</td>
<td style="text-align:left;">
UFM1 specific ligase 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000014216
</td>
<td style="text-align:left;">
calpain 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000023697
</td>
<td style="text-align:left;">
deoxyribose-phosphate aldolase
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000025800
</td>
<td style="text-align:left;">
karyopherin subunit alpha 6
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000039560
</td>
<td style="text-align:left;">
retinoic acid induced 14
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000048342
</td>
<td style="text-align:left;">
coiled-coil and C2 domain containing 2A
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000051382
</td>
<td style="text-align:left;">
phosphatidylinositol-4,5-bisphosphate 3-kinase catalytic subunit beta
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000053108
</td>
<td style="text-align:left;">
follistatin like 4
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000064726
</td>
<td style="text-align:left;">
BTB domain containing 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000064961
</td>
<td style="text-align:left;">
high mobility group 20B
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000065427
</td>
<td style="text-align:left;">
lysyl-tRNA synthetase 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000069188
</td>
<td style="text-align:left;">
sidekick cell adhesion molecule 2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000074603
</td>
<td style="text-align:left;">
dipeptidyl peptidase 8
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000075914
</td>
<td style="text-align:left;">
exosome component 7
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000076555
</td>
<td style="text-align:left;">
acetyl-CoA carboxylase beta
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000089199
</td>
<td style="text-align:left;">
chromogranin B
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000100485
</td>
<td style="text-align:left;">
SOS Ras/Rho guanine nucleotide exchange factor 2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000100504
</td>
<td style="text-align:left;">
glycogen phosphorylase L
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000103061
</td>
<td style="text-align:left;">
solute carrier family 7 member 6 opposite strand
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000104731
</td>
<td style="text-align:left;">
kelch domain containing 4
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000109466
</td>
<td style="text-align:left;">
kelch like family member 2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000111727
</td>
<td style="text-align:left;">
host cell factor C2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000113645
</td>
<td style="text-align:left;">
WW and C2 domain containing 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000116120
</td>
<td style="text-align:left;">
phenylalanyl-tRNA synthetase subunit beta
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000116141
</td>
<td style="text-align:left;">
microtubule affinity regulating kinase 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000116198
</td>
<td style="text-align:left;">
centrosomal protein 104
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000116604
</td>
<td style="text-align:left;">
myocyte enhancer factor 2D
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000119383
</td>
<td style="text-align:left;">
protein phosphatase 2 phosphatase activator
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000119414
</td>
<td style="text-align:left;">
protein phosphatase 6 catalytic subunit
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000119844
</td>
<td style="text-align:left;">
aftiphilin
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000119979
</td>
<td style="text-align:left;">
DENN domain containing 10
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000120885
</td>
<td style="text-align:left;">
clusterin
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000122012
</td>
<td style="text-align:left;">
synaptic vesicle glycoprotein 2C
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000123146
</td>
<td style="text-align:left;">
adhesion G protein-coupled receptor E5
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000124571
</td>
<td style="text-align:left;">
exportin 5
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000128342
</td>
<td style="text-align:left;">
LIF interleukin 6 family cytokine
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000129993
</td>
<td style="text-align:left;">
CBFA2/RUNX1 partner transcriptional co-repressor 3
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000130382
</td>
<td style="text-align:left;">
MLLT1 super elongation complex subunit
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000130560
</td>
<td style="text-align:left;">
UBA domain containing 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000132623
</td>
<td style="text-align:left;">
ankyrin repeat and EF-hand domain containing 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000133083
</td>
<td style="text-align:left;">
doublecortin like kinase 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000133962
</td>
<td style="text-align:left;">
cation channel sperm associated auxiliary subunit beta
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000138029
</td>
<td style="text-align:left;">
hydroxyacyl-CoA dehydrogenase trifunctional multienzyme complex subunit beta
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000138448
</td>
<td style="text-align:left;">
integrin subunit alpha V
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000140284
</td>
<td style="text-align:left;">
solute carrier family 27 member 2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000141456
</td>
<td style="text-align:left;">
proline, glutamate and leucine rich protein 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000142319
</td>
<td style="text-align:left;">
solute carrier family 6 member 3
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000142611
</td>
<td style="text-align:left;">
PR/SET domain 16
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000142949
</td>
<td style="text-align:left;">
protein tyrosine phosphatase receptor type F
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000145681
</td>
<td style="text-align:left;">
hyaluronan and proteoglycan link protein 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000146005
</td>
<td style="text-align:left;">
pleckstrin and Sec7 domain containing 2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000147601
</td>
<td style="text-align:left;">
telomeric repeat binding factor 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000148498
</td>
<td style="text-align:left;">
par-3 family cell polarity regulator
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000148824
</td>
<td style="text-align:left;">
mitochondrial ribosome associated GTPase 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000149547
</td>
<td style="text-align:left;">
EI24 autophagy associated transmembrane protein
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000150551
</td>
<td style="text-align:left;">
LY6/PLAUR domain containing 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000151572
</td>
<td style="text-align:left;">
anoctamin 4
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000152291
</td>
<td style="text-align:left;">
trans-golgi network protein 2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000152700
</td>
<td style="text-align:left;">
secretion associated Ras related GTPase 1B
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000154269
</td>
<td style="text-align:left;">
ectonucleotide pyrophosphatase/phosphodiesterase 3
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000158270
</td>
<td style="text-align:left;">
collectin subfamily member 12
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000158315
</td>
<td style="text-align:left;">
rhomboid like 2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000158683
</td>
<td style="text-align:left;">
polycystin 1 like 1, transient receptor potential channel interacting
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000159398
</td>
<td style="text-align:left;">
carboxylesterase 5A
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000159409
</td>
<td style="text-align:left;">
CUGBP Elav-like family member 3
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000162390
</td>
<td style="text-align:left;">
acyl-CoA thioesterase 11
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000162851
</td>
<td style="text-align:left;">
transcription factor B2, mitochondrial
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000163636
</td>
<td style="text-align:left;">
proteasome 26S subunit, non-ATPase 6
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000163686
</td>
<td style="text-align:left;">
abhydrolase domain containing 6, acylglycerol lipase
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000165392
</td>
<td style="text-align:left;">
WRN RecQ like helicase
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000166411
</td>
<td style="text-align:left;">
isocitrate dehydrogenase (NAD(+)) 3 catalytic subunit alpha
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000166796
</td>
<td style="text-align:left;">
lactate dehydrogenase C
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000167994
</td>
<td style="text-align:left;">
RAB3A interacting protein like 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000168038
</td>
<td style="text-align:left;">
unc-51 like kinase 4
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000168676
</td>
<td style="text-align:left;">
potassium channel tetramerization domain containing 19
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000168872
</td>
<td style="text-align:left;">
DEAD-box helicase 19A
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000169220
</td>
<td style="text-align:left;">
regulator of G protein signaling 14
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000169994
</td>
<td style="text-align:left;">
myosin VIIB
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000172893
</td>
<td style="text-align:left;">
7-dehydrocholesterol reductase
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000179820
</td>
<td style="text-align:left;">
myeloid associated differentiation marker
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000179919
</td>
<td style="text-align:left;">
olfactory receptor family 10 subfamily A member 7
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000182985
</td>
<td style="text-align:left;">
cell adhesion molecule 1
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000184988
</td>
<td style="text-align:left;">
transmembrane protein 106A
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000185963
</td>
<td style="text-align:left;">
BICD cargo adaptor 2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000186395
</td>
<td style="text-align:left;">
keratin 10
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000186487
</td>
<td style="text-align:left;">
myelin transcription factor 1 like
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000188730
</td>
<td style="text-align:left;">
von Willebrand factor C domain containing 2
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000188869
</td>
<td style="text-align:left;">
transmembrane channel like 3
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000196155
</td>
<td style="text-align:left;">
pleckstrin homology and RhoGEF domain containing G4
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000196369
</td>
<td style="text-align:left;">
SLIT-ROBO Rho GTPase activating protein 2B
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000196549
</td>
<td style="text-align:left;">
membrane metalloendopeptidase
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000197122
</td>
<td style="text-align:left;">
SRC proto-oncogene, non-receptor tyrosine kinase
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000198734
</td>
<td style="text-align:left;">
coagulation factor V
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000213699
</td>
<td style="text-align:left;">
solute carrier family 35 member F6
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000224383
</td>
<td style="text-align:left;">
proline rich 29
</td>
</tr>
</tbody>
</table>
</div>
<div id="go-terms-1" class="section level5 hasAnchor" number="5.3.4.4.3">
<h5><span class="header-section-number">5.3.4.4.3</span> GO terms<a href="all-samples.html#go-terms-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="all-samples.html#cb261-1" tabindex="-1"></a>snp_go <span class="ot">&lt;-</span> </span>
<span id="cb261-2"><a href="all-samples.html#cb261-2" tabindex="-1"></a>  <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,</span>
<span id="cb261-3"><a href="all-samples.html#cb261-3" tabindex="-1"></a>                       <span class="st">&quot;go_id&quot;</span>, <span class="st">&quot;name_1006&quot;</span>, <span class="st">&quot;definition_1006&quot;</span>, <span class="st">&quot;namespace_1003&quot;</span>),</span>
<span id="cb261-4"><a href="all-samples.html#cb261-4" tabindex="-1"></a>        <span class="at">filter =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>,</span>
<span id="cb261-5"><a href="all-samples.html#cb261-5" tabindex="-1"></a>        <span class="at">values =</span> snp_ensembl<span class="sc">$</span>Ensembl.ID,</span>
<span id="cb261-6"><a href="all-samples.html#cb261-6" tabindex="-1"></a>        <span class="at">mart =</span> ensembl,</span>
<span id="cb261-7"><a href="all-samples.html#cb261-7" tabindex="-1"></a>        <span class="at">uniqueRows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb261-8"><a href="all-samples.html#cb261-8" tabindex="-1"></a></span>
<span id="cb261-9"><a href="all-samples.html#cb261-9" tabindex="-1"></a>snp_go_tidy <span class="ot">&lt;-</span></span>
<span id="cb261-10"><a href="all-samples.html#cb261-10" tabindex="-1"></a>  snp_go <span class="sc">%&gt;%</span> </span>
<span id="cb261-11"><a href="all-samples.html#cb261-11" tabindex="-1"></a>  <span class="fu">group_by</span>(name_1006) <span class="sc">%&gt;%</span> </span>
<span id="cb261-12"><a href="all-samples.html#cb261-12" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">freq =</span> <span class="fu">n</span>())</span>
<span id="cb261-13"><a href="all-samples.html#cb261-13" tabindex="-1"></a></span>
<span id="cb261-14"><a href="all-samples.html#cb261-14" tabindex="-1"></a>snp_go <span class="sc">%&gt;%</span> </span>
<span id="cb261-15"><a href="all-samples.html#cb261-15" tabindex="-1"></a>  <span class="fu">filter</span>(namespace_1003 <span class="sc">==</span> <span class="st">&quot;biological_process&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb261-16"><a href="all-samples.html#cb261-16" tabindex="-1"></a>  <span class="fu">group_by</span>(name_1006) <span class="sc">%&gt;%</span> </span>
<span id="cb261-17"><a href="all-samples.html#cb261-17" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">freq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb261-18"><a href="all-samples.html#cb261-18" tabindex="-1"></a>  <span class="fu">filter</span>(freq <span class="sc">&gt;</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb261-19"><a href="all-samples.html#cb261-19" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_reorder</span>(name_1006, freq, <span class="at">.desc=</span><span class="cn">TRUE</span>), </span>
<span id="cb261-20"><a href="all-samples.html#cb261-20" tabindex="-1"></a>                <span class="at">y =</span> freq)) <span class="sc">+</span></span>
<span id="cb261-21"><a href="all-samples.html#cb261-21" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb261-22"><a href="all-samples.html#cb261-22" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb261-23"><a href="all-samples.html#cb261-23" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;GO Term&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb261-24"><a href="all-samples.html#cb261-24" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb261-25"><a href="all-samples.html#cb261-25" tabindex="-1"></a>  <span class="fu">coord_flip</span>() </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-77-1.png" width="672" /></p>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="all-samples.html#cb262-1" tabindex="-1"></a>snp_go <span class="sc">%&gt;%</span> </span>
<span id="cb262-2"><a href="all-samples.html#cb262-2" tabindex="-1"></a>  <span class="fu">filter</span>(namespace_1003 <span class="sc">==</span> <span class="st">&quot;molecular_function&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb262-3"><a href="all-samples.html#cb262-3" tabindex="-1"></a>  <span class="fu">group_by</span>(name_1006) <span class="sc">%&gt;%</span> </span>
<span id="cb262-4"><a href="all-samples.html#cb262-4" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">freq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb262-5"><a href="all-samples.html#cb262-5" tabindex="-1"></a>  <span class="fu">filter</span>(freq <span class="sc">&gt;</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb262-6"><a href="all-samples.html#cb262-6" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_reorder</span>(name_1006, freq, <span class="at">.desc=</span><span class="cn">TRUE</span>), </span>
<span id="cb262-7"><a href="all-samples.html#cb262-7" tabindex="-1"></a>                <span class="at">y =</span> freq)) <span class="sc">+</span></span>
<span id="cb262-8"><a href="all-samples.html#cb262-8" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb262-9"><a href="all-samples.html#cb262-9" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb262-10"><a href="all-samples.html#cb262-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;GO Term&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb262-11"><a href="all-samples.html#cb262-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb262-12"><a href="all-samples.html#cb262-12" tabindex="-1"></a>  <span class="fu">coord_flip</span>() </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-77-2.png" width="672" /></p>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="all-samples.html#cb263-1" tabindex="-1"></a>snp_go <span class="sc">%&gt;%</span> </span>
<span id="cb263-2"><a href="all-samples.html#cb263-2" tabindex="-1"></a>  <span class="fu">filter</span>(namespace_1003 <span class="sc">==</span> <span class="st">&quot;cellular_component&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb263-3"><a href="all-samples.html#cb263-3" tabindex="-1"></a>  <span class="fu">group_by</span>(name_1006) <span class="sc">%&gt;%</span> </span>
<span id="cb263-4"><a href="all-samples.html#cb263-4" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">freq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb263-5"><a href="all-samples.html#cb263-5" tabindex="-1"></a>  <span class="fu">filter</span>(freq <span class="sc">&gt;</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb263-6"><a href="all-samples.html#cb263-6" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_reorder</span>(name_1006, freq, <span class="at">.desc=</span><span class="cn">TRUE</span>), </span>
<span id="cb263-7"><a href="all-samples.html#cb263-7" tabindex="-1"></a>                <span class="at">y =</span> freq)) <span class="sc">+</span></span>
<span id="cb263-8"><a href="all-samples.html#cb263-8" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb263-9"><a href="all-samples.html#cb263-9" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb263-10"><a href="all-samples.html#cb263-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;GO Term&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb263-11"><a href="all-samples.html#cb263-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb263-12"><a href="all-samples.html#cb263-12" tabindex="-1"></a>  <span class="fu">coord_flip</span>() </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-77-3.png" width="672" /></p>
</div>
<div id="reactome-pathways" class="section level5 hasAnchor" number="5.3.4.4.4">
<h5><span class="header-section-number">5.3.4.4.4</span> Reactome pathways<a href="all-samples.html#reactome-pathways" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="all-samples.html#cb264-1" tabindex="-1"></a>all_reactome <span class="ot">&lt;-</span> </span>
<span id="cb264-2"><a href="all-samples.html#cb264-2" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(reactomePATHID2NAME)</span>
<span id="cb264-3"><a href="all-samples.html#cb264-3" tabindex="-1"></a></span>
<span id="cb264-4"><a href="all-samples.html#cb264-4" tabindex="-1"></a>snp_reactome <span class="ot">&lt;-</span> </span>
<span id="cb264-5"><a href="all-samples.html#cb264-5" tabindex="-1"></a>  <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,</span>
<span id="cb264-6"><a href="all-samples.html#cb264-6" tabindex="-1"></a>                       <span class="st">&quot;reactome&quot;</span>),</span>
<span id="cb264-7"><a href="all-samples.html#cb264-7" tabindex="-1"></a>        <span class="at">filter =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>,</span>
<span id="cb264-8"><a href="all-samples.html#cb264-8" tabindex="-1"></a>        <span class="at">values =</span> snp_ensembl<span class="sc">$</span>Ensembl.ID,</span>
<span id="cb264-9"><a href="all-samples.html#cb264-9" tabindex="-1"></a>        <span class="at">mart =</span> ensembl,</span>
<span id="cb264-10"><a href="all-samples.html#cb264-10" tabindex="-1"></a>        <span class="at">uniqueRows =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb264-11"><a href="all-samples.html#cb264-11" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>reactome <span class="sc">==</span> <span class="st">&quot;&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb264-12"><a href="all-samples.html#cb264-12" tabindex="-1"></a>  <span class="fu">merge</span>(., all_reactome, <span class="at">by.x =</span> <span class="st">&quot;reactome&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;DB_ID&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb264-13"><a href="all-samples.html#cb264-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">path_name =</span> <span class="fu">gsub</span>(<span class="st">&quot;Homo sapiens: &quot;</span>, <span class="st">&quot;&quot;</span>, path_name))</span>
<span id="cb264-14"><a href="all-samples.html#cb264-14" tabindex="-1"></a></span>
<span id="cb264-15"><a href="all-samples.html#cb264-15" tabindex="-1"></a>snp_reactome <span class="sc">%&gt;%</span> </span>
<span id="cb264-16"><a href="all-samples.html#cb264-16" tabindex="-1"></a>  <span class="fu">group_by</span>(path_name) <span class="sc">%&gt;%</span> </span>
<span id="cb264-17"><a href="all-samples.html#cb264-17" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">freq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb264-18"><a href="all-samples.html#cb264-18" tabindex="-1"></a>  <span class="fu">filter</span>(freq <span class="sc">&gt;</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb264-19"><a href="all-samples.html#cb264-19" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_reorder</span>(path_name, freq, <span class="at">.desc=</span><span class="cn">TRUE</span>), </span>
<span id="cb264-20"><a href="all-samples.html#cb264-20" tabindex="-1"></a>                <span class="at">y =</span> freq)) <span class="sc">+</span></span>
<span id="cb264-21"><a href="all-samples.html#cb264-21" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb264-22"><a href="all-samples.html#cb264-22" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb264-23"><a href="all-samples.html#cb264-23" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Reactome Pathway&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Frequeny&quot;</span>) <span class="sc">+</span></span>
<span id="cb264-24"><a href="all-samples.html#cb264-24" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb264-25"><a href="all-samples.html#cb264-25" tabindex="-1"></a>  <span class="fu">coord_flip</span>() </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-78-1.png" width="672" /></p>
</div>
</div>
<div id="reactome-pathway-over-representation" class="section level4 hasAnchor" number="5.3.4.5">
<h4><span class="header-section-number">5.3.4.5</span> Reactome Pathway Over-representation<a href="all-samples.html#reactome-pathway-over-representation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="all-samples.html#cb265-1" tabindex="-1"></a><span class="co"># Convert to entrez gene ID</span></span>
<span id="cb265-2"><a href="all-samples.html#cb265-2" tabindex="-1"></a>snp_entrez_id <span class="ot">&lt;-</span> </span>
<span id="cb265-3"><a href="all-samples.html#cb265-3" tabindex="-1"></a>  <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>, <span class="st">&quot;entrezgene_id&quot;</span>),</span>
<span id="cb265-4"><a href="all-samples.html#cb265-4" tabindex="-1"></a>        <span class="at">filter =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>,</span>
<span id="cb265-5"><a href="all-samples.html#cb265-5" tabindex="-1"></a>        <span class="at">values =</span> snp_ensembl<span class="sc">$</span>Ensembl.ID,</span>
<span id="cb265-6"><a href="all-samples.html#cb265-6" tabindex="-1"></a>        <span class="at">mart =</span> ensembl,</span>
<span id="cb265-7"><a href="all-samples.html#cb265-7" tabindex="-1"></a>        <span class="at">uniqueRows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb265-8"><a href="all-samples.html#cb265-8" tabindex="-1"></a></span>
<span id="cb265-9"><a href="all-samples.html#cb265-9" tabindex="-1"></a>snp_overrep_reactome <span class="ot">&lt;-</span> </span>
<span id="cb265-10"><a href="all-samples.html#cb265-10" tabindex="-1"></a>  <span class="fu">enrichPathway</span>(<span class="at">gene =</span> snp_entrez_id<span class="sc">$</span>entrezgene_id,</span>
<span id="cb265-11"><a href="all-samples.html#cb265-11" tabindex="-1"></a>                <span class="at">organism =</span> <span class="st">&quot;human&quot;</span>,</span>
<span id="cb265-12"><a href="all-samples.html#cb265-12" tabindex="-1"></a>                <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb265-13"><a href="all-samples.html#cb265-13" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;fdr&quot;</span>,</span>
<span id="cb265-14"><a href="all-samples.html#cb265-14" tabindex="-1"></a>                <span class="at">readable =</span> <span class="cn">TRUE</span>,</span>
<span id="cb265-15"><a href="all-samples.html#cb265-15" tabindex="-1"></a>                <span class="at">universe =</span> pv_genes_tidy<span class="sc">$</span>gene,</span>
<span id="cb265-16"><a href="all-samples.html#cb265-16" tabindex="-1"></a>                <span class="at">minGSSize =</span> <span class="dv">2</span>)<span class="sc">$</span>result</span></code></pre></div>
<pre><code>## Loading required package: org.Hs.eg.db</code></pre>
<pre><code>## </code></pre>
<pre><code>## No gene sets have size between 2 and 500 ...</code></pre>
<pre><code>## --&gt; return NULL...</code></pre>
</div>
<div id="exonic-cds-genes" class="section level4 hasAnchor" number="5.3.4.6">
<h4><span class="header-section-number">5.3.4.6</span> Exonic, CDS Genes<a href="all-samples.html#exonic-cds-genes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="identify-genes" class="section level5 hasAnchor" number="5.3.4.6.1">
<h5><span class="header-section-number">5.3.4.6.1</span> Identify genes<a href="all-samples.html#identify-genes" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="all-samples.html#cb270-1" tabindex="-1"></a>fst_sd_cds_tidy <span class="ot">&lt;-</span> </span>
<span id="cb270-2"><a href="all-samples.html#cb270-2" tabindex="-1"></a>  fst_sd_cds <span class="sc">%&gt;%</span></span>
<span id="cb270-3"><a href="all-samples.html#cb270-3" tabindex="-1"></a>  <span class="fu">merge</span>(., fst_sd_genes_tidy[,<span class="fu">c</span>(<span class="dv">12</span><span class="sc">:</span><span class="dv">14</span>)], <span class="at">by =</span> <span class="st">&quot;V12&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb270-4"><a href="all-samples.html#cb270-4" tabindex="-1"></a>  <span class="fu">merge</span>(., snp_ensembl, <span class="at">by.x =</span> <span class="st">&quot;gene&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;Gene&quot;</span>)</span>
<span id="cb270-5"><a href="all-samples.html#cb270-5" tabindex="-1"></a></span>
<span id="cb270-6"><a href="all-samples.html#cb270-6" tabindex="-1"></a>fst_sd_exons_tidy <span class="ot">&lt;-</span> </span>
<span id="cb270-7"><a href="all-samples.html#cb270-7" tabindex="-1"></a>  fst_sd_exons <span class="sc">%&gt;%</span></span>
<span id="cb270-8"><a href="all-samples.html#cb270-8" tabindex="-1"></a>  <span class="fu">merge</span>(., fst_sd_genes_tidy[,<span class="fu">c</span>(<span class="dv">12</span><span class="sc">:</span><span class="dv">14</span>)], <span class="at">by =</span> <span class="st">&quot;V12&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb270-9"><a href="all-samples.html#cb270-9" tabindex="-1"></a>  <span class="fu">merge</span>(., snp_ensembl, <span class="at">by.x =</span> <span class="st">&quot;gene&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;Gene&quot;</span>)</span></code></pre></div>
</div>
<div id="go-terms-2" class="section level5 hasAnchor" number="5.3.4.6.2">
<h5><span class="header-section-number">5.3.4.6.2</span> GO terms<a href="all-samples.html#go-terms-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="all-samples.html#cb271-1" tabindex="-1"></a>fst_sd_cds_tidy <span class="sc">%&gt;%</span> </span>
<span id="cb271-2"><a href="all-samples.html#cb271-2" tabindex="-1"></a>  <span class="fu">merge</span>(., snp_go, <span class="at">by.x =</span> <span class="st">&quot;Ensembl.ID&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb271-3"><a href="all-samples.html#cb271-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(name_1006, namespace_1003) <span class="sc">%&gt;%</span> </span>
<span id="cb271-4"><a href="all-samples.html#cb271-4" tabindex="-1"></a>  <span class="fu">arrange</span>(namespace_1003) <span class="sc">%&gt;%</span> </span>
<span id="cb271-5"><a href="all-samples.html#cb271-5" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb271-6"><a href="all-samples.html#cb271-6" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">&quot;basic&quot;</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
name_1006
</th>
<th style="text-align:left;">
namespace_1003
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
autophagy
</td>
<td style="text-align:left;">
biological_process
</td>
</tr>
<tr>
<td style="text-align:left;">
apoptotic process
</td>
<td style="text-align:left;">
biological_process
</td>
</tr>
<tr>
<td style="text-align:left;">
negative regulation of cell growth
</td>
<td style="text-align:left;">
biological_process
</td>
</tr>
<tr>
<td style="text-align:left;">
macroautophagy
</td>
<td style="text-align:left;">
biological_process
</td>
</tr>
<tr>
<td style="text-align:left;">
nucleus
</td>
<td style="text-align:left;">
cellular_component
</td>
</tr>
<tr>
<td style="text-align:left;">
cytoplasm
</td>
<td style="text-align:left;">
cellular_component
</td>
</tr>
<tr>
<td style="text-align:left;">
membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
</tr>
<tr>
<td style="text-align:left;">
endoplasmic reticulum
</td>
<td style="text-align:left;">
cellular_component
</td>
</tr>
<tr>
<td style="text-align:left;">
endoplasmic reticulum membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
</tr>
<tr>
<td style="text-align:left;">
cytosol
</td>
<td style="text-align:left;">
cellular_component
</td>
</tr>
<tr>
<td style="text-align:left;">
Golgi apparatus
</td>
<td style="text-align:left;">
cellular_component
</td>
</tr>
<tr>
<td style="text-align:left;">
nuclear membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
</tr>
<tr>
<td style="text-align:left;">
protein binding
</td>
<td style="text-align:left;">
molecular_function
</td>
</tr>
<tr>
<td style="text-align:left;">
importin-alpha family protein binding
</td>
<td style="text-align:left;">
molecular_function
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="all-samples.html#cb272-1" tabindex="-1"></a>fst_sd_exons_tidy <span class="sc">%&gt;%</span> </span>
<span id="cb272-2"><a href="all-samples.html#cb272-2" tabindex="-1"></a>  <span class="fu">merge</span>(., snp_go, <span class="at">by.x =</span> <span class="st">&quot;Ensembl.ID&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb272-3"><a href="all-samples.html#cb272-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(name_1006, namespace_1003) <span class="sc">%&gt;%</span> </span>
<span id="cb272-4"><a href="all-samples.html#cb272-4" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>name_1006 <span class="sc">==</span> <span class="st">&quot;&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb272-5"><a href="all-samples.html#cb272-5" tabindex="-1"></a>  <span class="fu">group_by</span>(name_1006, namespace_1003) <span class="sc">%&gt;%</span> </span>
<span id="cb272-6"><a href="all-samples.html#cb272-6" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">freq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb272-7"><a href="all-samples.html#cb272-7" tabindex="-1"></a>  <span class="fu">arrange</span>(namespace_1003) <span class="sc">%&gt;%</span> </span>
<span id="cb272-8"><a href="all-samples.html#cb272-8" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb272-9"><a href="all-samples.html#cb272-9" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">&quot;basic&quot;</span>)</span></code></pre></div>
<pre><code>## `summarise()` has grouped output by &#39;name_1006&#39;. You can
## override using the `.groups` argument.</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
name_1006
</th>
<th style="text-align:left;">
namespace_1003
</th>
<th style="text-align:right;">
freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
DNA-templated transcription
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
RNA export from nucleus
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
apoptotic process
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
autophagy
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
cell adhesion
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
cell differentiation
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
central nervous system development
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
glial cell differentiation
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
granulocyte differentiation
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
intracellular protein transport
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
macroautophagy
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
membrane protein ectodomain proteolysis
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
membrane protein intracellular domain proteolysis
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
membrane protein proteolysis
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
miRNA metabolic process
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
monoatomic ion transmembrane transport
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
muscle organ development
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
negative regulation of DNA-templated transcription
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
negative regulation of cell growth
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
negative regulation of cell population proliferation
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
negative regulation of glycolytic process
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
neurogenesis
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
positive regulation of neuroblast proliferation
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
positive regulation of proteasomal ubiquitin-dependent protein catabolic process
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
pre-miRNA export from nucleus
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
protein export from nucleus
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
protein transport
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
protein ubiquitination
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
proteolysis
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
regulation of aerobic respiration
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
regulation of immune response
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
regulation of protein binding
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
regulation of tumor necrosis factor-mediated signaling pathway
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
regulatory ncRNA-mediated gene silencing
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
response to hypoxia
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
skeletal system development
</td>
<td style="text-align:left;">
biological_process
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
Golgi apparatus
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
Golgi membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
Golgi-associated vesicle membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
P-body
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
RISC complex
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
RNA nuclear export complex
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
actin cytoskeleton
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
centrosome
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
collagen-containing extracellular matrix
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
cytoplasm
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
cytoplasmic ribonucleoprotein granule
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
cytoplasmic side of endoplasmic reticulum membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
cytosol
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
endoplasmic reticulum
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
endoplasmic reticulum membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
endosome
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
endosome membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
extracellular matrix
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
extracellular region
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
extracellular space
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
lumenal side of endoplasmic reticulum membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
lysosomal membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
lysosome
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
nuclear membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
nucleolus
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
nucleoplasm
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
nucleus
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
perineuronal net
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
plasma membrane
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
protein-containing complex
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
synapse
</td>
<td style="text-align:left;">
cellular_component
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
RISC complex binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
RNA binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
aspartic endopeptidase activity, intramembrane cleaving
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
cullin family protein binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
extracellular matrix structural constituent conferring compression resistance
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
hyaluronic acid binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
identical protein binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
importin-alpha family protein binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
mRNA binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
mechanosensitive monoatomic ion channel activity
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
metal ion binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
nuclear export signal receptor activity
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
peptidase activity
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
pre-miRNA binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
protein binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
protein homodimerization activity
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
small GTPase binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
tRNA binding
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
transcription corepressor activity
</td>
<td style="text-align:left;">
molecular_function
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
</div>
<div id="reactome-pathways-1" class="section level5 hasAnchor" number="5.3.4.6.3">
<h5><span class="header-section-number">5.3.4.6.3</span> Reactome Pathways<a href="all-samples.html#reactome-pathways-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="all-samples.html#cb274-1" tabindex="-1"></a>fst_sd_cds_tidy <span class="sc">%&gt;%</span> </span>
<span id="cb274-2"><a href="all-samples.html#cb274-2" tabindex="-1"></a>  <span class="fu">merge</span>(., snp_reactome, <span class="at">by.x =</span> <span class="st">&quot;Ensembl.ID&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb274-3"><a href="all-samples.html#cb274-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(reactome, path_name) <span class="sc">%&gt;%</span> </span>
<span id="cb274-4"><a href="all-samples.html#cb274-4" tabindex="-1"></a>  <span class="fu">arrange</span>(path_name) <span class="sc">%&gt;%</span> </span>
<span id="cb274-5"><a href="all-samples.html#cb274-5" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb274-6"><a href="all-samples.html#cb274-6" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">&quot;basic&quot;</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
reactome
</th>
<th style="text-align:left;">
path_name
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
:
</td>
<td style="text-align:left;">
:
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="all-samples.html#cb275-1" tabindex="-1"></a>fst_sd_exons_tidy <span class="sc">%&gt;%</span> </span>
<span id="cb275-2"><a href="all-samples.html#cb275-2" tabindex="-1"></a>  <span class="fu">merge</span>(., snp_reactome, <span class="at">by.x =</span> <span class="st">&quot;Ensembl.ID&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb275-3"><a href="all-samples.html#cb275-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(reactome, path_name) <span class="sc">%&gt;%</span> </span>
<span id="cb275-4"><a href="all-samples.html#cb275-4" tabindex="-1"></a>  <span class="fu">arrange</span>(path_name) <span class="sc">%&gt;%</span> </span>
<span id="cb275-5"><a href="all-samples.html#cb275-5" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb275-6"><a href="all-samples.html#cb275-6" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">&quot;basic&quot;</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
reactome
</th>
<th style="text-align:left;">
path_name
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
R-HSA-1280218
</td>
<td style="text-align:left;">
Adaptive Immune System
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-983168
</td>
<td style="text-align:left;">
Antigen processing: Ubiquitination &amp; Proteasome degradation
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-983169
</td>
<td style="text-align:left;">
Class I MHC mediated antigen processing &amp; presentation
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-73887
</td>
<td style="text-align:left;">
Death Receptor Signaling
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-3000178
</td>
<td style="text-align:left;">
ECM proteoglycans
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-1474244
</td>
<td style="text-align:left;">
Extracellular matrix organization
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-211000
</td>
<td style="text-align:left;">
Gene Silencing by RNA
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-74160
</td>
<td style="text-align:left;">
Gene expression (Transcription)
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-168256
</td>
<td style="text-align:left;">
Immune System
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-392499
</td>
<td style="text-align:left;">
Metabolism of proteins
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-203927
</td>
<td style="text-align:left;">
MicroRNA (miRNA) biogenesis
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-8951664
</td>
<td style="text-align:left;">
Neddylation
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-597592
</td>
<td style="text-align:left;">
Post-translational protein modification
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-5357905
</td>
<td style="text-align:left;">
Regulation of TNFR1 signaling
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-162582
</td>
<td style="text-align:left;">
Signal Transduction
</td>
</tr>
<tr>
<td style="text-align:left;">
R-HSA-75893
</td>
<td style="text-align:left;">
TNF signaling
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gwas" class="section level3 hasAnchor" number="5.3.5">
<h3><span class="header-section-number">5.3.5</span> GWAS<a href="all-samples.html#gwas" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="all-samples.html#cb276-1" tabindex="-1"></a>gwas_snp <span class="ot">&lt;-</span> </span>
<span id="cb276-2"><a href="all-samples.html#cb276-2" tabindex="-1"></a>  gwas_snp <span class="sc">%&gt;%</span> </span>
<span id="cb276-3"><a href="all-samples.html#cb276-3" tabindex="-1"></a>  <span class="fu">filter</span>(fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb276-4"><a href="all-samples.html#cb276-4" tabindex="-1"></a>  <span class="fu">merge</span>(., fst_sd_genes_tidy, <span class="at">by.x =</span> <span class="st">&quot;BP&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;V4&quot;</span>) </span></code></pre></div>
<p>HAPLN1:<br />
Predicted to enable hyaluronic acid binding activity. Predicted to be an extracellular matrix structural constituent conferring compression resistance. Predicted to be involved in nervous system development; positive regulation of neuroblast proliferation; and skeletal system development.</p>
<p>**SNPs in introns that exhibit differences in allele frequency between cases and controls - we say it potentially has some impact on survival. But how does SNP in intronic region impact survivability?</p>
<p>Polymorphism &amp; pathway - how do genetic changes affect pathways that affect higher physiological levels? Single gene, large effect - lots of SNPs in a gene</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="density-plot-filter-metrics.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section",
    "scroll_highlight": true
  },
  "info": false
});
});
</script>

</body>

</html>
