<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 PCA - High Coverage | Pv-WGS-Analysis</title>
  <meta name="description" content="McCosker Dissertation Chapter 3 Analyses" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="3 PCA - High Coverage | Pv-WGS-Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="McCosker Dissertation Chapter 3 Analyses" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 PCA - High Coverage | Pv-WGS-Analysis" />
  
  <meta name="twitter:description" content="McCosker Dissertation Chapter 3 Analyses" />
  

<meta name="author" content="Christina McCosker" />


<meta name="date" content="2025-03-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tidy-data-exploratory.html"/>
<link rel="next" href="density-plot-filter-metrics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Pv-WGS-Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a></li>
<li class="chapter" data-level="2" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html"><i class="fa fa-check"></i><b>2</b> Tidy Data &amp; Exploratory</a>
<ul>
<li class="chapter" data-level="2.1" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#libraries"><i class="fa fa-check"></i><b>2.1</b> Libraries</a></li>
<li class="chapter" data-level="2.2" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#data"><i class="fa fa-check"></i><b>2.2</b> Data</a></li>
<li class="chapter" data-level="2.3" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#descriptive-statistics"><i class="fa fa-check"></i><b>2.3</b> Descriptive Statistics</a></li>
<li class="chapter" data-level="2.4" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#exclude-samples"><i class="fa fa-check"></i><b>2.4</b> Exclude samples</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#sample-statistics"><i class="fa fa-check"></i><b>2.4.1</b> Sample statistics</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#map-of-strandings"><i class="fa fa-check"></i><b>2.5</b> Map of Strandings</a></li>
<li class="chapter" data-level="2.6" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#case-v-control-factors"><i class="fa fa-check"></i><b>2.6</b> Case v Control ~ Factors</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#duringafter-outbreak"><i class="fa fa-check"></i><b>2.6.1</b> During/After Outbreak</a></li>
<li class="chapter" data-level="2.6.2" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#sex"><i class="fa fa-check"></i><b>2.6.2</b> Sex</a></li>
<li class="chapter" data-level="2.6.3" data-path="tidy-data-exploratory.html"><a href="tidy-data-exploratory.html#age"><i class="fa fa-check"></i><b>2.6.3</b> Age</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html"><i class="fa fa-check"></i><b>3</b> PCA - High Coverage</a>
<ul>
<li class="chapter" data-level="3.1" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#libraries-1"><i class="fa fa-check"></i><b>3.1</b> Libraries</a></li>
<li class="chapter" data-level="3.2" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#data-1"><i class="fa fa-check"></i><b>3.2</b> Data</a></li>
<li class="chapter" data-level="3.3" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#vcf-to-gds"><i class="fa fa-check"></i><b>3.3</b> VCF to GDS</a></li>
<li class="chapter" data-level="3.4" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#ld-based-snp-pruning"><i class="fa fa-check"></i><b>3.4</b> LD-based SNP Pruning</a></li>
<li class="chapter" data-level="3.5" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#pca"><i class="fa fa-check"></i><b>3.5</b> PCA</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#pca-original"><i class="fa fa-check"></i><b>3.5.1</b> PCA Original</a></li>
<li class="chapter" data-level="3.5.2" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#pca-wo-outliers"><i class="fa fa-check"></i><b>3.5.2</b> PCA w/o Outliers</a></li>
<li class="chapter" data-level="3.5.3" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#pca-by-chromosome"><i class="fa fa-check"></i><b>3.5.3</b> PCA by Chromosome</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="pca---high-coverage.html"><a href="pca---high-coverage.html#relatedness"><i class="fa fa-check"></i><b>3.6</b> Relatedness</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html"><i class="fa fa-check"></i><b>4</b> Density Plot Filter Metrics</a>
<ul>
<li class="chapter" data-level="4.1" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#libraries-2"><i class="fa fa-check"></i><b>4.1</b> Libraries</a></li>
<li class="chapter" data-level="4.2" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#data-2"><i class="fa fa-check"></i><b>4.2</b> Data</a></li>
<li class="chapter" data-level="4.3" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#density-plots"><i class="fa fa-check"></i><b>4.3</b> Density Plots</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-qd"><i class="fa fa-check"></i><b>4.3.1</b> Metric: QD</a></li>
<li class="chapter" data-level="4.3.2" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-fs"><i class="fa fa-check"></i><b>4.3.2</b> Metric: FS</a></li>
<li class="chapter" data-level="4.3.3" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-sor"><i class="fa fa-check"></i><b>4.3.3</b> Metric: SOR</a></li>
<li class="chapter" data-level="4.3.4" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-mq"><i class="fa fa-check"></i><b>4.3.4</b> Metric: MQ</a></li>
<li class="chapter" data-level="4.3.5" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-mqranksum"><i class="fa fa-check"></i><b>4.3.5</b> Metric: MQRankSum</a></li>
<li class="chapter" data-level="4.3.6" data-path="density-plot-filter-metrics.html"><a href="density-plot-filter-metrics.html#metric-readposranksum"><i class="fa fa-check"></i><b>4.3.6</b> Metric: ReadPosRankSum</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html"><i class="fa fa-check"></i><b>5</b> GWAS - rrBLUP</a>
<ul>
<li class="chapter" data-level="5.1" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#libraries-3"><i class="fa fa-check"></i><b>5.1</b> Libraries</a></li>
<li class="chapter" data-level="5.2" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#data-3"><i class="fa fa-check"></i><b>5.2</b> Data</a></li>
<li class="chapter" data-level="5.3" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#calculating-grm"><i class="fa fa-check"></i><b>5.3</b> Calculating GRM</a></li>
<li class="chapter" data-level="5.4" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#population-structure"><i class="fa fa-check"></i><b>5.4</b> Population structure</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#heatmap"><i class="fa fa-check"></i><b>5.4.1</b> Heatmap</a></li>
<li class="chapter" data-level="5.4.2" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#pca-1"><i class="fa fa-check"></i><b>5.4.2</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#configure-data"><i class="fa fa-check"></i><b>5.5</b> Configure data</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#create-matrix"><i class="fa fa-check"></i><b>5.5.1</b> Create matrix</a></li>
<li class="chapter" data-level="5.5.2" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#configure-phenotype-data"><i class="fa fa-check"></i><b>5.5.2</b> Configure phenotype data</a></li>
<li class="chapter" data-level="5.5.3" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#configure-grm-snp-info"><i class="fa fa-check"></i><b>5.5.3</b> Configure grm &amp; snp info</a></li>
<li class="chapter" data-level="5.5.4" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#gwas-model-1-genomic-relationship-matrix"><i class="fa fa-check"></i><b>5.5.4</b> GWAS Model 1: genomic relationship matrix</a></li>
<li class="chapter" data-level="5.5.5" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#extract-results"><i class="fa fa-check"></i><b>5.5.5</b> Extract results</a></li>
<li class="chapter" data-level="5.5.6" data-path="gwas---rrblup.html"><a href="gwas---rrblup.html#manhattan-plot"><i class="fa fa-check"></i><b>5.5.6</b> Manhattan Plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="fst-based-approach.html"><a href="fst-based-approach.html"><i class="fa fa-check"></i><b>6</b> Fst-based Approach</a>
<ul>
<li class="chapter" data-level="6.1" data-path="fst-based-approach.html"><a href="fst-based-approach.html#libraries-4"><i class="fa fa-check"></i><b>6.1</b> Libraries</a></li>
<li class="chapter" data-level="6.2" data-path="fst-based-approach.html"><a href="fst-based-approach.html#data-4"><i class="fa fa-check"></i><b>6.2</b> Data</a></li>
<li class="chapter" data-level="6.3" data-path="fst-based-approach.html"><a href="fst-based-approach.html#visualize-snp-fst-distribution"><i class="fa fa-check"></i><b>6.3</b> Visualize SNP Fst distribution</a></li>
<li class="chapter" data-level="6.4" data-path="fst-based-approach.html"><a href="fst-based-approach.html#select-outlier-snps"><i class="fa fa-check"></i><b>6.4</b> Select outlier SNPs</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="fst-based-approach.html"><a href="fst-based-approach.html#top-10"><i class="fa fa-check"></i><b>6.4.1</b> Top 10%</a></li>
<li class="chapter" data-level="6.4.2" data-path="fst-based-approach.html"><a href="fst-based-approach.html#sd"><i class="fa fa-check"></i><b>6.4.2</b> &gt; 5SD</a></li>
<li class="chapter" data-level="6.4.3" data-path="fst-based-approach.html"><a href="fst-based-approach.html#extract-snps-600-bp-positions"><i class="fa fa-check"></i><b>6.4.3</b> Extract SNPs &amp; 600 bp positions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="snp-annotation.html"><a href="snp-annotation.html"><i class="fa fa-check"></i><b>7</b> SNP Annotation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="snp-annotation.html"><a href="snp-annotation.html#libraries-5"><i class="fa fa-check"></i><b>7.1</b> Libraries</a></li>
<li class="chapter" data-level="7.2" data-path="snp-annotation.html"><a href="snp-annotation.html#data-5"><i class="fa fa-check"></i><b>7.2</b> Data</a></li>
<li class="chapter" data-level="7.3" data-path="snp-annotation.html"><a href="snp-annotation.html#tidy-harbor-seal-genes"><i class="fa fa-check"></i><b>7.3</b> Tidy Harbor seal genes</a></li>
<li class="chapter" data-level="7.4" data-path="snp-annotation.html"><a href="snp-annotation.html#fst-5-sd"><i class="fa fa-check"></i><b>7.4</b> FST 5 SD</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="snp-annotation.html"><a href="snp-annotation.html#id-genes"><i class="fa fa-check"></i><b>7.4.1</b> ID genes</a></li>
<li class="chapter" data-level="7.4.2" data-path="snp-annotation.html"><a href="snp-annotation.html#go-terms"><i class="fa fa-check"></i><b>7.4.2</b> GO Terms</a></li>
<li class="chapter" data-level="7.4.3" data-path="snp-annotation.html"><a href="snp-annotation.html#kegg-pathway"><i class="fa fa-check"></i><b>7.4.3</b> KEGG Pathway</a></li>
<li class="chapter" data-level="7.4.4" data-path="snp-annotation.html"><a href="snp-annotation.html#biomart-annotation"><i class="fa fa-check"></i><b>7.4.4</b> Biomart annotation</a></li>
<li class="chapter" data-level="7.4.5" data-path="snp-annotation.html"><a href="snp-annotation.html#exonic-cds-genes"><i class="fa fa-check"></i><b>7.4.5</b> Exonic, CDS Genes</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="snp-annotation.html"><a href="snp-annotation.html#gwas"><i class="fa fa-check"></i><b>7.5</b> GWAS</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/christinamccosker/pv-wgs.git" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pv-WGS-Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pca---high-coverage" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> PCA - High Coverage<a href="pca---high-coverage.html#pca---high-coverage" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="libraries-1" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Libraries<a href="pca---high-coverage.html#libraries-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="pca---high-coverage.html#cb25-1" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb25-2"><a href="pca---high-coverage.html#cb25-2" tabindex="-1"></a><span class="fu">library</span>(SNPRelate)</span>
<span id="cb25-3"><a href="pca---high-coverage.html#cb25-3" tabindex="-1"></a><span class="fu">library</span>(gdsfmt)</span>
<span id="cb25-4"><a href="pca---high-coverage.html#cb25-4" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb25-5"><a href="pca---high-coverage.html#cb25-5" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb25-6"><a href="pca---high-coverage.html#cb25-6" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb25-7"><a href="pca---high-coverage.html#cb25-7" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb25-8"><a href="pca---high-coverage.html#cb25-8" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb25-9"><a href="pca---high-coverage.html#cb25-9" tabindex="-1"></a><span class="fu">library</span>(knitr)</span></code></pre></div>
</div>
<div id="data-1" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Data<a href="pca---high-coverage.html#data-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="pca---high-coverage.html#cb26-1" tabindex="-1"></a>meta <span class="ot">&lt;-</span> </span>
<span id="cb26-2"><a href="pca---high-coverage.html#cb26-2" tabindex="-1"></a>  <span class="fu">fread</span>(<span class="st">&quot;Output Files/metadata_tidy.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="pca---high-coverage.html#cb26-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq.id =</span> <span class="fu">gsub</span>(<span class="st">&quot;B5GAN&quot;</span>, <span class="st">&quot;&quot;</span>, seq.id))</span></code></pre></div>
</div>
<div id="vcf-to-gds" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> VCF to GDS<a href="pca---high-coverage.html#vcf-to-gds" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="pca---high-coverage.html#cb27-1" tabindex="-1"></a><span class="co"># Load VCF file and create GDS file</span></span>
<span id="cb27-2"><a href="pca---high-coverage.html#cb27-2" tabindex="-1"></a><span class="co"># snpgdsVCF2GDS(&quot;C:/Users/Christina/OneDrive - University of Maine System/pv-wgs/30xSamples_Genotype_nosex_edit.vcf&quot;, </span></span>
<span id="cb27-3"><a href="pca---high-coverage.html#cb27-3" tabindex="-1"></a><span class="co">#               &quot;C:/Users/Christina/OneDrive - University of Maine System/pv-wgs/30xsamples_Genotype.gds&quot;, </span></span>
<span id="cb27-4"><a href="pca---high-coverage.html#cb27-4" tabindex="-1"></a><span class="co">#               method=&quot;biallelic.only&quot;)</span></span>
<span id="cb27-5"><a href="pca---high-coverage.html#cb27-5" tabindex="-1"></a></span>
<span id="cb27-6"><a href="pca---high-coverage.html#cb27-6" tabindex="-1"></a><span class="fu">snpgdsSummary</span>(<span class="st">&quot;C:/Users/Christina/OneDrive - University of Maine System/pv-wgs/30xSamples_Genotype_nosex.gds&quot;</span>)</span></code></pre></div>
<pre><code>## The file name: C:\Users\Christina\OneDrive - University of Maine System\pv-wgs\30xsamples_Genotype_nosex.gds 
## The total number of samples: 20 
## The total number of SNPs: 3908010 
## SNP genotypes are stored in SNP-major mode (Sample X SNP).</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="pca---high-coverage.html#cb29-1" tabindex="-1"></a>geno <span class="ot">&lt;-</span> <span class="fu">snpgdsOpen</span>(<span class="st">&quot;C:/Users/Christina/OneDrive - University of Maine System/pv-wgs/30xSamples_Genotype_nosex.gds&quot;</span>)</span></code></pre></div>
</div>
<div id="ld-based-snp-pruning" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> LD-based SNP Pruning<a href="pca---high-coverage.html#ld-based-snp-pruning" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Recursively removes SNPs within a sliding window based on pairwise genotypic correlation - remove SNPs in linkage disequilibrium with each other. Conducted chromosome by chromosome (SNPs in a chromosome are independent of SNPs on other chromosomes).Generates set of SNPs that are independent of each other.<br />
Many SNPs are going to be in LD to each other based on ancestry. Researchers will LD prune to reduce the chance of false-negatives and look for ways to prioritize SNPs within significant GWAS hits to increase the chance of finding something casual.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="pca---high-coverage.html#cb30-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1000</span>)</span>
<span id="cb30-2"><a href="pca---high-coverage.html#cb30-2" tabindex="-1"></a></span>
<span id="cb30-3"><a href="pca---high-coverage.html#cb30-3" tabindex="-1"></a>snpset <span class="ot">&lt;-</span> <span class="fu">snpgdsLDpruning</span>(geno, <span class="at">ld.threshold=</span><span class="fl">0.2</span>, <span class="at">autosome.only=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## SNP pruning based on LD:
## Excluding 1,086,533 SNPs (monomorphic: TRUE, MAF: 0.005, missing rate: 0.05)
##     # of samples: 20
##     # of SNPs: 2,821,477
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.2
##     method: composite
## Chrom 1: |====================|====================|
##     2.04%, 6,682 / 327,390 (Wed Mar 19 15:59:01 2025)
## Chrom 2: |====================|====================|
##     1.98%, 7,169 / 361,795 (Wed Mar 19 15:59:02 2025)
## Chrom 3: |====================|====================|
##     1.94%, 6,041 / 310,679 (Wed Mar 19 15:59:02 2025)
## Chrom 4: |====================|====================|
##     2.05%, 7,295 / 355,492 (Wed Mar 19 15:59:03 2025)
## Chrom 5: |====================|====================|
##     1.81%, 3,448 / 190,039 (Wed Mar 19 15:59:03 2025)
## Chrom 6: |====================|====================|
##     2.20%, 5,258 / 239,224 (Wed Mar 19 15:59:04 2025)
## Chrom 7: |====================|====================|
##     2.21%, 3,714 / 168,421 (Wed Mar 19 15:59:05 2025)
## Chrom 8: |====================|====================|
##     2.31%, 2,762 / 119,709 (Wed Mar 19 15:59:05 2025)
## Chrom 9: |====================|====================|
##     2.94%, 4,966 / 169,064 (Wed Mar 19 15:59:05 2025)
## Chrom 10: |====================|====================|
##     2.02%, 4,055 / 200,553 (Wed Mar 19 15:59:06 2025)
## Chrom 11: |====================|====================|
##     2.00%, 5,089 / 254,638 (Wed Mar 19 15:59:06 2025)
## Chrom 12: |====================|====================|
##     2.28%, 6,946 / 305,075 (Wed Mar 19 15:59:07 2025)
## Chrom 13: |====================|====================|
##     2.06%, 5,692 / 276,329 (Wed Mar 19 15:59:07 2025)
## Chrom 14: |====================|====================|
##     2.17%, 5,753 / 264,542 (Wed Mar 19 15:59:08 2025)
## Chrom 16: |====================|====================|
##     1.76%, 6,427 / 365,060 (Wed Mar 19 15:59:08 2025)
## 81,297 markers are selected in total.</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="pca---high-coverage.html#cb32-1" tabindex="-1"></a>snpset.id <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">unname</span>(snpset))</span>
<span id="cb32-2"><a href="pca---high-coverage.html#cb32-2" tabindex="-1"></a></span>
<span id="cb32-3"><a href="pca---high-coverage.html#cb32-3" tabindex="-1"></a>snpset2 <span class="ot">&lt;-</span> <span class="fu">snpgdsLDpruning</span>(geno, <span class="at">ld.threshold=</span><span class="fl">0.1</span>, <span class="at">autosome.only=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## SNP pruning based on LD:
## Excluding 1,086,533 SNPs (monomorphic: TRUE, MAF: 0.005, missing rate: 0.05)
##     # of samples: 20
##     # of SNPs: 2,821,477
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.1
##     method: composite
## Chrom 1: |====================|====================|
##     2.11%, 6,895 / 327,390 (Wed Mar 19 15:59:15 2025)
## Chrom 2: |====================|====================|
##     2.05%, 7,419 / 361,795 (Wed Mar 19 15:59:15 2025)
## Chrom 3: |====================|====================|
##     2.02%, 6,267 / 310,679 (Wed Mar 19 15:59:16 2025)
## Chrom 4: |====================|====================|
##     2.12%, 7,519 / 355,492 (Wed Mar 19 15:59:16 2025)
## Chrom 5: |====================|====================|
##     1.86%, 3,536 / 190,039 (Wed Mar 19 15:59:17 2025)
## Chrom 6: |====================|====================|
##     2.26%, 5,411 / 239,224 (Wed Mar 19 15:59:17 2025)
## Chrom 7: |====================|====================|
##     2.28%, 3,840 / 168,421 (Wed Mar 19 15:59:18 2025)
## Chrom 8: |====================|====================|
##     2.36%, 2,821 / 119,709 (Wed Mar 19 15:59:18 2025)
## Chrom 9: |====================|====================|
##     2.96%, 4,999 / 169,064 (Wed Mar 19 15:59:18 2025)
## Chrom 10: |====================|====================|
##     2.05%, 4,117 / 200,553 (Wed Mar 19 15:59:19 2025)
## Chrom 11: |====================|====================|
##     2.06%, 5,240 / 254,638 (Wed Mar 19 15:59:19 2025)
## Chrom 12: |====================|====================|
##     2.33%, 7,098 / 305,075 (Wed Mar 19 15:59:20 2025)
## Chrom 13: |====================|====================|
##     2.11%, 5,825 / 276,329 (Wed Mar 19 15:59:20 2025)
## Chrom 14: |====================|====================|
##     2.21%, 5,837 / 264,542 (Wed Mar 19 15:59:21 2025)
## Chrom 16: |====================|====================|
##     1.83%, 6,670 / 365,060 (Wed Mar 19 15:59:21 2025)
## 83,494 markers are selected in total.</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="pca---high-coverage.html#cb34-1" tabindex="-1"></a>snpset.id2 <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">unname</span>(snpset2))</span>
<span id="cb34-2"><a href="pca---high-coverage.html#cb34-2" tabindex="-1"></a></span>
<span id="cb34-3"><a href="pca---high-coverage.html#cb34-3" tabindex="-1"></a>snpset3 <span class="ot">&lt;-</span> <span class="fu">snpgdsLDpruning</span>(geno, <span class="at">ld.threshold=</span><span class="fl">0.5</span>, <span class="at">autosome.only=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## SNP pruning based on LD:
## Excluding 1,086,533 SNPs (monomorphic: TRUE, MAF: 0.005, missing rate: 0.05)
##     # of samples: 20
##     # of SNPs: 2,821,477
##     using 1 thread
##     sliding window: 500,000 basepairs, Inf SNPs
##     |LD| threshold: 0.5
##     method: composite
## Chrom 1: |====================|====================|
##     4.11%, 13,448 / 327,390 (Wed Mar 19 15:59:27 2025)
## Chrom 2: |====================|====================|
##     4.09%, 14,799 / 361,795 (Wed Mar 19 15:59:28 2025)
## Chrom 3: |====================|====================|
##     4.07%, 12,641 / 310,679 (Wed Mar 19 15:59:29 2025)
## Chrom 4: |====================|====================|
##     4.18%, 14,845 / 355,492 (Wed Mar 19 15:59:30 2025)
## Chrom 5: |====================|====================|
##     3.94%, 7,494 / 190,039 (Wed Mar 19 15:59:31 2025)
## Chrom 6: |====================|====================|
##     4.45%, 10,653 / 239,224 (Wed Mar 19 15:59:32 2025)
## Chrom 7: |====================|====================|
##     4.51%, 7,601 / 168,421 (Wed Mar 19 15:59:32 2025)
## Chrom 8: |====================|====================|
##     4.50%, 5,391 / 119,709 (Wed Mar 19 15:59:33 2025)
## Chrom 9: |====================|====================|
##     5.04%, 8,524 / 169,064 (Wed Mar 19 15:59:34 2025)
## Chrom 10: |====================|====================|
##     4.16%, 8,339 / 200,553 (Wed Mar 19 15:59:34 2025)
## Chrom 11: |====================|====================|
##     4.16%, 10,582 / 254,638 (Wed Mar 19 15:59:35 2025)
## Chrom 12: |====================|====================|
##     4.42%, 13,487 / 305,075 (Wed Mar 19 15:59:36 2025)
## Chrom 13: |====================|====================|
##     4.24%, 11,711 / 276,329 (Wed Mar 19 15:59:37 2025)
## Chrom 14: |====================|====================|
##     4.23%, 11,181 / 264,542 (Wed Mar 19 15:59:38 2025)
## Chrom 16: |====================|====================|
##     3.81%, 13,903 / 365,060 (Wed Mar 19 15:59:39 2025)
## 164,599 markers are selected in total.</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="pca---high-coverage.html#cb36-1" tabindex="-1"></a>snpset.id3 <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">unname</span>(snpset3))</span></code></pre></div>
</div>
<div id="pca" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> PCA<a href="pca---high-coverage.html#pca" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="pca-original" class="section level3 hasAnchor" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> PCA Original<a href="pca---high-coverage.html#pca-original" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="run-pca" class="section level4 hasAnchor" number="3.5.1.1">
<h4><span class="header-section-number">3.5.1.1</span> Run PCA<a href="pca---high-coverage.html#run-pca" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="pca---high-coverage.html#cb37-1" tabindex="-1"></a>pca <span class="ot">&lt;-</span> </span>
<span id="cb37-2"><a href="pca---high-coverage.html#cb37-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb37-3"><a href="pca---high-coverage.html#cb37-3" tabindex="-1"></a>            <span class="at">snp.id=</span>snpset.id, </span>
<span id="cb37-4"><a href="pca---high-coverage.html#cb37-4" tabindex="-1"></a>            <span class="at">num.thread=</span><span class="dv">2</span>, <span class="at">autosome.only=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 81,297
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 2673881
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 15:59:40 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 15:59:40 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 15:59:40 2025    Done.</code></pre>
</div>
<div id="calculate-pc-percent-variation" class="section level4 hasAnchor" number="3.5.1.2">
<h4><span class="header-section-number">3.5.1.2</span> Calculate PC Percent Variation<a href="pca---high-coverage.html#calculate-pc-percent-variation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="pca---high-coverage.html#cb39-1" tabindex="-1"></a>pc.percent <span class="ot">&lt;-</span> </span>
<span id="cb39-2"><a href="pca---high-coverage.html#cb39-2" tabindex="-1"></a>  <span class="fu">round</span>(pca<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb39-3"><a href="pca---high-coverage.html#cb39-3" tabindex="-1"></a><span class="fu">head</span>(pc.percent)</span></code></pre></div>
<pre><code>## [1] 6.11 5.53 5.49 5.43 5.40 5.37</code></pre>
</div>
<div id="configure-plot-data" class="section level4 hasAnchor" number="3.5.1.3">
<h4><span class="header-section-number">3.5.1.3</span> Configure plot data<a href="pca---high-coverage.html#configure-plot-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="pca---high-coverage.html#cb41-1" tabindex="-1"></a>pca_data <span class="ot">&lt;-</span> </span>
<span id="cb41-2"><a href="pca---high-coverage.html#cb41-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca<span class="sc">$</span>sample.id,</span>
<span id="cb41-3"><a href="pca---high-coverage.html#cb41-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],   </span>
<span id="cb41-4"><a href="pca---high-coverage.html#cb41-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb41-5"><a href="pca---high-coverage.html#cb41-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb41-6"><a href="pca---high-coverage.html#cb41-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb41-7"><a href="pca---high-coverage.html#cb41-7" tabindex="-1"></a></span>
<span id="cb41-8"><a href="pca---high-coverage.html#cb41-8" tabindex="-1"></a>labelx <span class="ot">&lt;-</span> </span>
<span id="cb41-9"><a href="pca---high-coverage.html#cb41-9" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb41-10"><a href="pca---high-coverage.html#cb41-10" tabindex="-1"></a>labely <span class="ot">&lt;-</span> </span>
<span id="cb41-11"><a href="pca---high-coverage.html#cb41-11" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span></code></pre></div>
</div>
<div id="plot" class="section level4 hasAnchor" number="3.5.1.4">
<h4><span class="header-section-number">3.5.1.4</span> Plot<a href="pca---high-coverage.html#plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Seemingly no structure, but maybe outliers (5, 51, 58)</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="pca---high-coverage.html#cb42-1" tabindex="-1"></a>plot <span class="ot">&lt;-</span></span>
<span id="cb42-2"><a href="pca---high-coverage.html#cb42-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb42-3"><a href="pca---high-coverage.html#cb42-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>, <span class="at">colour=</span><span class="st">&quot;slategray&quot;</span>) <span class="sc">+</span></span>
<span id="cb42-4"><a href="pca---high-coverage.html#cb42-4" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb42-5"><a href="pca---high-coverage.html#cb42-5" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb42-6"><a href="pca---high-coverage.html#cb42-6" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>pca_data<span class="sc">$</span>sample.no) <span class="sc">+</span></span>
<span id="cb42-7"><a href="pca---high-coverage.html#cb42-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb42-8"><a href="pca---high-coverage.html#cb42-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb42-9"><a href="pca---high-coverage.html#cb42-9" tabindex="-1"></a>plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
</div>
<div id="plot-pca-factors" class="section level4 hasAnchor" number="3.5.1.5">
<h4><span class="header-section-number">3.5.1.5</span> Plot PCA ~ Factors<a href="pca---high-coverage.html#plot-pca-factors" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="pca---high-coverage.html#cb43-1" tabindex="-1"></a>classification <span class="ot">&lt;-</span> </span>
<span id="cb43-2"><a href="pca---high-coverage.html#cb43-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span>classification)) <span class="sc">+</span></span>
<span id="cb43-3"><a href="pca---high-coverage.html#cb43-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb43-4"><a href="pca---high-coverage.html#cb43-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>)) <span class="sc">+</span></span>
<span id="cb43-5"><a href="pca---high-coverage.html#cb43-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb43-6"><a href="pca---high-coverage.html#cb43-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb43-7"><a href="pca---high-coverage.html#cb43-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb43-8"><a href="pca---high-coverage.html#cb43-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb43-9"><a href="pca---high-coverage.html#cb43-9" tabindex="-1"></a>classification</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="pca---high-coverage.html#cb44-1" tabindex="-1"></a>pdv <span class="ot">&lt;-</span> </span>
<span id="cb44-2"><a href="pca---high-coverage.html#cb44-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span>pdv)) <span class="sc">+</span></span>
<span id="cb44-3"><a href="pca---high-coverage.html#cb44-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb44-4"><a href="pca---high-coverage.html#cb44-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>, <span class="st">&quot;coral&quot;</span>)) <span class="sc">+</span></span>
<span id="cb44-5"><a href="pca---high-coverage.html#cb44-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb44-6"><a href="pca---high-coverage.html#cb44-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb44-7"><a href="pca---high-coverage.html#cb44-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb44-8"><a href="pca---high-coverage.html#cb44-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb44-9"><a href="pca---high-coverage.html#cb44-9" tabindex="-1"></a>pdv</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-18-2.png" width="672" /></p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="pca---high-coverage.html#cb45-1" tabindex="-1"></a>sex <span class="ot">&lt;-</span> </span>
<span id="cb45-2"><a href="pca---high-coverage.html#cb45-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span>sex)) <span class="sc">+</span></span>
<span id="cb45-3"><a href="pca---high-coverage.html#cb45-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb45-4"><a href="pca---high-coverage.html#cb45-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>)) <span class="sc">+</span></span>
<span id="cb45-5"><a href="pca---high-coverage.html#cb45-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb45-6"><a href="pca---high-coverage.html#cb45-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb45-7"><a href="pca---high-coverage.html#cb45-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb45-8"><a href="pca---high-coverage.html#cb45-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb45-9"><a href="pca---high-coverage.html#cb45-9" tabindex="-1"></a>sex</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-18-3.png" width="672" /></p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="pca---high-coverage.html#cb46-1" tabindex="-1"></a>age <span class="ot">&lt;-</span></span>
<span id="cb46-2"><a href="pca---high-coverage.html#cb46-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span>age.class)) <span class="sc">+</span></span>
<span id="cb46-3"><a href="pca---high-coverage.html#cb46-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb46-4"><a href="pca---high-coverage.html#cb46-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>, <span class="st">&quot;coral&quot;</span>, <span class="st">&quot;magenta&quot;</span>)) <span class="sc">+</span></span>
<span id="cb46-5"><a href="pca---high-coverage.html#cb46-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb46-6"><a href="pca---high-coverage.html#cb46-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb46-7"><a href="pca---high-coverage.html#cb46-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb46-8"><a href="pca---high-coverage.html#cb46-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb46-9"><a href="pca---high-coverage.html#cb46-9" tabindex="-1"></a>age</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-18-4.png" width="672" /></p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="pca---high-coverage.html#cb47-1" tabindex="-1"></a>condition <span class="ot">&lt;-</span> </span>
<span id="cb47-2"><a href="pca---high-coverage.html#cb47-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span>condition)) <span class="sc">+</span></span>
<span id="cb47-3"><a href="pca---high-coverage.html#cb47-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb47-4"><a href="pca---high-coverage.html#cb47-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>, <span class="st">&quot;coral&quot;</span>)) <span class="sc">+</span></span>
<span id="cb47-5"><a href="pca---high-coverage.html#cb47-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb47-6"><a href="pca---high-coverage.html#cb47-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb47-7"><a href="pca---high-coverage.html#cb47-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb47-8"><a href="pca---high-coverage.html#cb47-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb47-9"><a href="pca---high-coverage.html#cb47-9" tabindex="-1"></a>condition</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-18-5.png" width="672" /></p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="pca---high-coverage.html#cb48-1" tabindex="-1"></a>seqrun <span class="ot">&lt;-</span> </span>
<span id="cb48-2"><a href="pca---high-coverage.html#cb48-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span><span class="fu">factor</span>(seq.run))) <span class="sc">+</span></span>
<span id="cb48-3"><a href="pca---high-coverage.html#cb48-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb48-4"><a href="pca---high-coverage.html#cb48-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>)) <span class="sc">+</span></span>
<span id="cb48-5"><a href="pca---high-coverage.html#cb48-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb48-6"><a href="pca---high-coverage.html#cb48-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb48-7"><a href="pca---high-coverage.html#cb48-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb48-8"><a href="pca---high-coverage.html#cb48-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb48-9"><a href="pca---high-coverage.html#cb48-9" tabindex="-1"></a>seqrun</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-18-6.png" width="672" /></p>
</div>
</div>
<div id="pca-wo-outliers" class="section level3 hasAnchor" number="3.5.2">
<h3><span class="header-section-number">3.5.2</span> PCA w/o Outliers<a href="pca---high-coverage.html#pca-wo-outliers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After two iterations. First removed 51, 58, then removed 5, 72.</p>
<div id="possible-outliers" class="section level4 hasAnchor" number="3.5.2.1">
<h4><span class="header-section-number">3.5.2.1</span> Possible outliers<a href="pca---high-coverage.html#possible-outliers" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="pca---high-coverage.html#cb49-1" tabindex="-1"></a>outlier_meta <span class="ot">&lt;-</span> </span>
<span id="cb49-2"><a href="pca---high-coverage.html#cb49-2" tabindex="-1"></a>  meta <span class="sc">%&gt;%</span> </span>
<span id="cb49-3"><a href="pca---high-coverage.html#cb49-3" tabindex="-1"></a>  <span class="fu">filter</span>(sample.no<span class="sc">==</span><span class="st">&quot;51&quot;</span> <span class="sc">|</span></span>
<span id="cb49-4"><a href="pca---high-coverage.html#cb49-4" tabindex="-1"></a>         sample.no<span class="sc">==</span><span class="st">&quot;58&quot;</span> <span class="sc">|</span></span>
<span id="cb49-5"><a href="pca---high-coverage.html#cb49-5" tabindex="-1"></a>         sample.no<span class="sc">==</span><span class="st">&quot;5&quot;</span> <span class="sc">|</span></span>
<span id="cb49-6"><a href="pca---high-coverage.html#cb49-6" tabindex="-1"></a>         sample.no<span class="sc">==</span><span class="st">&quot;72&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-7"><a href="pca---high-coverage.html#cb49-7" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>seq.id <span class="sc">&amp;</span> <span class="sc">!</span>coverage) <span class="sc">%&gt;%</span> </span>
<span id="cb49-8"><a href="pca---high-coverage.html#cb49-8" tabindex="-1"></a>  <span class="fu">setNames</span>(., <span class="at">nm=</span><span class="fu">c</span>(<span class="st">&quot;#&quot;</span>, <span class="st">&quot;sample&quot;</span>, <span class="st">&quot;run&quot;</span>, <span class="st">&quot;category&quot;</span>, <span class="st">&quot;strand date&quot;</span>, <span class="st">&quot;lat&quot;</span>, <span class="st">&quot;long&quot;</span>, </span>
<span id="cb49-9"><a href="pca---high-coverage.html#cb49-9" tabindex="-1"></a>             <span class="st">&quot;condition&quot;</span>, <span class="st">&quot;sex&quot;</span>, <span class="st">&quot;age&quot;</span>, <span class="st">&quot;pdv&quot;</span>, <span class="st">&quot;totaldna&quot;</span>, <span class="st">&quot;seqdepth&quot;</span>))</span>
<span id="cb49-10"><a href="pca---high-coverage.html#cb49-10" tabindex="-1"></a></span>
<span id="cb49-11"><a href="pca---high-coverage.html#cb49-11" tabindex="-1"></a>outlier_meta <span class="sc">%&gt;%</span> </span>
<span id="cb49-12"><a href="pca---high-coverage.html#cb49-12" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb49-13"><a href="pca---high-coverage.html#cb49-13" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">&quot;basic&quot;</span>, <span class="at">font_size=</span><span class="dv">12</span>,</span>
<span id="cb49-14"><a href="pca---high-coverage.html#cb49-14" tabindex="-1"></a>                <span class="at">position=</span><span class="st">&quot;center&quot;</span>)</span></code></pre></div>
<table class="table" style="font-size: 12px; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
#
</th>
<th style="text-align:left;">
sample
</th>
<th style="text-align:right;">
run
</th>
<th style="text-align:left;">
category
</th>
<th style="text-align:left;">
strand date
</th>
<th style="text-align:right;">
lat
</th>
<th style="text-align:right;">
long
</th>
<th style="text-align:left;">
condition
</th>
<th style="text-align:left;">
sex
</th>
<th style="text-align:left;">
age
</th>
<th style="text-align:left;">
pdv
</th>
<th style="text-align:right;">
totaldna
</th>
<th style="text-align:right;">
seqdepth
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
MME-18-568Pv
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
Non-survivor
</td>
<td style="text-align:left;">
8/31/2018
</td>
<td style="text-align:right;">
43.16199
</td>
<td style="text-align:right;">
-70.61980
</td>
<td style="text-align:left;">
Alive
</td>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
Yearling
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
1148
</td>
<td style="text-align:right;">
58.7219
</td>
</tr>
<tr>
<td style="text-align:right;">
51
</td>
<td style="text-align:left;">
MME-18-120Pv
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Non-survivor
</td>
<td style="text-align:left;">
7/3/2018
</td>
<td style="text-align:right;">
43.66643
</td>
<td style="text-align:right;">
-70.23860
</td>
<td style="text-align:left;">
Alive
</td>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
Yearling
</td>
<td style="text-align:left;">
N/A
</td>
<td style="text-align:right;">
1696
</td>
<td style="text-align:right;">
58.1716
</td>
</tr>
<tr>
<td style="text-align:right;">
58
</td>
<td style="text-align:left;">
IFAW20-069Pv
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Survivor
</td>
<td style="text-align:left;">
5/7/2020
</td>
<td style="text-align:right;">
41.67004
</td>
<td style="text-align:right;">
-69.94767
</td>
<td style="text-align:left;">
Mod Decomp
</td>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
Adult
</td>
<td style="text-align:left;">
Negative
</td>
<td style="text-align:right;">
4200
</td>
<td style="text-align:right;">
58.9652
</td>
</tr>
<tr>
<td style="text-align:right;">
72
</td>
<td style="text-align:left;">
MME-19-149Pv
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
Survivor
</td>
<td style="text-align:left;">
5/21/2019
</td>
<td style="text-align:right;">
43.83349
</td>
<td style="text-align:right;">
-69.64360
</td>
<td style="text-align:left;">
Alive
</td>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
Pup/Calf
</td>
<td style="text-align:left;">
Positive
</td>
<td style="text-align:right;">
4320
</td>
<td style="text-align:right;">
58.9553
</td>
</tr>
</tbody>
</table>
</div>
<div id="run-pca-1" class="section level4 hasAnchor" number="3.5.2.2">
<h4><span class="header-section-number">3.5.2.2</span> Run PCA<a href="pca---high-coverage.html#run-pca-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="pca---high-coverage.html#cb50-1" tabindex="-1"></a>samples_outlier <span class="ot">&lt;-</span></span>
<span id="cb50-2"><a href="pca---high-coverage.html#cb50-2" tabindex="-1"></a>  meta <span class="sc">%&gt;%</span> </span>
<span id="cb50-3"><a href="pca---high-coverage.html#cb50-3" tabindex="-1"></a>  <span class="fu">filter</span>(coverage<span class="sc">==</span><span class="st">&quot;high&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-4"><a href="pca---high-coverage.html#cb50-4" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>sample.no<span class="sc">==</span><span class="st">&quot;51&quot;</span> <span class="sc">&amp;</span> </span>
<span id="cb50-5"><a href="pca---high-coverage.html#cb50-5" tabindex="-1"></a>         <span class="sc">!</span>sample.no<span class="sc">==</span><span class="st">&quot;58&quot;</span> <span class="sc">&amp;</span></span>
<span id="cb50-6"><a href="pca---high-coverage.html#cb50-6" tabindex="-1"></a>         <span class="sc">!</span>sample.no<span class="sc">==</span><span class="st">&quot;5&quot;</span> <span class="sc">&amp;</span></span>
<span id="cb50-7"><a href="pca---high-coverage.html#cb50-7" tabindex="-1"></a>         <span class="sc">!</span>sample.no<span class="sc">==</span><span class="st">&quot;72&quot;</span>)</span>
<span id="cb50-8"><a href="pca---high-coverage.html#cb50-8" tabindex="-1"></a></span>
<span id="cb50-9"><a href="pca---high-coverage.html#cb50-9" tabindex="-1"></a>pca_outlier <span class="ot">&lt;-</span> </span>
<span id="cb50-10"><a href="pca---high-coverage.html#cb50-10" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb50-11"><a href="pca---high-coverage.html#cb50-11" tabindex="-1"></a>            <span class="at">snp.id=</span>snpset.id, </span>
<span id="cb50-12"><a href="pca---high-coverage.html#cb50-12" tabindex="-1"></a>            <span class="at">sample.id=</span>samples_outlier<span class="sc">$</span>seq.id,</span>
<span id="cb50-13"><a href="pca---high-coverage.html#cb50-13" tabindex="-1"></a>            <span class="at">num.thread=</span><span class="dv">2</span>,</span>
<span id="cb50-14"><a href="pca---high-coverage.html#cb50-14" tabindex="-1"></a>            <span class="at">autosome.only=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Excluding 11,473 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 16
##     # of SNPs: 69,824
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 1782304
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 15:59:46 2025    (internal increment: 81664)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 15:59:46 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 15:59:46 2025    Done.</code></pre>
</div>
<div id="calculate-pc-percent-variation-1" class="section level4 hasAnchor" number="3.5.2.3">
<h4><span class="header-section-number">3.5.2.3</span> Calculate PC Percent Variation<a href="pca---high-coverage.html#calculate-pc-percent-variation-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="pca---high-coverage.html#cb52-1" tabindex="-1"></a>pc.percent_outlier <span class="ot">&lt;-</span> </span>
<span id="cb52-2"><a href="pca---high-coverage.html#cb52-2" tabindex="-1"></a>  <span class="fu">round</span>(pca_outlier<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb52-3"><a href="pca---high-coverage.html#cb52-3" tabindex="-1"></a><span class="fu">head</span>(pc.percent_outlier)</span></code></pre></div>
<pre><code>## [1] 6.97 6.95 6.88 6.80 6.75 6.73</code></pre>
</div>
<div id="configure-plot-data-1" class="section level4 hasAnchor" number="3.5.2.4">
<h4><span class="header-section-number">3.5.2.4</span> Configure plot data<a href="pca---high-coverage.html#configure-plot-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="pca---high-coverage.html#cb54-1" tabindex="-1"></a>pca_data_outlier <span class="ot">&lt;-</span> </span>
<span id="cb54-2"><a href="pca---high-coverage.html#cb54-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_outlier<span class="sc">$</span>sample.id,</span>
<span id="cb54-3"><a href="pca---high-coverage.html#cb54-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_outlier<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],   </span>
<span id="cb54-4"><a href="pca---high-coverage.html#cb54-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_outlier<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb54-5"><a href="pca---high-coverage.html#cb54-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-6"><a href="pca---high-coverage.html#cb54-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb54-7"><a href="pca---high-coverage.html#cb54-7" tabindex="-1"></a></span>
<span id="cb54-8"><a href="pca---high-coverage.html#cb54-8" tabindex="-1"></a>labelx <span class="ot">&lt;-</span> </span>
<span id="cb54-9"><a href="pca---high-coverage.html#cb54-9" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_outlier[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb54-10"><a href="pca---high-coverage.html#cb54-10" tabindex="-1"></a>labely <span class="ot">&lt;-</span> </span>
<span id="cb54-11"><a href="pca---high-coverage.html#cb54-11" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_outlier[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span></code></pre></div>
</div>
<div id="plot-1" class="section level4 hasAnchor" number="3.5.2.5">
<h4><span class="header-section-number">3.5.2.5</span> Plot<a href="pca---high-coverage.html#plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Seemingly no structure, but maybe outliers (5, 51, 58)</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="pca---high-coverage.html#cb55-1" tabindex="-1"></a>plot_outlier <span class="ot">&lt;-</span></span>
<span id="cb55-2"><a href="pca---high-coverage.html#cb55-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data_outlier, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb55-3"><a href="pca---high-coverage.html#cb55-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>, <span class="at">colour=</span><span class="st">&quot;slategray&quot;</span>) <span class="sc">+</span></span>
<span id="cb55-4"><a href="pca---high-coverage.html#cb55-4" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb55-5"><a href="pca---high-coverage.html#cb55-5" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb55-6"><a href="pca---high-coverage.html#cb55-6" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>pca_data_outlier<span class="sc">$</span>sample.no) <span class="sc">+</span></span>
<span id="cb55-7"><a href="pca---high-coverage.html#cb55-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb55-8"><a href="pca---high-coverage.html#cb55-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb55-9"><a href="pca---high-coverage.html#cb55-9" tabindex="-1"></a>plot_outlier</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
</div>
<div id="plot-pca-factors-1" class="section level4 hasAnchor" number="3.5.2.6">
<h4><span class="header-section-number">3.5.2.6</span> Plot PCA ~ Factors<a href="pca---high-coverage.html#plot-pca-factors-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="pca---high-coverage.html#cb56-1" tabindex="-1"></a>classification_outlier <span class="ot">&lt;-</span> </span>
<span id="cb56-2"><a href="pca---high-coverage.html#cb56-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data_outlier, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span>classification)) <span class="sc">+</span></span>
<span id="cb56-3"><a href="pca---high-coverage.html#cb56-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb56-4"><a href="pca---high-coverage.html#cb56-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>)) <span class="sc">+</span></span>
<span id="cb56-5"><a href="pca---high-coverage.html#cb56-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb56-6"><a href="pca---high-coverage.html#cb56-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb56-7"><a href="pca---high-coverage.html#cb56-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb56-8"><a href="pca---high-coverage.html#cb56-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb56-9"><a href="pca---high-coverage.html#cb56-9" tabindex="-1"></a>classification_outlier</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="pca---high-coverage.html#cb57-1" tabindex="-1"></a>pdv_outlier <span class="ot">&lt;-</span> </span>
<span id="cb57-2"><a href="pca---high-coverage.html#cb57-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data_outlier, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span>pdv)) <span class="sc">+</span></span>
<span id="cb57-3"><a href="pca---high-coverage.html#cb57-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb57-4"><a href="pca---high-coverage.html#cb57-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>, <span class="st">&quot;coral&quot;</span>)) <span class="sc">+</span></span>
<span id="cb57-5"><a href="pca---high-coverage.html#cb57-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb57-6"><a href="pca---high-coverage.html#cb57-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb57-7"><a href="pca---high-coverage.html#cb57-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb57-8"><a href="pca---high-coverage.html#cb57-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb57-9"><a href="pca---high-coverage.html#cb57-9" tabindex="-1"></a>pdv_outlier</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-24-2.png" width="672" /></p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="pca---high-coverage.html#cb58-1" tabindex="-1"></a>sex_outlier <span class="ot">&lt;-</span> </span>
<span id="cb58-2"><a href="pca---high-coverage.html#cb58-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data_outlier, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span>sex)) <span class="sc">+</span></span>
<span id="cb58-3"><a href="pca---high-coverage.html#cb58-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb58-4"><a href="pca---high-coverage.html#cb58-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>)) <span class="sc">+</span></span>
<span id="cb58-5"><a href="pca---high-coverage.html#cb58-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb58-6"><a href="pca---high-coverage.html#cb58-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb58-7"><a href="pca---high-coverage.html#cb58-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb58-8"><a href="pca---high-coverage.html#cb58-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb58-9"><a href="pca---high-coverage.html#cb58-9" tabindex="-1"></a>sex_outlier</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-24-3.png" width="672" /></p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="pca---high-coverage.html#cb59-1" tabindex="-1"></a>age_outlier <span class="ot">&lt;-</span></span>
<span id="cb59-2"><a href="pca---high-coverage.html#cb59-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data_outlier, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span>age.class)) <span class="sc">+</span></span>
<span id="cb59-3"><a href="pca---high-coverage.html#cb59-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb59-4"><a href="pca---high-coverage.html#cb59-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>, <span class="st">&quot;coral&quot;</span>, <span class="st">&quot;magenta&quot;</span>)) <span class="sc">+</span></span>
<span id="cb59-5"><a href="pca---high-coverage.html#cb59-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb59-6"><a href="pca---high-coverage.html#cb59-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb59-7"><a href="pca---high-coverage.html#cb59-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb59-8"><a href="pca---high-coverage.html#cb59-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb59-9"><a href="pca---high-coverage.html#cb59-9" tabindex="-1"></a>age_outlier</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-24-4.png" width="672" /></p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="pca---high-coverage.html#cb60-1" tabindex="-1"></a>condition_outlier <span class="ot">&lt;-</span> </span>
<span id="cb60-2"><a href="pca---high-coverage.html#cb60-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data_outlier, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span>condition)) <span class="sc">+</span></span>
<span id="cb60-3"><a href="pca---high-coverage.html#cb60-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb60-4"><a href="pca---high-coverage.html#cb60-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>, <span class="st">&quot;coral&quot;</span>)) <span class="sc">+</span></span>
<span id="cb60-5"><a href="pca---high-coverage.html#cb60-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb60-6"><a href="pca---high-coverage.html#cb60-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb60-7"><a href="pca---high-coverage.html#cb60-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb60-8"><a href="pca---high-coverage.html#cb60-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb60-9"><a href="pca---high-coverage.html#cb60-9" tabindex="-1"></a>condition_outlier</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-24-5.png" width="672" /></p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="pca---high-coverage.html#cb61-1" tabindex="-1"></a>seqrun_outlier <span class="ot">&lt;-</span> </span>
<span id="cb61-2"><a href="pca---high-coverage.html#cb61-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(pca_data_outlier, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2, <span class="at">color=</span><span class="fu">factor</span>(seq.run))) <span class="sc">+</span></span>
<span id="cb61-3"><a href="pca---high-coverage.html#cb61-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb61-4"><a href="pca---high-coverage.html#cb61-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray&quot;</span>, <span class="st">&quot;#58baba&quot;</span>)) <span class="sc">+</span></span>
<span id="cb61-5"><a href="pca---high-coverage.html#cb61-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span>labelx,</span>
<span id="cb61-6"><a href="pca---high-coverage.html#cb61-6" tabindex="-1"></a>       <span class="at">y=</span>labely) <span class="sc">+</span></span>
<span id="cb61-7"><a href="pca---high-coverage.html#cb61-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb61-8"><a href="pca---high-coverage.html#cb61-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb61-9"><a href="pca---high-coverage.html#cb61-9" tabindex="-1"></a>seqrun_outlier</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-24-6.png" width="672" /></p>
</div>
<div id="combine" class="section level4 hasAnchor" number="3.5.2.7">
<h4><span class="header-section-number">3.5.2.7</span> Combine<a href="pca---high-coverage.html#combine" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="pca---high-coverage.html#cb62-1" tabindex="-1"></a>plot_outlier</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="pca---high-coverage.html#cb63-1" tabindex="-1"></a><span class="fu">grid.arrange</span>(classification_outlier, pdv_outlier, sex_outlier, </span>
<span id="cb63-2"><a href="pca---high-coverage.html#cb63-2" tabindex="-1"></a>             age_outlier, condition_outlier, seqrun_outlier, <span class="at">ncol=</span><span class="dv">3</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-25-2.png" width="672" /></p>
</div>
</div>
<div id="pca-by-chromosome" class="section level3 hasAnchor" number="3.5.3">
<h3><span class="header-section-number">3.5.3</span> PCA by Chromosome<a href="pca---high-coverage.html#pca-by-chromosome" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="chromosome-1" class="section level4 hasAnchor" number="3.5.3.1">
<h4><span class="header-section-number">3.5.3.1</span> Chromosome 1<a href="pca---high-coverage.html#chromosome-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="pca---high-coverage.html#cb64-1" tabindex="-1"></a>pca_chrom1 <span class="ot">&lt;-</span> </span>
<span id="cb64-2"><a href="pca---high-coverage.html#cb64-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb64-3"><a href="pca---high-coverage.html#cb64-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb64-4"><a href="pca---high-coverage.html#cb64-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb64-5"><a href="pca---high-coverage.html#cb64-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;1&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 6,682 SNPs according to chromosome 1
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 6,682
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 224580
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 15:59:56 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 15:59:56 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 15:59:56 2025    Done.</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="pca---high-coverage.html#cb66-1" tabindex="-1"></a>chrom1 <span class="ot">&lt;-</span> </span>
<span id="cb66-2"><a href="pca---high-coverage.html#cb66-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom1<span class="sc">$</span>sample.id,</span>
<span id="cb66-3"><a href="pca---high-coverage.html#cb66-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom1<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb66-4"><a href="pca---high-coverage.html#cb66-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom1<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb66-5"><a href="pca---high-coverage.html#cb66-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb66-6"><a href="pca---high-coverage.html#cb66-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb66-7"><a href="pca---high-coverage.html#cb66-7" tabindex="-1"></a></span>
<span id="cb66-8"><a href="pca---high-coverage.html#cb66-8" tabindex="-1"></a>pc.percent_chrom1 <span class="ot">&lt;-</span> </span>
<span id="cb66-9"><a href="pca---high-coverage.html#cb66-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom1<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb66-10"><a href="pca---high-coverage.html#cb66-10" tabindex="-1"></a></span>
<span id="cb66-11"><a href="pca---high-coverage.html#cb66-11" tabindex="-1"></a>labelx_chrom1 <span class="ot">&lt;-</span> </span>
<span id="cb66-12"><a href="pca---high-coverage.html#cb66-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom1[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb66-13"><a href="pca---high-coverage.html#cb66-13" tabindex="-1"></a>labely_chrom1 <span class="ot">&lt;-</span> </span>
<span id="cb66-14"><a href="pca---high-coverage.html#cb66-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom1[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb66-15"><a href="pca---high-coverage.html#cb66-15" tabindex="-1"></a></span>
<span id="cb66-16"><a href="pca---high-coverage.html#cb66-16" tabindex="-1"></a>chrom1_plot <span class="ot">&lt;-</span></span>
<span id="cb66-17"><a href="pca---high-coverage.html#cb66-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom1, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb66-18"><a href="pca---high-coverage.html#cb66-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb66-19"><a href="pca---high-coverage.html#cb66-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 1&quot;</span>,</span>
<span id="cb66-20"><a href="pca---high-coverage.html#cb66-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom1,</span>
<span id="cb66-21"><a href="pca---high-coverage.html#cb66-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom1) <span class="sc">+</span></span>
<span id="cb66-22"><a href="pca---high-coverage.html#cb66-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom1<span class="sc">$</span>sample.no,   </span>
<span id="cb66-23"><a href="pca---high-coverage.html#cb66-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb66-24"><a href="pca---high-coverage.html#cb66-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb66-25"><a href="pca---high-coverage.html#cb66-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb66-26"><a href="pca---high-coverage.html#cb66-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-2" class="section level4 hasAnchor" number="3.5.3.2">
<h4><span class="header-section-number">3.5.3.2</span> Chromosome 2<a href="pca---high-coverage.html#chromosome-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="pca---high-coverage.html#cb67-1" tabindex="-1"></a>pca_chrom2 <span class="ot">&lt;-</span> </span>
<span id="cb67-2"><a href="pca---high-coverage.html#cb67-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb67-3"><a href="pca---high-coverage.html#cb67-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb67-4"><a href="pca---high-coverage.html#cb67-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb67-5"><a href="pca---high-coverage.html#cb67-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;2&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 7,169 SNPs according to chromosome 2
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 7,169
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 241119
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 15:59:57 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 15:59:57 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 15:59:57 2025    Done.</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="pca---high-coverage.html#cb69-1" tabindex="-1"></a>chrom2 <span class="ot">&lt;-</span> </span>
<span id="cb69-2"><a href="pca---high-coverage.html#cb69-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom2<span class="sc">$</span>sample.id,</span>
<span id="cb69-3"><a href="pca---high-coverage.html#cb69-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom2<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb69-4"><a href="pca---high-coverage.html#cb69-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom2<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb69-5"><a href="pca---high-coverage.html#cb69-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb69-6"><a href="pca---high-coverage.html#cb69-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb69-7"><a href="pca---high-coverage.html#cb69-7" tabindex="-1"></a></span>
<span id="cb69-8"><a href="pca---high-coverage.html#cb69-8" tabindex="-1"></a>pc.percent_chrom2 <span class="ot">&lt;-</span> </span>
<span id="cb69-9"><a href="pca---high-coverage.html#cb69-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom2<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb69-10"><a href="pca---high-coverage.html#cb69-10" tabindex="-1"></a></span>
<span id="cb69-11"><a href="pca---high-coverage.html#cb69-11" tabindex="-1"></a>labelx_chrom2 <span class="ot">&lt;-</span> </span>
<span id="cb69-12"><a href="pca---high-coverage.html#cb69-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom2[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb69-13"><a href="pca---high-coverage.html#cb69-13" tabindex="-1"></a>labely_chrom2 <span class="ot">&lt;-</span> </span>
<span id="cb69-14"><a href="pca---high-coverage.html#cb69-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom2[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb69-15"><a href="pca---high-coverage.html#cb69-15" tabindex="-1"></a></span>
<span id="cb69-16"><a href="pca---high-coverage.html#cb69-16" tabindex="-1"></a>chrom2_plot <span class="ot">&lt;-</span></span>
<span id="cb69-17"><a href="pca---high-coverage.html#cb69-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom2, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb69-18"><a href="pca---high-coverage.html#cb69-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb69-19"><a href="pca---high-coverage.html#cb69-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 2&quot;</span>,</span>
<span id="cb69-20"><a href="pca---high-coverage.html#cb69-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom2,</span>
<span id="cb69-21"><a href="pca---high-coverage.html#cb69-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom2) <span class="sc">+</span></span>
<span id="cb69-22"><a href="pca---high-coverage.html#cb69-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom2<span class="sc">$</span>sample.no,   </span>
<span id="cb69-23"><a href="pca---high-coverage.html#cb69-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb69-24"><a href="pca---high-coverage.html#cb69-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb69-25"><a href="pca---high-coverage.html#cb69-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb69-26"><a href="pca---high-coverage.html#cb69-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-3" class="section level4 hasAnchor" number="3.5.3.3">
<h4><span class="header-section-number">3.5.3.3</span> Chromosome 3<a href="pca---high-coverage.html#chromosome-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="pca---high-coverage.html#cb70-1" tabindex="-1"></a>pca_chrom3 <span class="ot">&lt;-</span> </span>
<span id="cb70-2"><a href="pca---high-coverage.html#cb70-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb70-3"><a href="pca---high-coverage.html#cb70-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb70-4"><a href="pca---high-coverage.html#cb70-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb70-5"><a href="pca---high-coverage.html#cb70-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;3&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 6,041 SNPs according to chromosome 3
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 6,041
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 209503
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 15:59:59 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 15:59:59 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 15:59:59 2025    Done.</code></pre>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="pca---high-coverage.html#cb72-1" tabindex="-1"></a>chrom3 <span class="ot">&lt;-</span> </span>
<span id="cb72-2"><a href="pca---high-coverage.html#cb72-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom3<span class="sc">$</span>sample.id,</span>
<span id="cb72-3"><a href="pca---high-coverage.html#cb72-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom3<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb72-4"><a href="pca---high-coverage.html#cb72-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom3<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb72-5"><a href="pca---high-coverage.html#cb72-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb72-6"><a href="pca---high-coverage.html#cb72-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb72-7"><a href="pca---high-coverage.html#cb72-7" tabindex="-1"></a></span>
<span id="cb72-8"><a href="pca---high-coverage.html#cb72-8" tabindex="-1"></a>pc.percent_chrom3 <span class="ot">&lt;-</span> </span>
<span id="cb72-9"><a href="pca---high-coverage.html#cb72-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom3<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb72-10"><a href="pca---high-coverage.html#cb72-10" tabindex="-1"></a></span>
<span id="cb72-11"><a href="pca---high-coverage.html#cb72-11" tabindex="-1"></a>labelx_chrom3 <span class="ot">&lt;-</span> </span>
<span id="cb72-12"><a href="pca---high-coverage.html#cb72-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom3[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb72-13"><a href="pca---high-coverage.html#cb72-13" tabindex="-1"></a>labely_chrom3 <span class="ot">&lt;-</span> </span>
<span id="cb72-14"><a href="pca---high-coverage.html#cb72-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom3[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb72-15"><a href="pca---high-coverage.html#cb72-15" tabindex="-1"></a></span>
<span id="cb72-16"><a href="pca---high-coverage.html#cb72-16" tabindex="-1"></a>chrom3_plot <span class="ot">&lt;-</span></span>
<span id="cb72-17"><a href="pca---high-coverage.html#cb72-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom3, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb72-18"><a href="pca---high-coverage.html#cb72-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb72-19"><a href="pca---high-coverage.html#cb72-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 3&quot;</span>,</span>
<span id="cb72-20"><a href="pca---high-coverage.html#cb72-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom3,</span>
<span id="cb72-21"><a href="pca---high-coverage.html#cb72-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom3) <span class="sc">+</span></span>
<span id="cb72-22"><a href="pca---high-coverage.html#cb72-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom3<span class="sc">$</span>sample.no,   </span>
<span id="cb72-23"><a href="pca---high-coverage.html#cb72-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb72-24"><a href="pca---high-coverage.html#cb72-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb72-25"><a href="pca---high-coverage.html#cb72-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb72-26"><a href="pca---high-coverage.html#cb72-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-4" class="section level4 hasAnchor" number="3.5.3.4">
<h4><span class="header-section-number">3.5.3.4</span> Chromosome 4<a href="pca---high-coverage.html#chromosome-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="pca---high-coverage.html#cb73-1" tabindex="-1"></a>pca_chrom4 <span class="ot">&lt;-</span> </span>
<span id="cb73-2"><a href="pca---high-coverage.html#cb73-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb73-3"><a href="pca---high-coverage.html#cb73-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb73-4"><a href="pca---high-coverage.html#cb73-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb73-5"><a href="pca---high-coverage.html#cb73-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;4&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 7,295 SNPs according to chromosome 4
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 7,295
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 245953
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:01 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:01 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:01 2025    Done.</code></pre>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="pca---high-coverage.html#cb75-1" tabindex="-1"></a>chrom4 <span class="ot">&lt;-</span> </span>
<span id="cb75-2"><a href="pca---high-coverage.html#cb75-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom4<span class="sc">$</span>sample.id,</span>
<span id="cb75-3"><a href="pca---high-coverage.html#cb75-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom4<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb75-4"><a href="pca---high-coverage.html#cb75-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom4<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb75-5"><a href="pca---high-coverage.html#cb75-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb75-6"><a href="pca---high-coverage.html#cb75-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb75-7"><a href="pca---high-coverage.html#cb75-7" tabindex="-1"></a></span>
<span id="cb75-8"><a href="pca---high-coverage.html#cb75-8" tabindex="-1"></a>pc.percent_chrom4 <span class="ot">&lt;-</span> </span>
<span id="cb75-9"><a href="pca---high-coverage.html#cb75-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom4<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb75-10"><a href="pca---high-coverage.html#cb75-10" tabindex="-1"></a></span>
<span id="cb75-11"><a href="pca---high-coverage.html#cb75-11" tabindex="-1"></a>labelx_chrom4 <span class="ot">&lt;-</span> </span>
<span id="cb75-12"><a href="pca---high-coverage.html#cb75-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom4[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb75-13"><a href="pca---high-coverage.html#cb75-13" tabindex="-1"></a>labely_chrom4 <span class="ot">&lt;-</span> </span>
<span id="cb75-14"><a href="pca---high-coverage.html#cb75-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom4[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb75-15"><a href="pca---high-coverage.html#cb75-15" tabindex="-1"></a></span>
<span id="cb75-16"><a href="pca---high-coverage.html#cb75-16" tabindex="-1"></a>chrom4_plot <span class="ot">&lt;-</span></span>
<span id="cb75-17"><a href="pca---high-coverage.html#cb75-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom4, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb75-18"><a href="pca---high-coverage.html#cb75-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb75-19"><a href="pca---high-coverage.html#cb75-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 4&quot;</span>,</span>
<span id="cb75-20"><a href="pca---high-coverage.html#cb75-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom4,</span>
<span id="cb75-21"><a href="pca---high-coverage.html#cb75-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom4) <span class="sc">+</span></span>
<span id="cb75-22"><a href="pca---high-coverage.html#cb75-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom4<span class="sc">$</span>sample.no,   </span>
<span id="cb75-23"><a href="pca---high-coverage.html#cb75-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb75-24"><a href="pca---high-coverage.html#cb75-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb75-25"><a href="pca---high-coverage.html#cb75-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb75-26"><a href="pca---high-coverage.html#cb75-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-5" class="section level4 hasAnchor" number="3.5.3.5">
<h4><span class="header-section-number">3.5.3.5</span> Chromosome 5<a href="pca---high-coverage.html#chromosome-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="pca---high-coverage.html#cb76-1" tabindex="-1"></a>pca_chrom5 <span class="ot">&lt;-</span> </span>
<span id="cb76-2"><a href="pca---high-coverage.html#cb76-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb76-3"><a href="pca---high-coverage.html#cb76-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb76-4"><a href="pca---high-coverage.html#cb76-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb76-5"><a href="pca---high-coverage.html#cb76-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;5&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 3,448 SNPs according to chromosome 5
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 3,448
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 119026
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:02 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:02 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:02 2025    Done.</code></pre>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="pca---high-coverage.html#cb78-1" tabindex="-1"></a>chrom5 <span class="ot">&lt;-</span> </span>
<span id="cb78-2"><a href="pca---high-coverage.html#cb78-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom5<span class="sc">$</span>sample.id,</span>
<span id="cb78-3"><a href="pca---high-coverage.html#cb78-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom5<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb78-4"><a href="pca---high-coverage.html#cb78-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom5<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb78-5"><a href="pca---high-coverage.html#cb78-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb78-6"><a href="pca---high-coverage.html#cb78-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb78-7"><a href="pca---high-coverage.html#cb78-7" tabindex="-1"></a></span>
<span id="cb78-8"><a href="pca---high-coverage.html#cb78-8" tabindex="-1"></a>pc.percent_chrom5 <span class="ot">&lt;-</span> </span>
<span id="cb78-9"><a href="pca---high-coverage.html#cb78-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom5<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb78-10"><a href="pca---high-coverage.html#cb78-10" tabindex="-1"></a></span>
<span id="cb78-11"><a href="pca---high-coverage.html#cb78-11" tabindex="-1"></a>labelx_chrom5 <span class="ot">&lt;-</span> </span>
<span id="cb78-12"><a href="pca---high-coverage.html#cb78-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom5[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb78-13"><a href="pca---high-coverage.html#cb78-13" tabindex="-1"></a>labely_chrom5 <span class="ot">&lt;-</span> </span>
<span id="cb78-14"><a href="pca---high-coverage.html#cb78-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom5[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb78-15"><a href="pca---high-coverage.html#cb78-15" tabindex="-1"></a></span>
<span id="cb78-16"><a href="pca---high-coverage.html#cb78-16" tabindex="-1"></a>chrom5_plot <span class="ot">&lt;-</span></span>
<span id="cb78-17"><a href="pca---high-coverage.html#cb78-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom5, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb78-18"><a href="pca---high-coverage.html#cb78-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb78-19"><a href="pca---high-coverage.html#cb78-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 5&quot;</span>,</span>
<span id="cb78-20"><a href="pca---high-coverage.html#cb78-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom5,</span>
<span id="cb78-21"><a href="pca---high-coverage.html#cb78-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom5) <span class="sc">+</span></span>
<span id="cb78-22"><a href="pca---high-coverage.html#cb78-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom5<span class="sc">$</span>sample.no,   </span>
<span id="cb78-23"><a href="pca---high-coverage.html#cb78-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb78-24"><a href="pca---high-coverage.html#cb78-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb78-25"><a href="pca---high-coverage.html#cb78-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb78-26"><a href="pca---high-coverage.html#cb78-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-6" class="section level4 hasAnchor" number="3.5.3.6">
<h4><span class="header-section-number">3.5.3.6</span> Chromosome 6<a href="pca---high-coverage.html#chromosome-6" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="pca---high-coverage.html#cb79-1" tabindex="-1"></a>pca_chrom6 <span class="ot">&lt;-</span> </span>
<span id="cb79-2"><a href="pca---high-coverage.html#cb79-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb79-3"><a href="pca---high-coverage.html#cb79-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb79-4"><a href="pca---high-coverage.html#cb79-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb79-5"><a href="pca---high-coverage.html#cb79-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;6&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 5,258 SNPs according to chromosome 6
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 5,258
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 172351
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:04 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:04 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:04 2025    Done.</code></pre>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="pca---high-coverage.html#cb81-1" tabindex="-1"></a>chrom6 <span class="ot">&lt;-</span> </span>
<span id="cb81-2"><a href="pca---high-coverage.html#cb81-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom6<span class="sc">$</span>sample.id,</span>
<span id="cb81-3"><a href="pca---high-coverage.html#cb81-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom6<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb81-4"><a href="pca---high-coverage.html#cb81-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom6<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb81-5"><a href="pca---high-coverage.html#cb81-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb81-6"><a href="pca---high-coverage.html#cb81-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb81-7"><a href="pca---high-coverage.html#cb81-7" tabindex="-1"></a></span>
<span id="cb81-8"><a href="pca---high-coverage.html#cb81-8" tabindex="-1"></a>pc.percent_chrom6 <span class="ot">&lt;-</span> </span>
<span id="cb81-9"><a href="pca---high-coverage.html#cb81-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom6<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb81-10"><a href="pca---high-coverage.html#cb81-10" tabindex="-1"></a></span>
<span id="cb81-11"><a href="pca---high-coverage.html#cb81-11" tabindex="-1"></a>labelx_chrom6 <span class="ot">&lt;-</span> </span>
<span id="cb81-12"><a href="pca---high-coverage.html#cb81-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom6[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb81-13"><a href="pca---high-coverage.html#cb81-13" tabindex="-1"></a>labely_chrom6 <span class="ot">&lt;-</span> </span>
<span id="cb81-14"><a href="pca---high-coverage.html#cb81-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom6[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb81-15"><a href="pca---high-coverage.html#cb81-15" tabindex="-1"></a></span>
<span id="cb81-16"><a href="pca---high-coverage.html#cb81-16" tabindex="-1"></a>chrom6_plot <span class="ot">&lt;-</span></span>
<span id="cb81-17"><a href="pca---high-coverage.html#cb81-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom6, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb81-18"><a href="pca---high-coverage.html#cb81-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb81-19"><a href="pca---high-coverage.html#cb81-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 6&quot;</span>,</span>
<span id="cb81-20"><a href="pca---high-coverage.html#cb81-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom6,</span>
<span id="cb81-21"><a href="pca---high-coverage.html#cb81-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom6) <span class="sc">+</span></span>
<span id="cb81-22"><a href="pca---high-coverage.html#cb81-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom6<span class="sc">$</span>sample.no,   </span>
<span id="cb81-23"><a href="pca---high-coverage.html#cb81-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb81-24"><a href="pca---high-coverage.html#cb81-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb81-25"><a href="pca---high-coverage.html#cb81-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb81-26"><a href="pca---high-coverage.html#cb81-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-7" class="section level4 hasAnchor" number="3.5.3.7">
<h4><span class="header-section-number">3.5.3.7</span> Chromosome 7<a href="pca---high-coverage.html#chromosome-7" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="pca---high-coverage.html#cb82-1" tabindex="-1"></a>pca_chrom7 <span class="ot">&lt;-</span> </span>
<span id="cb82-2"><a href="pca---high-coverage.html#cb82-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb82-3"><a href="pca---high-coverage.html#cb82-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb82-4"><a href="pca---high-coverage.html#cb82-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb82-5"><a href="pca---high-coverage.html#cb82-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;7&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 3,714 SNPs according to chromosome 7
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 3,714
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 118494
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:06 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:06 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:06 2025    Done.</code></pre>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="pca---high-coverage.html#cb84-1" tabindex="-1"></a>chrom7 <span class="ot">&lt;-</span> </span>
<span id="cb84-2"><a href="pca---high-coverage.html#cb84-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom7<span class="sc">$</span>sample.id,</span>
<span id="cb84-3"><a href="pca---high-coverage.html#cb84-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom7<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb84-4"><a href="pca---high-coverage.html#cb84-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom7<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb84-5"><a href="pca---high-coverage.html#cb84-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb84-6"><a href="pca---high-coverage.html#cb84-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb84-7"><a href="pca---high-coverage.html#cb84-7" tabindex="-1"></a></span>
<span id="cb84-8"><a href="pca---high-coverage.html#cb84-8" tabindex="-1"></a>pc.percent_chrom7 <span class="ot">&lt;-</span> </span>
<span id="cb84-9"><a href="pca---high-coverage.html#cb84-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom7<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb84-10"><a href="pca---high-coverage.html#cb84-10" tabindex="-1"></a></span>
<span id="cb84-11"><a href="pca---high-coverage.html#cb84-11" tabindex="-1"></a>labelx_chrom7 <span class="ot">&lt;-</span> </span>
<span id="cb84-12"><a href="pca---high-coverage.html#cb84-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom7[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb84-13"><a href="pca---high-coverage.html#cb84-13" tabindex="-1"></a>labely_chrom7 <span class="ot">&lt;-</span> </span>
<span id="cb84-14"><a href="pca---high-coverage.html#cb84-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom7[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb84-15"><a href="pca---high-coverage.html#cb84-15" tabindex="-1"></a></span>
<span id="cb84-16"><a href="pca---high-coverage.html#cb84-16" tabindex="-1"></a>chrom7_plot <span class="ot">&lt;-</span></span>
<span id="cb84-17"><a href="pca---high-coverage.html#cb84-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom7, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb84-18"><a href="pca---high-coverage.html#cb84-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb84-19"><a href="pca---high-coverage.html#cb84-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 7&quot;</span>,</span>
<span id="cb84-20"><a href="pca---high-coverage.html#cb84-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom7,</span>
<span id="cb84-21"><a href="pca---high-coverage.html#cb84-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom7) <span class="sc">+</span></span>
<span id="cb84-22"><a href="pca---high-coverage.html#cb84-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom7<span class="sc">$</span>sample.no,   </span>
<span id="cb84-23"><a href="pca---high-coverage.html#cb84-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb84-24"><a href="pca---high-coverage.html#cb84-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb84-25"><a href="pca---high-coverage.html#cb84-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb84-26"><a href="pca---high-coverage.html#cb84-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-8" class="section level4 hasAnchor" number="3.5.3.8">
<h4><span class="header-section-number">3.5.3.8</span> Chromosome 8<a href="pca---high-coverage.html#chromosome-8" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="pca---high-coverage.html#cb85-1" tabindex="-1"></a>pca_chrom8 <span class="ot">&lt;-</span> </span>
<span id="cb85-2"><a href="pca---high-coverage.html#cb85-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb85-3"><a href="pca---high-coverage.html#cb85-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb85-4"><a href="pca---high-coverage.html#cb85-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb85-5"><a href="pca---high-coverage.html#cb85-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;8&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 2,762 SNPs according to chromosome 8
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 2,762
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 84324
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:08 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:08 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:08 2025    Done.</code></pre>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="pca---high-coverage.html#cb87-1" tabindex="-1"></a>chrom8 <span class="ot">&lt;-</span> </span>
<span id="cb87-2"><a href="pca---high-coverage.html#cb87-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom8<span class="sc">$</span>sample.id,</span>
<span id="cb87-3"><a href="pca---high-coverage.html#cb87-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom8<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb87-4"><a href="pca---high-coverage.html#cb87-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom8<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb87-5"><a href="pca---high-coverage.html#cb87-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-6"><a href="pca---high-coverage.html#cb87-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb87-7"><a href="pca---high-coverage.html#cb87-7" tabindex="-1"></a></span>
<span id="cb87-8"><a href="pca---high-coverage.html#cb87-8" tabindex="-1"></a>pc.percent_chrom8 <span class="ot">&lt;-</span> </span>
<span id="cb87-9"><a href="pca---high-coverage.html#cb87-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom8<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb87-10"><a href="pca---high-coverage.html#cb87-10" tabindex="-1"></a></span>
<span id="cb87-11"><a href="pca---high-coverage.html#cb87-11" tabindex="-1"></a>labelx_chrom8 <span class="ot">&lt;-</span> </span>
<span id="cb87-12"><a href="pca---high-coverage.html#cb87-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom8[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb87-13"><a href="pca---high-coverage.html#cb87-13" tabindex="-1"></a>labely_chrom8 <span class="ot">&lt;-</span> </span>
<span id="cb87-14"><a href="pca---high-coverage.html#cb87-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom8[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb87-15"><a href="pca---high-coverage.html#cb87-15" tabindex="-1"></a></span>
<span id="cb87-16"><a href="pca---high-coverage.html#cb87-16" tabindex="-1"></a>chrom8_plot <span class="ot">&lt;-</span></span>
<span id="cb87-17"><a href="pca---high-coverage.html#cb87-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom8, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb87-18"><a href="pca---high-coverage.html#cb87-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb87-19"><a href="pca---high-coverage.html#cb87-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 8&quot;</span>,</span>
<span id="cb87-20"><a href="pca---high-coverage.html#cb87-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom8,</span>
<span id="cb87-21"><a href="pca---high-coverage.html#cb87-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom8) <span class="sc">+</span></span>
<span id="cb87-22"><a href="pca---high-coverage.html#cb87-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom8<span class="sc">$</span>sample.no,   </span>
<span id="cb87-23"><a href="pca---high-coverage.html#cb87-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb87-24"><a href="pca---high-coverage.html#cb87-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb87-25"><a href="pca---high-coverage.html#cb87-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb87-26"><a href="pca---high-coverage.html#cb87-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-9" class="section level4 hasAnchor" number="3.5.3.9">
<h4><span class="header-section-number">3.5.3.9</span> Chromosome 9<a href="pca---high-coverage.html#chromosome-9" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="pca---high-coverage.html#cb88-1" tabindex="-1"></a>pca_chrom9 <span class="ot">&lt;-</span> </span>
<span id="cb88-2"><a href="pca---high-coverage.html#cb88-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb88-3"><a href="pca---high-coverage.html#cb88-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb88-4"><a href="pca---high-coverage.html#cb88-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb88-5"><a href="pca---high-coverage.html#cb88-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;9&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 4,966 SNPs according to chromosome 9
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 4,966
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 142246
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:09 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:09 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:09 2025    Done.</code></pre>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="pca---high-coverage.html#cb90-1" tabindex="-1"></a>chrom9 <span class="ot">&lt;-</span> </span>
<span id="cb90-2"><a href="pca---high-coverage.html#cb90-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom9<span class="sc">$</span>sample.id,</span>
<span id="cb90-3"><a href="pca---high-coverage.html#cb90-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom9<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb90-4"><a href="pca---high-coverage.html#cb90-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom9<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb90-5"><a href="pca---high-coverage.html#cb90-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-6"><a href="pca---high-coverage.html#cb90-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb90-7"><a href="pca---high-coverage.html#cb90-7" tabindex="-1"></a></span>
<span id="cb90-8"><a href="pca---high-coverage.html#cb90-8" tabindex="-1"></a>pc.percent_chrom9 <span class="ot">&lt;-</span> </span>
<span id="cb90-9"><a href="pca---high-coverage.html#cb90-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom9<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb90-10"><a href="pca---high-coverage.html#cb90-10" tabindex="-1"></a></span>
<span id="cb90-11"><a href="pca---high-coverage.html#cb90-11" tabindex="-1"></a>labelx_chrom9 <span class="ot">&lt;-</span> </span>
<span id="cb90-12"><a href="pca---high-coverage.html#cb90-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom9[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb90-13"><a href="pca---high-coverage.html#cb90-13" tabindex="-1"></a>labely_chrom9 <span class="ot">&lt;-</span> </span>
<span id="cb90-14"><a href="pca---high-coverage.html#cb90-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom9[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb90-15"><a href="pca---high-coverage.html#cb90-15" tabindex="-1"></a></span>
<span id="cb90-16"><a href="pca---high-coverage.html#cb90-16" tabindex="-1"></a>chrom9_plot <span class="ot">&lt;-</span></span>
<span id="cb90-17"><a href="pca---high-coverage.html#cb90-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom9, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb90-18"><a href="pca---high-coverage.html#cb90-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb90-19"><a href="pca---high-coverage.html#cb90-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 9&quot;</span>,</span>
<span id="cb90-20"><a href="pca---high-coverage.html#cb90-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom9,</span>
<span id="cb90-21"><a href="pca---high-coverage.html#cb90-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom9) <span class="sc">+</span></span>
<span id="cb90-22"><a href="pca---high-coverage.html#cb90-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom9<span class="sc">$</span>sample.no,   </span>
<span id="cb90-23"><a href="pca---high-coverage.html#cb90-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb90-24"><a href="pca---high-coverage.html#cb90-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb90-25"><a href="pca---high-coverage.html#cb90-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb90-26"><a href="pca---high-coverage.html#cb90-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-10" class="section level4 hasAnchor" number="3.5.3.10">
<h4><span class="header-section-number">3.5.3.10</span> Chromosome 10<a href="pca---high-coverage.html#chromosome-10" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="pca---high-coverage.html#cb91-1" tabindex="-1"></a>pca_chrom10 <span class="ot">&lt;-</span> </span>
<span id="cb91-2"><a href="pca---high-coverage.html#cb91-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb91-3"><a href="pca---high-coverage.html#cb91-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb91-4"><a href="pca---high-coverage.html#cb91-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb91-5"><a href="pca---high-coverage.html#cb91-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;10&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 4,055 SNPs according to chromosome 10
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 4,055
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 133216
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:11 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:11 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:11 2025    Done.</code></pre>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="pca---high-coverage.html#cb93-1" tabindex="-1"></a>chrom10 <span class="ot">&lt;-</span> </span>
<span id="cb93-2"><a href="pca---high-coverage.html#cb93-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom10<span class="sc">$</span>sample.id,</span>
<span id="cb93-3"><a href="pca---high-coverage.html#cb93-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom10<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb93-4"><a href="pca---high-coverage.html#cb93-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom10<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb93-5"><a href="pca---high-coverage.html#cb93-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb93-6"><a href="pca---high-coverage.html#cb93-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb93-7"><a href="pca---high-coverage.html#cb93-7" tabindex="-1"></a></span>
<span id="cb93-8"><a href="pca---high-coverage.html#cb93-8" tabindex="-1"></a>pc.percent_chrom10 <span class="ot">&lt;-</span> </span>
<span id="cb93-9"><a href="pca---high-coverage.html#cb93-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom10<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb93-10"><a href="pca---high-coverage.html#cb93-10" tabindex="-1"></a></span>
<span id="cb93-11"><a href="pca---high-coverage.html#cb93-11" tabindex="-1"></a>labelx_chrom10 <span class="ot">&lt;-</span> </span>
<span id="cb93-12"><a href="pca---high-coverage.html#cb93-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom10[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb93-13"><a href="pca---high-coverage.html#cb93-13" tabindex="-1"></a>labely_chrom10 <span class="ot">&lt;-</span> </span>
<span id="cb93-14"><a href="pca---high-coverage.html#cb93-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom10[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb93-15"><a href="pca---high-coverage.html#cb93-15" tabindex="-1"></a></span>
<span id="cb93-16"><a href="pca---high-coverage.html#cb93-16" tabindex="-1"></a>chrom10_plot <span class="ot">&lt;-</span></span>
<span id="cb93-17"><a href="pca---high-coverage.html#cb93-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom10, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb93-18"><a href="pca---high-coverage.html#cb93-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb93-19"><a href="pca---high-coverage.html#cb93-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 10&quot;</span>,</span>
<span id="cb93-20"><a href="pca---high-coverage.html#cb93-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom10,</span>
<span id="cb93-21"><a href="pca---high-coverage.html#cb93-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom10) <span class="sc">+</span></span>
<span id="cb93-22"><a href="pca---high-coverage.html#cb93-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom10<span class="sc">$</span>sample.no,   </span>
<span id="cb93-23"><a href="pca---high-coverage.html#cb93-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb93-24"><a href="pca---high-coverage.html#cb93-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb93-25"><a href="pca---high-coverage.html#cb93-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb93-26"><a href="pca---high-coverage.html#cb93-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-11" class="section level4 hasAnchor" number="3.5.3.11">
<h4><span class="header-section-number">3.5.3.11</span> Chromosome 11<a href="pca---high-coverage.html#chromosome-11" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="pca---high-coverage.html#cb94-1" tabindex="-1"></a>pca_chrom11 <span class="ot">&lt;-</span> </span>
<span id="cb94-2"><a href="pca---high-coverage.html#cb94-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb94-3"><a href="pca---high-coverage.html#cb94-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb94-4"><a href="pca---high-coverage.html#cb94-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb94-5"><a href="pca---high-coverage.html#cb94-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;11&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 5,089 SNPs according to chromosome 11
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 5,089
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 167381
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:12 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:12 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:12 2025    Done.</code></pre>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="pca---high-coverage.html#cb96-1" tabindex="-1"></a>chrom11 <span class="ot">&lt;-</span> </span>
<span id="cb96-2"><a href="pca---high-coverage.html#cb96-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom11<span class="sc">$</span>sample.id,</span>
<span id="cb96-3"><a href="pca---high-coverage.html#cb96-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom11<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb96-4"><a href="pca---high-coverage.html#cb96-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom11<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb96-5"><a href="pca---high-coverage.html#cb96-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb96-6"><a href="pca---high-coverage.html#cb96-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb96-7"><a href="pca---high-coverage.html#cb96-7" tabindex="-1"></a></span>
<span id="cb96-8"><a href="pca---high-coverage.html#cb96-8" tabindex="-1"></a>pc.percent_chrom11 <span class="ot">&lt;-</span> </span>
<span id="cb96-9"><a href="pca---high-coverage.html#cb96-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom11<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb96-10"><a href="pca---high-coverage.html#cb96-10" tabindex="-1"></a></span>
<span id="cb96-11"><a href="pca---high-coverage.html#cb96-11" tabindex="-1"></a>labelx_chrom11 <span class="ot">&lt;-</span> </span>
<span id="cb96-12"><a href="pca---high-coverage.html#cb96-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom11[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb96-13"><a href="pca---high-coverage.html#cb96-13" tabindex="-1"></a>labely_chrom11 <span class="ot">&lt;-</span> </span>
<span id="cb96-14"><a href="pca---high-coverage.html#cb96-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom11[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb96-15"><a href="pca---high-coverage.html#cb96-15" tabindex="-1"></a></span>
<span id="cb96-16"><a href="pca---high-coverage.html#cb96-16" tabindex="-1"></a>chrom11_plot <span class="ot">&lt;-</span></span>
<span id="cb96-17"><a href="pca---high-coverage.html#cb96-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom11, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb96-18"><a href="pca---high-coverage.html#cb96-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb96-19"><a href="pca---high-coverage.html#cb96-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 11&quot;</span>,</span>
<span id="cb96-20"><a href="pca---high-coverage.html#cb96-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom11,</span>
<span id="cb96-21"><a href="pca---high-coverage.html#cb96-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom11) <span class="sc">+</span></span>
<span id="cb96-22"><a href="pca---high-coverage.html#cb96-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom11<span class="sc">$</span>sample.no,   </span>
<span id="cb96-23"><a href="pca---high-coverage.html#cb96-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb96-24"><a href="pca---high-coverage.html#cb96-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb96-25"><a href="pca---high-coverage.html#cb96-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb96-26"><a href="pca---high-coverage.html#cb96-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-12" class="section level4 hasAnchor" number="3.5.3.12">
<h4><span class="header-section-number">3.5.3.12</span> Chromosome 12<a href="pca---high-coverage.html#chromosome-12" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="pca---high-coverage.html#cb97-1" tabindex="-1"></a>pca_chrom12 <span class="ot">&lt;-</span> </span>
<span id="cb97-2"><a href="pca---high-coverage.html#cb97-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb97-3"><a href="pca---high-coverage.html#cb97-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb97-4"><a href="pca---high-coverage.html#cb97-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb97-5"><a href="pca---high-coverage.html#cb97-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;12&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 6,946 SNPs according to chromosome 12
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 6,946
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 222110
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:14 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:14 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:14 2025    Done.</code></pre>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="pca---high-coverage.html#cb99-1" tabindex="-1"></a>chrom12 <span class="ot">&lt;-</span> </span>
<span id="cb99-2"><a href="pca---high-coverage.html#cb99-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom12<span class="sc">$</span>sample.id,</span>
<span id="cb99-3"><a href="pca---high-coverage.html#cb99-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom12<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb99-4"><a href="pca---high-coverage.html#cb99-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom12<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb99-5"><a href="pca---high-coverage.html#cb99-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb99-6"><a href="pca---high-coverage.html#cb99-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb99-7"><a href="pca---high-coverage.html#cb99-7" tabindex="-1"></a></span>
<span id="cb99-8"><a href="pca---high-coverage.html#cb99-8" tabindex="-1"></a>pc.percent_chrom12 <span class="ot">&lt;-</span> </span>
<span id="cb99-9"><a href="pca---high-coverage.html#cb99-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom12<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb99-10"><a href="pca---high-coverage.html#cb99-10" tabindex="-1"></a></span>
<span id="cb99-11"><a href="pca---high-coverage.html#cb99-11" tabindex="-1"></a>labelx_chrom12 <span class="ot">&lt;-</span> </span>
<span id="cb99-12"><a href="pca---high-coverage.html#cb99-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom12[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb99-13"><a href="pca---high-coverage.html#cb99-13" tabindex="-1"></a>labely_chrom12 <span class="ot">&lt;-</span> </span>
<span id="cb99-14"><a href="pca---high-coverage.html#cb99-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom12[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb99-15"><a href="pca---high-coverage.html#cb99-15" tabindex="-1"></a></span>
<span id="cb99-16"><a href="pca---high-coverage.html#cb99-16" tabindex="-1"></a>chrom12_plot <span class="ot">&lt;-</span></span>
<span id="cb99-17"><a href="pca---high-coverage.html#cb99-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom12, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb99-18"><a href="pca---high-coverage.html#cb99-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb99-19"><a href="pca---high-coverage.html#cb99-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 12&quot;</span>,</span>
<span id="cb99-20"><a href="pca---high-coverage.html#cb99-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom12,</span>
<span id="cb99-21"><a href="pca---high-coverage.html#cb99-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom12) <span class="sc">+</span></span>
<span id="cb99-22"><a href="pca---high-coverage.html#cb99-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom12<span class="sc">$</span>sample.no,   </span>
<span id="cb99-23"><a href="pca---high-coverage.html#cb99-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb99-24"><a href="pca---high-coverage.html#cb99-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb99-25"><a href="pca---high-coverage.html#cb99-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb99-26"><a href="pca---high-coverage.html#cb99-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-13" class="section level4 hasAnchor" number="3.5.3.13">
<h4><span class="header-section-number">3.5.3.13</span> Chromosome 13<a href="pca---high-coverage.html#chromosome-13" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="pca---high-coverage.html#cb100-1" tabindex="-1"></a>pca_chrom13 <span class="ot">&lt;-</span> </span>
<span id="cb100-2"><a href="pca---high-coverage.html#cb100-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb100-3"><a href="pca---high-coverage.html#cb100-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb100-4"><a href="pca---high-coverage.html#cb100-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb100-5"><a href="pca---high-coverage.html#cb100-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;13&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 5,692 SNPs according to chromosome 13
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 5,692
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 187299
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:15 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:15 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:15 2025    Done.</code></pre>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="pca---high-coverage.html#cb102-1" tabindex="-1"></a>chrom13 <span class="ot">&lt;-</span> </span>
<span id="cb102-2"><a href="pca---high-coverage.html#cb102-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom13<span class="sc">$</span>sample.id,</span>
<span id="cb102-3"><a href="pca---high-coverage.html#cb102-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom13<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb102-4"><a href="pca---high-coverage.html#cb102-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom13<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb102-5"><a href="pca---high-coverage.html#cb102-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb102-6"><a href="pca---high-coverage.html#cb102-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb102-7"><a href="pca---high-coverage.html#cb102-7" tabindex="-1"></a></span>
<span id="cb102-8"><a href="pca---high-coverage.html#cb102-8" tabindex="-1"></a>pc.percent_chrom13 <span class="ot">&lt;-</span> </span>
<span id="cb102-9"><a href="pca---high-coverage.html#cb102-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom13<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb102-10"><a href="pca---high-coverage.html#cb102-10" tabindex="-1"></a></span>
<span id="cb102-11"><a href="pca---high-coverage.html#cb102-11" tabindex="-1"></a>labelx_chrom13 <span class="ot">&lt;-</span> </span>
<span id="cb102-12"><a href="pca---high-coverage.html#cb102-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom13[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb102-13"><a href="pca---high-coverage.html#cb102-13" tabindex="-1"></a>labely_chrom13 <span class="ot">&lt;-</span> </span>
<span id="cb102-14"><a href="pca---high-coverage.html#cb102-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom13[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb102-15"><a href="pca---high-coverage.html#cb102-15" tabindex="-1"></a></span>
<span id="cb102-16"><a href="pca---high-coverage.html#cb102-16" tabindex="-1"></a>chrom13_plot <span class="ot">&lt;-</span></span>
<span id="cb102-17"><a href="pca---high-coverage.html#cb102-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom13, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb102-18"><a href="pca---high-coverage.html#cb102-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb102-19"><a href="pca---high-coverage.html#cb102-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 13&quot;</span>,</span>
<span id="cb102-20"><a href="pca---high-coverage.html#cb102-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom13,</span>
<span id="cb102-21"><a href="pca---high-coverage.html#cb102-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom13) <span class="sc">+</span></span>
<span id="cb102-22"><a href="pca---high-coverage.html#cb102-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom13<span class="sc">$</span>sample.no,   </span>
<span id="cb102-23"><a href="pca---high-coverage.html#cb102-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb102-24"><a href="pca---high-coverage.html#cb102-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb102-25"><a href="pca---high-coverage.html#cb102-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb102-26"><a href="pca---high-coverage.html#cb102-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-14" class="section level4 hasAnchor" number="3.5.3.14">
<h4><span class="header-section-number">3.5.3.14</span> Chromosome 14<a href="pca---high-coverage.html#chromosome-14" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="pca---high-coverage.html#cb103-1" tabindex="-1"></a>pca_chrom14 <span class="ot">&lt;-</span> </span>
<span id="cb103-2"><a href="pca---high-coverage.html#cb103-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb103-3"><a href="pca---high-coverage.html#cb103-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb103-4"><a href="pca---high-coverage.html#cb103-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb103-5"><a href="pca---high-coverage.html#cb103-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;14&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 5,753 SNPs according to chromosome 14
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 5,753
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 182618
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:17 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:17 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:17 2025    Done.</code></pre>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="pca---high-coverage.html#cb105-1" tabindex="-1"></a>chrom14 <span class="ot">&lt;-</span> </span>
<span id="cb105-2"><a href="pca---high-coverage.html#cb105-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom14<span class="sc">$</span>sample.id,</span>
<span id="cb105-3"><a href="pca---high-coverage.html#cb105-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom14<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb105-4"><a href="pca---high-coverage.html#cb105-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom14<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb105-5"><a href="pca---high-coverage.html#cb105-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb105-6"><a href="pca---high-coverage.html#cb105-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb105-7"><a href="pca---high-coverage.html#cb105-7" tabindex="-1"></a></span>
<span id="cb105-8"><a href="pca---high-coverage.html#cb105-8" tabindex="-1"></a>pc.percent_chrom14 <span class="ot">&lt;-</span> </span>
<span id="cb105-9"><a href="pca---high-coverage.html#cb105-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom14<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb105-10"><a href="pca---high-coverage.html#cb105-10" tabindex="-1"></a></span>
<span id="cb105-11"><a href="pca---high-coverage.html#cb105-11" tabindex="-1"></a>labelx_chrom14 <span class="ot">&lt;-</span> </span>
<span id="cb105-12"><a href="pca---high-coverage.html#cb105-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom14[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb105-13"><a href="pca---high-coverage.html#cb105-13" tabindex="-1"></a>labely_chrom14 <span class="ot">&lt;-</span> </span>
<span id="cb105-14"><a href="pca---high-coverage.html#cb105-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom14[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb105-15"><a href="pca---high-coverage.html#cb105-15" tabindex="-1"></a></span>
<span id="cb105-16"><a href="pca---high-coverage.html#cb105-16" tabindex="-1"></a>chrom14_plot <span class="ot">&lt;-</span></span>
<span id="cb105-17"><a href="pca---high-coverage.html#cb105-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom14, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb105-18"><a href="pca---high-coverage.html#cb105-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb105-19"><a href="pca---high-coverage.html#cb105-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 14&quot;</span>,</span>
<span id="cb105-20"><a href="pca---high-coverage.html#cb105-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom14,</span>
<span id="cb105-21"><a href="pca---high-coverage.html#cb105-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom14) <span class="sc">+</span></span>
<span id="cb105-22"><a href="pca---high-coverage.html#cb105-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom14<span class="sc">$</span>sample.no,   </span>
<span id="cb105-23"><a href="pca---high-coverage.html#cb105-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb105-24"><a href="pca---high-coverage.html#cb105-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb105-25"><a href="pca---high-coverage.html#cb105-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb105-26"><a href="pca---high-coverage.html#cb105-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
<div id="chromosome-16" class="section level4 hasAnchor" number="3.5.3.15">
<h4><span class="header-section-number">3.5.3.15</span> Chromosome 16<a href="pca---high-coverage.html#chromosome-16" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="pca---high-coverage.html#cb106-1" tabindex="-1"></a>pca_chrom16 <span class="ot">&lt;-</span> </span>
<span id="cb106-2"><a href="pca---high-coverage.html#cb106-2" tabindex="-1"></a>  <span class="fu">snpgdsPCA</span>(geno, </span>
<span id="cb106-3"><a href="pca---high-coverage.html#cb106-3" tabindex="-1"></a>            <span class="at">snp.id =</span> snpset.id, </span>
<span id="cb106-4"><a href="pca---high-coverage.html#cb106-4" tabindex="-1"></a>            <span class="at">num.thread =</span> <span class="dv">2</span>, </span>
<span id="cb106-5"><a href="pca---high-coverage.html#cb106-5" tabindex="-1"></a>            <span class="at">autosome.only =</span> <span class="st">&quot;16&quot;</span>)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Keeping 6,427 SNPs according to chromosome 16
## Excluding 0 SNP (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 20
##     # of SNPs: 6,427
##     using 2 threads
##     # of principal components: 32
## PCA:    the sum of all selected genotypes (0,1,2) = 223661
## CPU capabilities: Double-Precision SSE2
## Wed Mar 19 16:00:18 2025    (internal increment: 65328)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 1s
## Wed Mar 19 16:00:19 2025    Begin (eigenvalues and eigenvectors)
## Wed Mar 19 16:00:19 2025    Done.</code></pre>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="pca---high-coverage.html#cb108-1" tabindex="-1"></a>chrom16 <span class="ot">&lt;-</span> </span>
<span id="cb108-2"><a href="pca---high-coverage.html#cb108-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pca_chrom16<span class="sc">$</span>sample.id,</span>
<span id="cb108-3"><a href="pca---high-coverage.html#cb108-3" tabindex="-1"></a>             <span class="at">EV1 =</span> pca_chrom16<span class="sc">$</span>eigenvect[,<span class="dv">1</span>],    </span>
<span id="cb108-4"><a href="pca---high-coverage.html#cb108-4" tabindex="-1"></a>             <span class="at">EV2 =</span> pca_chrom16<span class="sc">$</span>eigenvect[,<span class="dv">2</span>],    </span>
<span id="cb108-5"><a href="pca---high-coverage.html#cb108-5" tabindex="-1"></a>             <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb108-6"><a href="pca---high-coverage.html#cb108-6" tabindex="-1"></a>  <span class="fu">merge</span>(., meta, <span class="at">by.x=</span><span class="st">&quot;sample.id&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;seq.id&quot;</span>)</span>
<span id="cb108-7"><a href="pca---high-coverage.html#cb108-7" tabindex="-1"></a></span>
<span id="cb108-8"><a href="pca---high-coverage.html#cb108-8" tabindex="-1"></a>pc.percent_chrom16 <span class="ot">&lt;-</span> </span>
<span id="cb108-9"><a href="pca---high-coverage.html#cb108-9" tabindex="-1"></a>  <span class="fu">round</span>(pca_chrom16<span class="sc">$</span>varprop<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb108-10"><a href="pca---high-coverage.html#cb108-10" tabindex="-1"></a></span>
<span id="cb108-11"><a href="pca---high-coverage.html#cb108-11" tabindex="-1"></a>labelx_chrom16 <span class="ot">&lt;-</span> </span>
<span id="cb108-12"><a href="pca---high-coverage.html#cb108-12" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom16[<span class="dv">1</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb108-13"><a href="pca---high-coverage.html#cb108-13" tabindex="-1"></a>labely_chrom16 <span class="ot">&lt;-</span> </span>
<span id="cb108-14"><a href="pca---high-coverage.html#cb108-14" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;EV1 (&quot;</span>, pc.percent_chrom16[<span class="dv">2</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb108-15"><a href="pca---high-coverage.html#cb108-15" tabindex="-1"></a></span>
<span id="cb108-16"><a href="pca---high-coverage.html#cb108-16" tabindex="-1"></a>chrom16_plot <span class="ot">&lt;-</span></span>
<span id="cb108-17"><a href="pca---high-coverage.html#cb108-17" tabindex="-1"></a>  <span class="fu">ggplot</span>(chrom16, <span class="fu">aes</span>(<span class="at">x=</span>EV1, <span class="at">y=</span>EV2)) <span class="sc">+</span></span>
<span id="cb108-18"><a href="pca---high-coverage.html#cb108-18" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb108-19"><a href="pca---high-coverage.html#cb108-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Chromosome 15&quot;</span>,</span>
<span id="cb108-20"><a href="pca---high-coverage.html#cb108-20" tabindex="-1"></a>       <span class="at">x=</span>labelx_chrom16,</span>
<span id="cb108-21"><a href="pca---high-coverage.html#cb108-21" tabindex="-1"></a>       <span class="at">y=</span>labely_chrom16) <span class="sc">+</span></span>
<span id="cb108-22"><a href="pca---high-coverage.html#cb108-22" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">label=</span>chrom16<span class="sc">$</span>sample.no,   </span>
<span id="cb108-23"><a href="pca---high-coverage.html#cb108-23" tabindex="-1"></a>                  <span class="at">max.overlaps=</span><span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb108-24"><a href="pca---high-coverage.html#cb108-24" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb108-25"><a href="pca---high-coverage.html#cb108-25" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(),</span>
<span id="cb108-26"><a href="pca---high-coverage.html#cb108-26" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span>
<span id="cb108-27"><a href="pca---high-coverage.html#cb108-27" tabindex="-1"></a>chrom16_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-40-1.png" width="672" /></p>
</div>
<div id="view" class="section level4 hasAnchor" number="3.5.3.16">
<h4><span class="header-section-number">3.5.3.16</span> View<a href="pca---high-coverage.html#view" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="pca---high-coverage.html#cb109-1" tabindex="-1"></a>chrom1_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-1.png" width="672" /></p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="pca---high-coverage.html#cb110-1" tabindex="-1"></a>chrom2_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-2.png" width="672" /></p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="pca---high-coverage.html#cb111-1" tabindex="-1"></a>chrom3_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-3.png" width="672" /></p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="pca---high-coverage.html#cb112-1" tabindex="-1"></a>chrom4_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-4.png" width="672" /></p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="pca---high-coverage.html#cb113-1" tabindex="-1"></a>chrom5_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-5.png" width="672" /></p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="pca---high-coverage.html#cb114-1" tabindex="-1"></a>chrom6_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-6.png" width="672" /></p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="pca---high-coverage.html#cb115-1" tabindex="-1"></a>chrom7_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-7.png" width="672" /></p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="pca---high-coverage.html#cb116-1" tabindex="-1"></a>chrom8_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-8.png" width="672" /></p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="pca---high-coverage.html#cb117-1" tabindex="-1"></a>chrom9_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-9.png" width="672" /></p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="pca---high-coverage.html#cb118-1" tabindex="-1"></a>chrom10_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-10.png" width="672" /></p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="pca---high-coverage.html#cb119-1" tabindex="-1"></a>chrom11_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-11.png" width="672" /></p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="pca---high-coverage.html#cb120-1" tabindex="-1"></a>chrom12_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-12.png" width="672" /></p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="pca---high-coverage.html#cb121-1" tabindex="-1"></a>chrom13_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-13.png" width="672" /></p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="pca---high-coverage.html#cb122-1" tabindex="-1"></a>chrom14_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-14.png" width="672" /></p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="pca---high-coverage.html#cb123-1" tabindex="-1"></a>chrom16_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-41-15.png" width="672" /></p>
</div>
</div>
</div>
<div id="relatedness" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Relatedness<a href="pca---high-coverage.html#relatedness" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Drop SNPs with missing info in &gt;5% individuals.<br />
Exclude rare SNPs, &lt;5% MAF.
k0 is high for all individuals - no relatedness</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="pca---high-coverage.html#cb124-1" tabindex="-1"></a>sample.id <span class="ot">&lt;-</span> </span>
<span id="cb124-2"><a href="pca---high-coverage.html#cb124-2" tabindex="-1"></a>  <span class="fu">read.gdsn</span>(<span class="fu">index.gdsn</span>(geno, <span class="st">&quot;sample.id&quot;</span>))</span>
<span id="cb124-3"><a href="pca---high-coverage.html#cb124-3" tabindex="-1"></a></span>
<span id="cb124-4"><a href="pca---high-coverage.html#cb124-4" tabindex="-1"></a>ibd <span class="ot">&lt;-</span> </span>
<span id="cb124-5"><a href="pca---high-coverage.html#cb124-5" tabindex="-1"></a>  <span class="fu">snpgdsIBDMoM</span>(geno, </span>
<span id="cb124-6"><a href="pca---high-coverage.html#cb124-6" tabindex="-1"></a>               <span class="at">sample.id=</span>sample.id, <span class="at">snp.id=</span>snpset.id,</span>
<span id="cb124-7"><a href="pca---high-coverage.html#cb124-7" tabindex="-1"></a>               <span class="at">maf=</span><span class="fl">0.05</span>, <span class="at">missing.rate=</span><span class="fl">0.05</span>, </span>
<span id="cb124-8"><a href="pca---high-coverage.html#cb124-8" tabindex="-1"></a>               <span class="at">num.thread=</span><span class="dv">2</span>, <span class="at">autosome.only=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## IBD analysis (PLINK method of moment) on genotypes:
## Excluding 50,189 SNPs (monomorphic: TRUE, MAF: 0.05, missing rate: 0.05)
##     # of samples: 20
##     # of SNPs: 31,108
##     using 2 threads
## PLINK IBD:    the sum of all selected genotypes (0,1,2) = 777386
## Wed Mar 19 16:00:40 2025    (internal increment: 65536)
## [..................................................]  0%, ETC: ---        [==================================================] 100%, completed, 0s
## Wed Mar 19 16:00:40 2025    Done.</code></pre>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="pca---high-coverage.html#cb126-1" tabindex="-1"></a>ibd.coeff <span class="ot">&lt;-</span> <span class="fu">snpgdsIBDSelection</span>(ibd)</span>
<span id="cb126-2"><a href="pca---high-coverage.html#cb126-2" tabindex="-1"></a></span>
<span id="cb126-3"><a href="pca---high-coverage.html#cb126-3" tabindex="-1"></a><span class="fu">plot</span>(ibd.coeff<span class="sc">$</span>k0, ibd.coeff<span class="sc">$</span>k1, <span class="at">xlim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb126-4"><a href="pca---high-coverage.html#cb126-4" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;k0&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;k1&quot;</span>, <span class="at">main=</span><span class="st">&quot;YRI samples (MoM)&quot;</span>)</span>
<span id="cb126-5"><a href="pca---high-coverage.html#cb126-5" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>), <span class="at">col=</span><span class="st">&quot;red&quot;</span>, <span class="at">lty=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-42-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="tidy-data-exploratory.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="density-plot-filter-metrics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section",
    "scroll_highlight": true
  },
  "info": false
});
});
</script>

</body>

</html>
